---
bibliography: references.bib
csl: genomebiology.csl
site: "bookdown::bookdown_site"
output:
  bookdown::pdf_document2:
    toc: FALSE
---

```{r preliminary-setup, include=FALSE}
	# Load libraries
	#biocLite libraries
  #install.packages("BiocManager")
 #BiocManager::install(version = "3.11")
  #BiocManager::install(c("ggtree","treeio","impute","DESeq2","topGO","goseq","GO.db","vsn"))
	library(ggtree)
  library(treeio)
  library(impute)
  library(DESeq2)
  library(topGO)
  library(goseq)
  library(GO.db)
  library(vsn)
	
	#devtools
  # library( devtools )
	library(hutan)
		# devtools::install_github("caseywdunn/hutan")
	library("agalmar")
    #devtools::install_github("caseywdunn/agalmar")
 
	#CRAN libraries
  library(ggplot2)
	library(bookdown)
  library(knitr)
	library(fields)
	library(picante)
	library(RColorBrewer)
	library(SparseM)
	library(tidyverse)
  library(digest)
  library(parallel)
  library(magrittr)
  library(PhylogeneticEM)
  library(GGally)
  library(geiger)
  library(gridExtra)
  library(emmeans)
  library(FSA)
  library(lme4)
  library(ggpubr)
  library(corrplot)
  library(kableExtra)
  library(ggpubr)
  library(ggrepel)
  library(gtools)

  	## The minimum number of counts to pass gene sampling criteria
	min_count <- 1 
	
	## p value cutoff for evaluating differential expression significance
	p_value_threshold <- 0.05 
	
	#
	focal_species <- c( "Agalma elegans", "Nanomia bijuga", "Bargmannia elongata", "Frillagalma vityazi", "Diphyes dispar", "Physalia physalis", "Apolemia lanosa" )
	
## Gene tree parameters
	
	### The default branch length pace holder used in the gene tree inference software
	default_length_val <- 1e-06   
	
	### Exclude trees with branches that exceed length threshold
	edge_length_max_threshold <- 2

	### Exclude trees that exceed specified fraction of branches with default values
	fraction_default_max_threshold <- 0.25
	
	### The minimum number of tips with expression data for a tree to be considered 
	min_tips <- 3
	
	### The maximum root depth of gene trees, where the root of the species tree is 1
	### Gene trees that exceed this threshold are removed
	max_root_depth <- 5
	
	# Set system computational parameters
	cores <- detectCores() - 1
	if ( cores < 1 ) {
		cores <- 1
	}

	set.seed( 23456 )
	
	cores <- detectCores() - 1

```

```{r load-data, include=FALSE}
#Generated by ancestral_trait_recon.R

  load("manuscript_checkpoint_ace.RData")
  

	source( "functions.R" )
```

```{r summarize-trees, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}
gene_tree_summary<-list()

#loop the summary over all of the col values of interest
for(i in 1:length(focal_tpm_values)){
gene_tree_summary[[i]]<-summarize_trees(gene_trees_ace_calibrated_pruned,focal_tpm_values[i])
}

#put all of the rows together
gene_tree_summary<-gene_tree_summary %>% dplyr::bind_rows() %>% arrange(gene)

```

```{r blast by gene_tree,verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}
#this gives you gene trees with the top blast hit for that gene tree

genes_to_tips =
		nodes_ace %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,blast_lookup, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( blast_hit =names(which.max(table(blast_hit))) ) %>% # A convoluted way to get the most frequent blast hit for each gene
	  dplyr::mutate( blast_hit = stringr::str_replace(blast_hit, "swissprot|", ""))

```

```{r subtree, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

phyldog_species_numbered_tree_original <-phyldog_species_numbered_tree	
	
	# Need to get the phyldog node numbers for tips from elsewhere
	node_map = 
		nodes_raw %>% 
		dplyr::select( S, species ) %>% 
		na.omit() %>% 
		dplyr::distinct()
	
	# Reorder to match tree tips
	node_map = 
		node_map[ match( phyldog_species_numbered_tree$tip.label, node_map$species ), ]
	
#	tiplabels( node_map$S )
	
#	kable(node_map, row.names=FALSE)
	
		observed_phyldog_nodes = 
		unique( c( edges_ace$S_parent, edges_ace$S_child ) )
	
	phyldog_species_numbered_tree = 
		ape::drop.tip( 
			phyldog_species_numbered_tree, 
			which( ! node_map$S %in% observed_phyldog_nodes )
		)
	
	node_map = node_map[ node_map$S %in% observed_phyldog_nodes, ]
	
	# make sure order matches tree tips
	node_map = 
		node_map[ match( phyldog_species_numbered_tree$tip.label, node_map$species ), ]

	# Create a vector of phyldog node labels in the order of the ape nodes
	S = c( node_map$S, phyldog_species_numbered_tree$node.label ) 
	
		##Have a human readable name for each of the branches instead of the numbers given by S
S_new<-data.frame( S_new=c("F","G","H","I","J","K","L","X","A","B","C","D","E" ),S=S)

	

```

```{r match edges, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE }
  edges_ace_original<- edges_ace

	# Only retain edges where both parent and child is a speciation event
	edges_ace %<>% dplyr::filter( (Ev_parent=="S") & (Ev_child=="S") )
	#dim(edges_ace)
	
	# Create an edge matrix that is in terms of phyldog node numbers rather than ape node numbers
	phyldog_edges = S[ as.vector(phyldog_species_numbered_tree$edge) ]
	dim(phyldog_edges) = dim( phyldog_species_numbered_tree$edge )
	
	# For each row in edges_ace, identify the expected parent in the species tree for S_child
	expected_parent = phyldog_edges[ match(edges_ace$S_child, phyldog_edges[,2]), 1 ]
	
	# Filter to retain only the edges where S_parent matches the expected parent of S_child in the species tree
	
	edges_ace %<>% dplyr::filter( (S_parent==expected_parent) )
	#dim(edges_ace)
	
	# Create a tidy data frame of changes in expression
	edges_tidy = edges_ace %>% dplyr::select( gene_tree, S_child, length, node_depth_child, node_age_child, node_age_parent, node_parent, node_child, terminal )
	edges_tidy = cbind( edges_tidy, edges_ace[, grepl("scaled", names(edges_ace))] )
	colnames(edges_tidy) = colnames(edges_tidy) %>% sub( "_scaled", "", .)
	edges_tidy %<>% tidyr::gather( names(edges_tidy)[10:18], key = "treatment", value = "change" )
	
	# Change child node to factor
	edges_tidy %<>% dplyr::mutate( S_child = as.factor(S_child) )
	
	# Remove tau for now since it is on a different scale
	edges_tidy %<>% filter(treatment != "tau")
	
	# Remove rows with missing values
	edges_tidy %<>% na.omit()
	
	# Add blast reports
	edges_tidy %<>%
		dplyr::left_join( genes_to_tips, key="gene_tree")
	
	edges_tidy$change_abs = abs( edges_tidy$change )
	
	#remove zooid values from branches where there shouldn't be any (due to topological issues)
	
	edges_tidy<-edges_tidy %>% group_by(S_child,treatment) %>% filter(n()>300) %>% ungroup()
	
```

```{r match edges sim, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

edges_sim_original<-edges_sim

genes_to_tips_sim =
		nodes_sim %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,blast_lookup, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( blast_hit =names(which.max(table(blast_hit))) ) %>% # A convoluted way to get the most frequent blast hit for each gene
		dplyr::mutate( blast_hit = stringr::str_replace(blast_hit, "swissprot|", ""))

	# Only retain edges where both parent and child is a speciation event
	edges_sim %<>% dplyr::filter( (Ev_parent=="S") & (Ev_child=="S") )
	#dim(edges_sim)
	
	# Create an edge matrix that is in terms of phyldog node numbers rather than ape node numbers
	phyldog_edges = S[ as.vector(phyldog_species_numbered_tree$edge) ]
	dim(phyldog_edges) = dim( phyldog_species_numbered_tree$edge )
	
	# For each row in edges_sim, identify the expected parent in the species tree for S_child
	expected_parent = phyldog_edges[ match(edges_sim$S_child, phyldog_edges[,2]), 1 ]
	
	# Filter to retain only the edges where S_parent matches the expected parent of S_child in the species tree
	
	edges_sim %<>% dplyr::filter( S_parent==expected_parent) 
	#dim(edges_sim)
	
	# Create a tidy data frame of changes in expression
	edges_sim_tidy = edges_sim %>% dplyr::select( gene_tree, S_child, length, node_depth_child, node_age_child, node_age_parent, node_parent, node_child, terminal )
	edges_sim_tidy = cbind( edges_sim_tidy, edges_sim[, grepl("scaled", names(edges_sim))] )
	colnames(edges_sim_tidy) = colnames(edges_sim_tidy) %>% sub( "_scaled", "", .)
	edges_sim_tidy %<>% tidyr::gather( names(edges_sim_tidy)[10:18], key = "treatment", value = "change" )
	
	# Change child node to factor
	edges_sim_tidy %<>% dplyr::mutate( S_child = as.factor(S_child) )
	
	# Remove tau for now since it is on a different scale
	edges_sim_tidy %<>% dplyr::filter(treatment != "tau")
	
	# Remove rows with missing values
	edges_sim_tidy %<>% na.omit()
	
	# Add blast reports
	edges_sim_tidy %<>%
		dplyr::left_join( genes_to_tips_sim, key="gene_tree")
	
	edges_sim_tidy$change_abs = abs( edges_sim_tidy$change )
	
	edges_sim_tidy<-edges_sim_tidy %>% group_by(S_child,treatment) %>% filter(n()>300) %>% ungroup()
	
	digest_sim<-lapply(gene_trees_sim_calibrated_pruned,digest::digest) %>% unlist()


```

```{r go-annots , verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

parseGO<-function(GOannot){
 GO<-readr::read_tsv(GOannot, col_names=TRUE) 
 GO_bygene<-strsplit(GO$GO_term,";")
 names(GO_bygene)<-GO$sequence_id
 return(GO_bygene)
}

getannots<-function(GobyGo){
  x<-list()
  for(i in 1:length(GobyGo)){
  x[[i]]<-data.frame(go=rep(names(GobyGo)[[i]],length(GobyGo[[i]])),gene=GobyGo[[i]])
  x[[i]]$go<-as.character(x[[i]]$go)
  x[[i]]$gene<-as.character(x[[i]]$gene)
  }
  x<-x %>% dplyr::bind_rows()
  return(x)
}

Agalma_GO<- parseGO(GOannot="GO_terms/Agalma_GO_anno.txt")
Agalma_GO<-Agalma_GO[names(Agalma_GO) %in% rownames(e$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`@x)]
Agalma_annot <- getannots(topGO::inverseList(Agalma_GO))
Nanomia_GO<-parseGO(GOannot="GO_terms/Nanomia_GO_anno.txt")
Nanomia_GO<-Nanomia_GO[names(Nanomia_GO) %in% rownames(e$`HWI-ST625-51-C02UNACXX-7-NANOMIA`@x)]
Nanomia_annot <- getannots(topGO::inverseList(Nanomia_GO))
Bargmannia_GO<-parseGO(GOannot="GO_terms/Bargmannia_GO_anno.txt")
Bargmannia_GO<-Bargmannia_GO[names(Bargmannia_GO) %in% rownames(e$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`@x)]
Bargmannia_annot <- getannots(topGO::inverseList(Bargmannia_GO))
Frillagalma_GO<-parseGO(GOannot="GO_terms/Frillagalma_GO_anno.txt")
Frillagalma_GO<-Frillagalma_GO[names(Frillagalma_GO) %in% rownames(e$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`@x)]
Frillagalma_annot <- getannots(topGO::inverseList(Frillagalma_GO))
Diphyes_GO<-parseGO(GOannot="GO_terms/Diphyes_GO_anno.txt")
Diphyes_GO<-Diphyes_GO[names(Diphyes_GO) %in% rownames(e$`K00162-189-HJTYGBBXX-7-NCAGTG`@x)]
Diphyes_annot <- getannots(topGO::inverseList(Diphyes_GO))
Physalia_GO<-parseGO(GOannot="GO_terms/Physalia_GO_anno.txt")
Physalia_GO<-Physalia_GO[names(Physalia_GO) %in% rownames(e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@x)]
Physalia_annot <- getannots(topGO::inverseList(Physalia_GO))
Apolemia_GO<-parseGO(GOannot="GO_terms/Apolemia_GO_anno.txt")
Apolemia_GO<-Apolemia_GO[names(Apolemia_GO) %in% rownames(e$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`@x)]
Apolemia_annot <- getannots(topGO::inverseList(Apolemia_GO))
GO<- readr::read_tsv("GO_terms/AllGOcat.txt", col_names=TRUE)

reduced_nodes<-nodes_ace[,c("gene_tree","sequence_ids")]

GO_to_tips =
		nodes_ace %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,GO, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( GO =names(which.max(table(GO_term))) ) 

GO_by_tree<-strsplit(GO_to_tips$GO,";")
names(GO_by_tree)<-GO_to_tips$gene_tree

readable_term<-function(golist){
x1<-lapply(golist, Term)
if(length(x1)>5){
  x1<-x1[1:5] %>% paste(collapse=", ")
}
else{
x1<-x1 %>% paste(collapse=", ")
}
return(x1)
}

genes_to_tips =
		nodes_ace %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,blast_lookup, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( blast_hit =names(which.max(table(blast_hit))) ) %>% # A convoluted way to get the most frequent blast hit for each gene
		dplyr::mutate( blast_hit = stringr::str_replace(blast_hit, "swissprot|", ""))

	lengthfunction<-function(object,GOlist){
	length<-as.integer(object@lengths)
	names(length)<-rownames(object@x)
	length<-length[names(length) %in% names(GOlist)]
	return(length)
	}
  
	Physalia_length<-lengthfunction(e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`,Physalia_GO)
	Nanomia_length<-lengthfunction(e$`HWI-ST625-51-C02UNACXX-7-NANOMIA`,Nanomia_GO)
	Diphyes_length<-lengthfunction(e$`K00162-189-HJTYGBBXX-7-NCAGTG`,Diphyes_GO)
	Apolemia_length<-lengthfunction(e$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`,Apolemia_GO)
	Agalma_length<-lengthfunction(e$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`,Agalma_GO)
	Bargmannia_length<-lengthfunction(e$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`,Bargmannia_GO)
	Frillagalma_length<-lengthfunction(e$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`,Frillagalma_GO)
```

```{r linear models, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

edges_tidy_original<-edges_tidy
edges_sim_tidy_original<-edges_sim_tidy
edges_tidy <- edges_tidy[!edges_tidy$treatment=="Bradev",]
edges_sim_tidy <- edges_sim_tidy[!edges_sim_tidy$treatment=="Bradev",]

gene_trees_sim_digests<-unlist(lapply(gene_trees_sim_calibrated_pruned, digest))

change_deciles<-quantile(edges_tidy$change,probs= seq(0, 1, by= 0.1))

linear_model<- edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lm(change~S_child*treatment,data=.)
g_linear_model<- edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% glm(change~S_child*treatment,data=.)
lm_mixed<- edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lmer(change~S_child*treatment+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE))
lm_mixed_S_null<-edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lmer(change~S_child+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE), REML=FALSE)
lm_mixed_null<-edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lmer(change~treatment+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE), REML=FALSE)
lm_mixed_interact_null<-edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lmer(change~S_child+treatment+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE),REML=FALSE)

#check simulated data

linear_modelsim<-edges_sim_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lm(change~S_child*treatment,data=.)
linear_mixed_modelsim<-edges_sim_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>%lmer(change~S_child*treatment+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE))
lm_mixed_modelsim_S_null<-edges_sim_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lmer(change~S_child+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE), REML=FALSE)
lm_mixed_modelsim_null<-edges_sim_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lmer(change~treatment+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE), REML=FALSE)
lm_mixed_modelsim_interact_null<-edges_sim_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% lmer(change~S_child+treatment+(1|gene_tree),data=., control = lmerControl(calc.derivs = FALSE),REML=FALSE)

anova_linear_sim<-anova(linear_modelsim)

#try shuffling values along the branch
xshuffle<-edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% transform(.,change=sample(change))
linear_modelshuffle<-lm(change~S_child*treatment,data=xshuffle)
linear_mixed_modelshuffle<-lmer(change~S_child*treatment+(1|gene_tree),data=xshuffle, control = lmerControl(calc.derivs = FALSE))
linear_mixed_modelshuffle_S_null<-lmer(change~S_child+(1|gene_tree),data=xshuffle, control = lmerControl(calc.derivs = FALSE), REML=FALSE)
linear_mixed_modelshuffle_null<-lmer(change~treatment+(1|gene_tree),data=xshuffle, control = lmerControl(calc.derivs = FALSE), REML=FALSE)
linear_mixed_modelshuffle_interact_null<-lmer(change~S_child+treatment+(1|gene_tree),data=xshuffle, control = lmerControl(calc.derivs = FALSE),REML=FALSE)
anova_linear_modelshuffle<- anova(linear_modelshuffle)

#check if largest changes have an effect, they do not.
#x1<- edges_tidy %>% filter(change>change_deciles[2] & change<change_deciles[10]) %>% dplyr::select(change,gene_tree,S_child,treatment)
#x3<-lm(change~S_child*treatment,data=x1)

anovalm<-anova(linear_model)
#anova(x3)
#S_child, treatment and S_child:treatment are all significant

S_child_marginal<- emmeans(linear_model, pairwise ~ S_child | treatment) 

#cld(treatment_marginal,
#alpha=0.05,
#Letters=letters,
#adjust="tukey")

Sum_shuffle = Summarize(change ~ S_child*treatment,  data=xshuffle,digits=3)

Sum_sim= edges_sim_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% Summarize(change ~ S_child*treatment,  data=.,digits=3)

Sum = edges_tidy %>% dplyr::select(change,gene_tree,S_child,treatment) %>% Summarize(change ~ S_child*treatment,  data=.,digits=3)

#add more memorable branch names with letters
Sum = data.frame(S_new = S_new$S_new[match(Sum$S_child, S_new$S)], S_child=Sum$S_child) %>% dplyr::bind_cols(Sum, .)

Sum_sim = data.frame(S_new = S_new$S_new[match(Sum_sim$S_child, S_new$S)], S_child=Sum_sim$S_child) %>% dplyr::bind_cols(Sum_sim, .)

pd = position_dodge(.2)

branch_change<- ggplot(Sum, aes(x = S_new,
                y = mean,
                color = treatment)) +
    geom_errorbar(aes(ymin = mean - sd,
                      ymax = mean + sd),
                   width=.2, size=0.7, position=pd) +
    geom_point(shape=15, size=4, position=pd) +
    theme_bw() +
    theme(axis.title = element_text(face = "plain", size=12)) +
    ylab("Mean change along branch") + xlab("") + ggtitle("Empirical") +theme(plot.title = element_text(hjust = 0.5)) +ylim(c(-15,15)) + scale_color_manual(values=c("#F9B326","#F97F06","#BC98D3","#F4A6AE","#AEC64A","#F7D114","#7ECBE0"))

leg<- get_legend(branch_change) 

branch_change<-branch_change + guides(color=FALSE)

sim_branch_change<- ggplot(Sum_sim, aes(x = S_new,
                y = mean,
                color = treatment)) +
    geom_errorbar(aes(ymin = mean - sd,
                      ymax = mean + sd),
                   width=.2, size=0.7, position=pd) +
    geom_point(shape=15, size=4, position=pd) +
    theme_bw() +
    theme(axis.title = element_text(face = "plain", size=12)) +
    ylab("") + ggtitle("Simulated") + xlab("") + guides(color=FALSE) + theme(plot.title = element_text(hjust = 0.5)) +ylim(c(-15,15)) + scale_color_manual(values=c("#F9B326","#F97F06","#BC98D3","#F4A6AE","#AEC64A","#F7D114","#7ECBE0"))

tree_reference<- ggtree(phyldog_species_numbered_tree, branch.length="none") + geom_tiplab(aes(label=label), fontface='italic')+xlim(-6,15) +geom_label(aes(x=branch,label=S_new$S_new), size=2, label.size = 0.1)

layout<-rbind(c(1,1,NA),c(2,3,4))

change_figure<- grid.arrange(tree_reference,branch_change,sim_branch_change,leg, bottom="Branch",widths = c(2,2,0.5),layout_matrix = layout)


#the vast majority of branches show small or no change in expression along a branch - conversely, a handful of gene trees show large changes along a branch

var_change_obs_data<-edges_tidy %>% group_by(S_child,treatment) %>% summarise(mean_abs_change=mean(change_abs), var_abs_change=var(change_abs),mean_change=mean(change),var_change=var(change),node_age_parent=node_age_parent[1])

##add human readable branch data
var_change_obs_data<-data.frame(S_new = S_new$S_new[match(var_change_obs_data$S_child, S_new$S)], S_child=var_change_obs_data$S_child) %>% bind_cols(var_change_obs_data, .)

var_change_obs<-var_change_obs_data%>% ggplot()+geom_point(aes(node_age_parent,var_abs_change,col=S_new)) + facet_wrap(~treatment) + xlab(" ") + ylab("")+ggtitle(label="Empirical") + theme(plot.title = element_text(hjust = 0.5))+ guides(col=guide_legend(title="Branch"))

legend<- get_legend(var_change_obs) 

var_change_obs<-var_change_obs + guides(color=FALSE)

#For all treatments, mean abs change is a function of the node_age of the parent - with greater mean abs change the younger the node age of the parent.  

var_change_sim_data<- edges_sim_tidy %>% group_by(S_child,treatment) %>% summarise(mean_abs_change=mean(change_abs), var_abs_change=var(change_abs),mean_change=mean(change),var_change=var(change),node_age_parent=node_age_parent[1])
  
##add human readable branch data
var_change_sim_data<-data.frame(S_new = S_new$S_new[match(var_change_sim_data$S_child, S_new$S)], S_child=var_change_sim_data$S_child) %>% bind_cols(var_change_sim_data, .)
    
var_change_sim<-var_change_sim_data %>% ggplot()+geom_point(aes(node_age_parent,var_abs_change,col=S_new)) + facet_wrap(~treatment) + xlab(" ") + ylab(" ")+ggtitle(label="Simulated") + theme(plot.title = element_text(hjust = 0.5))+ guides(color=FALSE)

#this pattern holds with simulated values, suggesting that variance is driven by the internal structure of the tree

layout2<-rbind(c(1,2),c(3,2))

```

```{r covariance, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}
  edges_cov<-edges_ace_original

	# Only retain edges where both parent and child is a speciation event
	edges_cov %<>% dplyr::filter( (Ev_parent=="S") & (Ev_child=="S") )
	#dim(edges_ace)
	
	# Create an edge matrix that is in terms of phyldog node numbers rather than ape node numbers
	phyldog_edges = S[ as.vector(phyldog_species_numbered_tree$edge) ]
	dim(phyldog_edges) = dim( phyldog_species_numbered_tree$edge )
	
	# For each row in edges_ace, identify the expected parent in the species tree for S_child
	expected_parent = phyldog_edges[ match(edges_cov$S_child, phyldog_edges[,2]), 1 ]
	
	# Filter to retain only the edges where S_parent matches the expected parent of S_child in the species tree
	
	edges_cov %<>% dplyr::filter( (S_parent==expected_parent) )
	#dim(edges_ace)
	
	# Create a tidy data frame of changes in expression
	edges_tidy_cov = edges_cov %>% dplyr::select( gene_tree, S_child, length, node_depth_child, node_age_child, node_age_parent, node_parent, node_child, terminal )
	edges_tidy_cov = cbind( edges_tidy_cov, edges_cov[, grepl("scaled", names(edges_cov))] )
	colnames(edges_tidy_cov) = colnames(edges_tidy_cov) %>% sub( "_scaled", "", .)
	edges_tidy_cov_superset<-	edges_tidy_cov[,c(1:2,7:8,10:14,16:17)]
	edges_tidy_cov<-edges_tidy_cov[,c(1:2,10:14,16:17)]
	#edges_tidy_cov<-edges_tidy_cov %>% filter(S_child %in% c("9","26","N37","35","N37"))

correlation_branches<-function(child,combn){
  correlation<-function(combn,child){
  u<- edges_tidy_cov %>% dplyr::filter(S_child == child) %>% dplyr::select(combn)
  not_all_na <- function(x) any(!is.na(x))
  u <- u%>% dplyr::select_if(not_all_na) %>% na.omit() 
  if(nrow(u)>100){
  cor_u <- cor(u)
  return(cor_u)
  }
  else{
    return(NA)
  }
}
  
combn1<-lapply(combn, correlation, child) 
names(combn1)<- lapply(combn,short_pairs_tpm) 
combn1<-combn1[lapply(combn1,length)>0]
combn1<-lapply(combn1,function(x){
  if(length(x)>1){
  x=x[,1]
  x=x[x!=1]
  x<-unname(x)
  return(x)
  }
})
combn1<- combn1 %>% unlist()
return(combn1)
}

focal_tpm_values<- focal_tpm_values[focal_tpm_values!="Bradev"]
pairs_tpm<-combn(focal_tpm_values,2, simplify=FALSE)
S_children<-edges_tidy_cov$S_child %>% unique()
	
Correlations<-lapply(S_children, correlation_branches,pairs_tpm)
Correlations<-lapply(Correlations,as.data.frame) 
Correlations<-lapply(Correlations,tibble::rownames_to_column)
for(i in 1:length(Correlations)){
  names(Correlations[[i]])<-c("zooid",S_children[[i]])
}
Correlations_new<-Reduce(dplyr::full_join,list(Correlations[[1]],Correlations[[2]],Correlations[[3]],Correlations[[4]],Correlations[[5]],Correlations[[6]],Correlations[[7]],Correlations[[8]],Correlations[[9]],Correlations[[10]],Correlations[[11]],Correlations[[12]]))

rm(Correlations)

Correlations_new<-tidyr::gather(Correlations_new,"S_child","Correlation",-zooid)

Correlations<-Correlations_new %>% tidyr::spread(key=zooid, value=Correlation) 

species_tree_annotated<- ape::chronos( phyldog_species_numbered_tree, lambda=1, model="correlated", quiet=TRUE )

tags<-species_tree_annotated %>% as_tibble %>% tibble::add_column(.,S= c(S_child=node_map$S[match(node_map$species,species_tree_annotated$tip.label)], species_tree_annotated$node.label))

tags<-dplyr::left_join(tags,Correlations,by=c("S"="S_child")) 

species_tree_annotated<- as.treedata(tags)

Correlation_pairs<-Correlations %>% dplyr::select(-S_child) %>% names()
Correlation_plots<-lapply(Correlation_pairs,function(pair){
tree<-ggtree(species_tree_annotated) + ggplot2::aes_string(color=as.name(pair)) + ggplot2::scale_color_continuous(low='darkgreen', high='red') +theme(legend.position="right") +geom_tiplab() +xlim(c(0,2))
return(tree)
})

# for(i in 1:length(Correlation_plots)){
# pdf(paste0("Corplot",i,".pdf"))
# print(Correlation_plots[[i]])
# dev.off()
# }

trait<-Correlations_new 
trait$S_child<-factor(trait$S_child, levels=species_tree_annotated@data$S)
trait_plot<- trait %>%  filter(S_child%in%c("4","6","9","18","26","32","35"))%>% ggplot()+geom_tile(aes(zooid,S_child,fill=Correlation))+theme(axis.text.x=element_text(angle = 30, hjust = 1))+ ggplot2::scale_fill_continuous(low='#BBDEF2', high='#0B4A6D',limits=c(0.5,0.9))

trait_plot2<- trait %>%  filter(S_child%in%c("N37"))%>% ggplot()+geom_tile(aes(zooid,S_child,fill=Correlation))+theme(axis.text.x=element_text(angle = 30, hjust = 1))+ ggplot2::scale_fill_continuous(low='#BBDEF2', high='#0B4A6D',limits=c(0.5,0.9))

trait_plot3<- trait %>%  filter(S_child%in%c("N23"))%>% ggplot()+geom_tile(aes(zooid,S_child,fill=Correlation))+theme(axis.text.x=element_text(angle = 30, hjust = 1))+ ggplot2::scale_fill_continuous(low='#BBDEF2', high='#0B4A6D',limits=c(0.5,0.9))

trait_plot4<- trait %>%  filter(S_child%in%c("N15"))%>% ggplot()+geom_tile(aes(zooid,S_child,fill=Correlation))+theme(axis.text.x=element_text(angle = 30, hjust = 1))+ ggplot2::scale_fill_continuous(low='#BBDEF2', high='#0B4A6D',limits=c(0.5,0.9))

trait_plot5<- trait %>%  filter(S_child%in%c("N12"))%>% ggplot()+geom_tile(aes(zooid,S_child,fill=Correlation))+theme(axis.text.x=element_text(angle = 30, hjust = 1))+ ggplot2::scale_fill_continuous(low='#BBDEF2', high='#0B4A6D',limits=c(0.5,0.9))

trait_plot7<- trait %>%  filter(S_child%in%c("N7"))%>% ggplot()+geom_tile(aes(zooid,S_child,fill=Correlation))+theme(axis.text.x=element_text(angle = 30, hjust = 1))+ ggplot2::scale_fill_continuous(low='#BBDEF2', high='#0B4A6D',limits=c(0.5,0.9))

trait_plot_all<- trait %>% ggplot()+geom_tile(aes(zooid,S_child,fill=Correlation))+theme(axis.text.x=element_text(angle = 30, hjust = 1))+ ggplot2::scale_fill_continuous(low='#BBDEF2', high='#0B4A6D',limits=c(0.5,0.9))

p11 <- ggtree(species_tree_annotated) +
  xlim(0,3) +
  geom_tiplab()

###Export all of these plots to build figure

#cor_plot <- ggcorr(u, label = TRUE, palette = "RdYlBu")
	
S_20<-edges_tidy_cov %>% dplyr::filter(S_child == "18") %>% .[,c(3,5)] %>% na.omit() %>% ggcorr(., label = TRUE, palette = "RdYlBu")

legend4<-get_legend(S_20)	

S_40<-edges_tidy_cov %>% dplyr::filter(S_child == "9") %>% .[,c(3,5,6,7,8)] %>% na.omit() %>% ggcorr(., label = TRUE, palette = "RdYlBu")+ guides(fill=FALSE) + ggtitle("Branch H") + theme(plot.title = element_text(hjust = 0.5))

S_55<-edges_tidy_cov %>% dplyr::filter(S_child == "26") %>% .[,c(3:9)] %>% na.omit() %>% ggcorr(., label = TRUE, palette = "RdYlBu")+ guides(fill=FALSE) + ggtitle("Branch J") + theme(plot.title = element_text(hjust = 0.5))

S_70<-edges_tidy_cov %>% dplyr::filter(S_child == "N37") %>% .[,c(4:7)] %>% na.omit() %>% ggcorr(., label = TRUE, palette = "RdYlBu")+ guides(fill=FALSE) + ggtitle("Branch E") + theme(plot.title = element_text(hjust = 0.5))

S_79<-edges_tidy_cov %>% dplyr::filter(S_child == "32") %>% .[,c(4:7)] %>% na.omit() %>% ggcorr(., label = TRUE, palette = "RdYlBu")+ guides(fill=FALSE) + ggtitle("Branch K") + theme(plot.title = element_text(hjust = 0.5))

S_76<-edges_tidy_cov %>% dplyr::filter(S_child == "35") %>% .[,c(4:7)] %>% na.omit() %>% ggcorr(., label = TRUE, palette = "RdYlBu")+ guides(fill=FALSE) + ggtitle("Branch L") + theme(plot.title = element_text(hjust = 0.5))

```

# Evolution of gene expression across species and specialized zooids in Siphonophora {-}
Catriona Munro^1,2^,  Stefan Siebert^3^, Felipe Zapata^4^, Mark Howison^5^, Steven H.D. Haddock^6^, Casey W. Dunn^7^
\newline
\newline
^1^ Department of Ecology and Evolutionary Biology, Brown University, Providence, RI 02912, USA \newline
^2^ Current address: Collège de France, PSL Research University, CNRS, Inserm, Center for Interdisciplinary Research in Biology, 75005 Paris, France \newline
^3^ XXXXXX Insert Stefan address XXXXXXXXXXX \newline
^4^ Department of Ecology and Evolutionary Biology, University of California Los Angeles, Los Angeles, CA 90095, USA \newline
^5^ Watson Institute for International and Public Affairs, Brown University, Providence, RI 02912, USA \newline
^6^ Monterey Bay Aquarium Research Institute, Moss Landing, CA 95039, USA \newline
^7^ Department of Ecology and Evolutionary Biology, Yale University, New Haven, CT 06520, USA



## Abstract {-}

## Introduction {-}

Gene expression is an important component of phenotypic diversity within and among units of biological organization, including cells, tissues, organs, and species [@Brawand:2011du; @gilad2005multi; @king1975evolution; @rifkin2003evolution]. There are a number of ways in which gene expression can change, including in the relative magnitude of expression, or temporal or spatial shifts in expression [@carroll2005evolution; @romero2012comparative; @wray2003evolution]. With high throughput RNA-seq data, we are able to quantify the relative expression level of a large number of genes within and among species, homologous organs, and developmental timepoints. Based primarily on qualitative distance measures and principal component analysis (PCA) of expression across 1:1 orthologs, most studies suggest that gene expression is more conserved among homologous organs of the same type across different species than among different organs within the same species; suggesting that molecular, cellular and developmental pathways are highly conserved among species [@Brawand:2011du; @breschi2016gene; @cardoso2019gene; @clarke2017evolutionary; @gilad2015reanalysis; @khaitovich2004neutral; @merkin2012evolutionary; @10.1093/molbev/msz212]. Other studies found the opposite result [@lin2014comparison; @pankey2014predictable; @tschopp2014relative; @yang2013organ], however re-analyses contest these findings [@gilad2015reanalysis; @gu2015understanding; @sudmant2015meta]. These global patterns of gene expression, however, fail to account for gene-specific patterns, which may vary in organ and species specificity, and clustering patterns may be driven by the behavior of a small subset of genes [@breschi2016gene].  In this study, we take a gene-centric approach using phylogenetic comparative methods (PCMs) to compare gene expression patterns between different siphonophore zooids, and one specialized tissue, and map expression values directly onto gene trees, in order to investigate evolutionary changes gene expression of different tissues across branches in gene trees.

Only a handful of studies have applied PCMs to compare gene expression values across species [@clarke2017evolutionary; @Chang:2014hq; @Chen229096; @Eng:2009gp; @gu2004statistical; @Oakley:2005ky; @Rohlfs:2015bd; @stern2018evolution; @Whitney:2011fw]. PCMs were developed to address challenges in comparing trait data across species, particularly the non-independence of traits due to the evolutionary history of the species [@dunn2013phylogenetic; @Felsenstein:1985ua; @Felsenstein:2008cl; @FitzJohn:2012ey;@Grafen:1989um; @Revell:2012jd; @Pagel:1999gc; @Uyeda:2014jl]. Multiple pairwise comparisons are frequently used in comparative genomic work, however re-analysis of the same data using PCMs has been shown to support very different conclusions; underscoring the importance of considering species relationships when making statements about evolutionary processes [@dunn2018pairwise].

Analyses of gene expression that do take phylogeny into account typically focus on strict orthologs and map expression values onto the species tree [@Chen229096;Oakley:2005ky; @Rohlfs:2015bd; @stern2018evolution], while other analyses use the expression values themselves to build neighbor-joining trees [@Brawand:2011du; @stern2018evolution]. However, in addition to understanding species relationships, understanding the homology of genes is critical for understanding the evolution of gene expression. Gene phylogenies are often used to study species relationships and the history of gene duplication and loss, but they are more useful than that -- they can also be used to study the evolution of gene traits such as expression. Gene phylogenies are hypotheses about the evolutionary relationship of gene sequences to one another, and gene topology reflects a number of evolutionary events, including speciation events, gene duplication and loss, and molecular evolution.

Siphonophores are highly complex, colonial "superorganisms" consisting of asexually produced bodies (termed zooids) that are homologous to solitary free-living polyps and medusae, but that share a common gastrovascular cavity [@dunn2006evolution;@mackie1963; @mackie1986aggregates;@mackie1987siphonophore;@totton1965synopsis]. The functional specialization of siphonophore zooids has been of central interest to zoologists since the 19th century, in part because siphonophores are considered to have a greater level of functional specialization than any other colonial animal, but also because these zooids are highly interdependent [@Beklemishev1969; @mackie1963].

Efforts to investigate the functional specialization of siphonophores have been limited, in part because there have been few detailed investigations of zooid structure. In the last half century, the microanatomy of siphonophore zooids and tissues has been investigated in only a handful of siphonophore species [@BARDI2007; @carre1969etude;@church2015histology; @Mackie1960]. This leaves many unknowns about how zooid structure and function differ across zooid types and species. Recent *in situ* gene expression analyses in siphonophores have described where a small number of pre-selected genes are expressed at high spatial resolution [@church2015histology; @Siebert:2011gv;@siebert2015stem], but these methods are limited since they require a large number of specimens per gene and siphonophores are relatively difficult to collect. RNA-seq analyses of hand-dissected specimens [@macrander2015rna;@sanders2014differential;@sanders2015interspecific;@Siebert:2011gv], in contrast, can describe the expression of a very large number of genes at low spatial resolution. The fact that so much data is obtained from each specimen is particularly advantageous in difficult-to-collect organisms like siphonophores. An earlier RNA-seq study of gene expression in two zooid types in a single siphonophore species showed the potential of this method to better understand differences between zooids [@Siebert:2011gv]. 

(ref:zooids-definition) Schematic of the siphonophore *Nanomia bijuga*, with highlighted zooids and pneumatophore, with explanations of known function. NGZ - nectosomal growth zone. SGZ - siphosomal growth zone. Diagram by Freya Goetz (http://commons.wikimedia.org/wiki/File:Nanomia_bijuga_whole_animal_and_growth_zones.svg)

```{r zooids-definition, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.5\\textheight', out.width='0.5\\textheight', fig.cap='(ref:zooids-definition)', fig.align = 'center'}

knitr::include_graphics("Figures/Siphonophoresare.pdf")

```

In this study we use RNA-seq data to conduct differential expression analyses among siphonophore zooids, and one specialized tissue (the pneumatophore, a gas filled float), within siphonophore species, and also map RNA-seq data directly onto gene phylogenies in order to compare gene expression signatures within zooid types across species. We also investigate differential gene expression patterns within species, in particular focusing on three species-specific zooid types that represent lineage specific zooid diversification events. 

## Results {-}

### Differential expression within species {-}

```{r differentially-expressed, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

top_res<- function(pair,object){
  
treat1=pair[1]
treat2=pair[2]
  
if( (length(unique(object@ddsMF@colData$treatment)[unique(object@ddsMF@colData$treatment) %in% c(treat1,treat2) ==TRUE])==2)&((length(object@ddsMF@colData$treatment[object@ddsMF@colData$treatment==treat1])>=2)&(length(object@ddsMF@colData$treatment[object@ddsMF@colData$treatment==treat2])>=2))){

res<-DESeq2::results(object@ddsMF,contrast=c( "treatment", treat1, treat2 ), pAdjustMethod="bonferroni", alpha=0.05 )

resOrdered <- res[order(res$padj),] 

resSig <- subset(resOrdered, padj < 0.05) %>% as.data.frame(.)

resSig <- subset(resSig,log2FoldChange > 0 )

resSig<-tibble::rownames_to_column(resSig, var="sequence_ids") 
resSig$sequence_ids<-as.numeric(resSig$sequence_ids)

resSig<-dplyr::left_join(resSig,reduced_nodes, by="sequence_ids")

resSig<-dplyr::left_join(resSig,blast_lookup, by="sequence_ids")

resSig<-dplyr::left_join(resSig,GO,by="sequence_ids")

return(resSig)
}
}


significant_de_genes<-function(object){
all_pairs_dge<- object@ddsMF$treatment %>% unique() %>% as.character() %>% gtools::permutations(n=length(.), r=2,v=., repeats.allowed = TRUE) %>% as.data.frame(stringsAsFactors=FALSE) 

all_pairs_dge<- all_pairs_dge %>% split(., seq(nrow(.))) %>% unname() %>% lapply(.,as.character)

x<-lapply(all_pairs_dge, top_res, object)

short_pairs_function<- function(pairs){
      pair_short<-pairs%>%
      stringr::str_split(.," ")%>%
      unlist() %>%
      stringr::str_sub(.,1,3) %>%
      stringr::str_c(.,collapse="")
      return(pair_short)
}

names(x)<-lapply(all_pairs_dge,short_pairs_function) %>% unlist()

return(x)
}

dge_sig<-lapply(dge, significant_de_genes)

#Agalma

##This is clunky, but due to the different sampling between species I have to do this manually

Agalma_necdev<-c( dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevPalmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevGonfem$sequence_ids) %>% unique()

Agalma_Palpon<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatPne$sequence_ids) %>% unique()

Agalma_PalponB<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatPne$sequence_ids) %>% unique()

Agalma_gasmat<-c( dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatPalmat$sequence_ids) %>% unique()

Agalma_Gonfem<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemPalmat$sequence_ids) %>% unique()

Agalma_Gonmal<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalPalmat$sequence_ids) %>% unique()

Agalma_Pne<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PnePalmat$sequence_ids) %>% unique()

Agalma_Pne_unique<-Agalma_Pne[!Agalma_Pne %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_gasmat,Agalma_Gonfem,Agalma_necdev, Agalma_Gonmal)]

Agalma_necdev_unique<-Agalma_necdev[!Agalma_necdev %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_Gonfem,Agalma_Gonmal,Agalma_gasmat, Agalma_Pne)]

Agalma_Gonmal_unique<-Agalma_Gonmal[!Agalma_Gonmal %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_gasmat,Agalma_Gonfem,Agalma_necdev, Agalma_Pne)]

Agalma_Gonfem_unique<-Agalma_Gonfem[!Agalma_Gonfem %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_gasmat,Agalma_Gonmal,Agalma_necdev, Agalma_Pne)]

Agalma_gasmat_unique<-Agalma_gasmat[!Agalma_gasmat %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_Gonfem,Agalma_Gonmal,Agalma_necdev, Agalma_Pne)]

Agalma_PalponB_unique<-Agalma_PalponB[!Agalma_PalponB %in% c(Agalma_necdev,Agalma_Gonfem,Agalma_Gonmal,Agalma_gasmat, Agalma_Pne)]

Agalma_Palpon_unique<-Agalma_Palpon[!Agalma_Palpon %in% c(Agalma_necdev,Agalma_Gonfem,Agalma_Gonmal,Agalma_gasmat, Agalma_Pne)]

## Bargmannia

Bargmannia_necdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGasyelmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGonmal$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevPne$sequence_ids) %>% unique

Bargmannia_Gaswhimat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGasyelmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGonmal$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatPne$sequence_ids) %>% unique

Bargmannia_Gasyelmat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGonmal$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatPne$sequence_ids) %>% unique

Bargmannia_Gonmal<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalPne$sequence_ids) %>% unique

Bargmannia_Pne<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGonmal$sequence_ids) %>% unique

Bargmannia_Gasyeldev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevPne$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGonmal$sequence_ids) %>% unique

Bargmannia_Gasdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevPne$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGonmal$sequence_ids) %>% unique


Bargmannia_Gasdev_unique<-Bargmannia_Gasdev[!Bargmannia_Gasdev %in% c(Bargmannia_Gasyeldev,Bargmannia_Pne,Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_Gaswhimat,Bargmannia_necdev)]

Bargmannia_Gasyeldev_unique<-Bargmannia_Gasyeldev[!Bargmannia_Gasyeldev %in% c(Bargmannia_Gasdev,Bargmannia_Pne,Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_Gaswhimat,Bargmannia_necdev)]

Bargmannia_necdev_unique<-Bargmannia_necdev[!Bargmannia_necdev %in% c(Bargmannia_Pne,Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_Gaswhimat)]

#exclude developing gastrozooids due to significant overlap (note I will include it for the developing zooid because I want to know genes involved specifically in development of this zooid)

Bargmannia_Gaswhimat_unique<-Bargmannia_Gaswhimat[!Bargmannia_Gaswhimat %in% c(Bargmannia_Pne,Bargmannia_Gonmal, Bargmannia_necdev,Bargmannia_Gasyelmat)]

Bargmannia_Gasyelmat_unique<-Bargmannia_Gasyelmat[!Bargmannia_Gasyelmat %in% c(Bargmannia_Pne,Bargmannia_Gonmal, Bargmannia_necdev,Bargmannia_Gaswhimat)]

Bargmannia_Gasmat_unique<-Bargmannia_Gaswhimat[!Bargmannia_Gaswhimat %in% c(Bargmannia_Pne,Bargmannia_Gonmal, Bargmannia_necdev)]

Bargmannia_gonmal_unique<-Bargmannia_Gonmal[!Bargmannia_Gonmal %in% c(Bargmannia_Pne,Bargmannia_Gasyelmat, Bargmannia_necdev,Bargmannia_Gaswhimat)]

Bargmannia_Pne_unique<-Bargmannia_Pne[!Bargmannia_Pne %in% c(Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_necdev,Bargmannia_Gaswhimat)]

##Nanomia

Nanomia_Necdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevGasmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevPalmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevPne$sequence_ids) %>% unique

Nanomia_Gasmat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatPalmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatPne$sequence_id) %>% unique

Nanomia_Gasdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevGasmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevPalmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevPne$sequence_id) %>% unique

Nanomia_Palmat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatPne$sequence_id) %>% unique

Nanomia_Paldev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevPne$sequence_id) %>% unique

Nanomia_Gonfem<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemPne$sequence_id) %>% unique

Nanomia_Pne<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PnePaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PnePaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneGonfem$sequence_id) %>% unique

Nanomia_Necdev_unique<-Nanomia_Necdev[!Nanomia_Necdev %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Paldev,Nanomia_Palmat,Nanomia_Gasdev,Nanomia_Gasmat)]

#exclude developing gastrozooids due to significant overlap (note I will include it for the developing zooid because I want to know genes involved specifically in development of this zooid)
Nanomia_Gasmat_unique<-Nanomia_Gasmat[!Nanomia_Gasmat %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Paldev,Nanomia_Palmat,Nanomia_Necdev)]

#exclude developing palpons for the same reason (note I will include it for the developing zooid because I want to know genes involved specifically in development of this zooid)
Nanomia_Palmat_unique<-Nanomia_Palmat[!Nanomia_Palmat %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev)]

Nanomia_Gonfem_unique<-Nanomia_Gonfem[!Nanomia_Gonfem %in% c(Nanomia_Pne,Nanomia_Palmat,Nanomia_Paldev, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev)]

Nanomia_Pne_unique<-Nanomia_Pne[!Nanomia_Pne %in% c(Nanomia_Gonfem,Nanomia_Palmat,Nanomia_Paldev, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev)]

Nanomia_Gasdev_unique<-Nanomia_Gasdev[!Nanomia_Gasdev %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Paldev,Nanomia_Palmat,Nanomia_Necdev, Nanomia_Gasmat)]

Nanomia_Paldev_unique<-Nanomia_Paldev[!Nanomia_Paldev %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev, Nanomia_Palmat)]

### 


Frillagalma_Gasdev<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevBradev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevPne$sequence_ids) %>% unique()

Frillagalma_Gasmat<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatBradev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatPne$sequence_ids) %>% unique()

Frillagalma_Bradev<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevPne$sequence_ids) %>% unique()

Frillagalma_Gonmal<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalPne$sequence_ids) %>% unique()

Frillagalma_Gonfem<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemPne$sequence_ids) %>% unique()

Frillagalma_Necdev<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevPne$sequence_ids) %>% unique()

Frillagalma_Palmat<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatPne$sequence_ids) %>% unique()

Frillagalma_Pne<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PnePalmat$sequence_ids) %>% unique()

Frillagalma_Pne_unique<- Frillagalma_Pne[!Frillagalma_Pne%in% c(Frillagalma_Bradev, Frillagalma_Gasdev,Frillagalma_Gasmat, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Bradev_unique<- Frillagalma_Bradev[!Frillagalma_Bradev%in% c(Frillagalma_Pne, Frillagalma_Gasdev,Frillagalma_Gasmat, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gasdev_unique<- Frillagalma_Gasdev[!Frillagalma_Gasdev%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gasmat_unique<- Frillagalma_Gasmat[!Frillagalma_Gasmat%in% c(Frillagalma_Pne, Frillagalma_Bradev, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gonfem_unique<- Frillagalma_Gonfem[!Frillagalma_Gonfem%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gonmal_unique<- Frillagalma_Gonmal[!Frillagalma_Gonmal%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonfem, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Necdev_unique<- Frillagalma_Necdev[!Frillagalma_Necdev%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Palmat)]

Frillagalma_Palmat_unique<- Frillagalma_Palmat[!Frillagalma_Palmat%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev)]

### Apolemia

Apolemia_Pne <- dge_sig$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`$GasdevPne$sequence_ids 
Apolemia_Gasdev <- dge_sig$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`$PneGasdev$sequence_ids

Apolemia_Pne_unique<-Apolemia_Pne[!Apolemia_Pne %in% Apolemia_Gasdev]
Apolemia_Gasdev_unique<-Apolemia_Gasdev[!Apolemia_Gasdev %in% Apolemia_Pne]

### Physalia

Physalia_Gasdev <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevTendev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevTenpalmat$sequence_ids) %>% unique()

Physalia_Gasmat <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatTendev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatTenpalmat$sequence_ids) %>% unique()

Physalia_Tendev <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevTenpalmat$sequence_ids) %>% unique()

Physalia_Tenpalmat <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatTendev$sequence_ids) %>% unique()

Physalia_Tenpaldev <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevTenpalmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevTendev$sequence_ids) %>% unique()

Physalia_Gasdev_unique <- Physalia_Gasdev[!Physalia_Gasdev %in% c(Physalia_Gasmat,Physalia_Tenpalmat, Physalia_Tenpaldev, Physalia_Tendev)]

Physalia_Gasmat_unique <- Physalia_Gasmat[!Physalia_Gasmat %in% c(Physalia_Tenpalmat, Physalia_Tenpaldev, Physalia_Tendev)]

Physalia_Tenpalmat_unique <- Physalia_Tenpalmat[!Physalia_Tenpalmat %in% c(Physalia_Gasmat,Physalia_Gasdev, Physalia_Tendev)]

Physalia_Tenpaldev_unique <- Physalia_Tenpaldev[!Physalia_Tenpaldev %in% c(Physalia_Gasmat,Physalia_Gasdev, Physalia_Tendev, Physalia_Tenpalmat)]

Physalia_Tendev_unique <- Physalia_Tendev[!Physalia_Tendev %in% c(Physalia_Gasmat,Physalia_Gasdev, Physalia_Tenpaldev, Physalia_Tenpalmat)]

### Diphyes

Diphyes_Gasdev <- c(dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasdevGasmat$sequence_ids, dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasdevGon$sequence_ids) %>% unique()

Diphyes_Gasmat <- c(dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasmatGasdev$sequence_ids, dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasmatGon$sequence_ids) %>% unique()

Diphyes_Gon <- c(dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GonGasdev$sequence_ids, dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GonGasmat$sequence_ids) %>% unique()

Diphyes_Gasdev_unique<-Diphyes_Gasdev[!Diphyes_Gasdev %in% c(Diphyes_Gasmat, Diphyes_Gon)]

Diphyes_Gasmat_unique<-Diphyes_Gasmat[!Diphyes_Gasmat %in% Diphyes_Gon]

Diphyes_Gon_unique<-Diphyes_Gon[!Diphyes_Gon %in% c(Diphyes_Gasmat, Diphyes_Gasdev)]


## Now put it all into one tidy dataframe

unique_zooids<-tibble(
  sequence_ids=c(Agalma_Pne_unique,Agalma_necdev_unique, Agalma_Gonmal_unique,Agalma_Gonfem_unique,Agalma_gasmat_unique, Agalma_PalponB_unique, Agalma_Palpon_unique,Bargmannia_Gasdev_unique,Bargmannia_Gasyeldev_unique,Bargmannia_necdev_unique,Bargmannia_Gaswhimat_unique,Bargmannia_Gasyelmat_unique,Bargmannia_gonmal_unique,Bargmannia_Pne_unique,Nanomia_Necdev_unique,Nanomia_Gasmat_unique,Nanomia_Palmat_unique,Nanomia_Gonfem_unique,Nanomia_Pne_unique,Nanomia_Gasdev_unique,Nanomia_Paldev_unique,Frillagalma_Pne_unique,Frillagalma_Bradev_unique,Frillagalma_Gasdev_unique,Frillagalma_Gasmat_unique,Frillagalma_Gonfem_unique,Frillagalma_Gonmal_unique,Frillagalma_Necdev_unique,Frillagalma_Palmat_unique,Apolemia_Pne_unique,Apolemia_Gasdev_unique, Physalia_Gasdev_unique,Physalia_Gasmat_unique,Physalia_Tenpalmat_unique,Physalia_Tenpaldev_unique,Physalia_Tendev_unique,Diphyes_Gasdev_unique,Diphyes_Gasmat_unique,Diphyes_Gon_unique),

species=c(rep("Agalma elegans",length(c(Agalma_Pne_unique,Agalma_necdev_unique, Agalma_Gonmal_unique,Agalma_Gonfem_unique,Agalma_gasmat_unique, Agalma_PalponB_unique, Agalma_Palpon_unique))),
          rep("Bargmannia elongata",length(c(Bargmannia_Gasdev_unique,Bargmannia_Gasyeldev_unique,Bargmannia_necdev_unique,Bargmannia_Gaswhimat_unique,Bargmannia_Gasyelmat_unique,Bargmannia_gonmal_unique,Bargmannia_Pne_unique))),
        rep("Nanomia bijuga",length(c(Nanomia_Necdev_unique,Nanomia_Gasmat_unique,Nanomia_Palmat_unique,Nanomia_Gonfem_unique,Nanomia_Pne_unique,Nanomia_Gasdev_unique,Nanomia_Paldev_unique))),
        rep("Frillagalma vityazi",length(c(Frillagalma_Pne_unique,Frillagalma_Bradev_unique,Frillagalma_Gasdev_unique,Frillagalma_Gasmat_unique,Frillagalma_Gonfem_unique,Frillagalma_Gonmal_unique,Frillagalma_Necdev_unique,Frillagalma_Palmat_unique))),
        rep("Apolemia lanosa",length(c(Apolemia_Pne_unique,Apolemia_Gasdev_unique))),
        rep("Physalia physalis",length(c(Physalia_Gasdev_unique,Physalia_Gasmat_unique,Physalia_Tenpalmat_unique,Physalia_Tenpaldev_unique,Physalia_Tendev_unique))),
        rep("Diphyes dispar",length(c(Diphyes_Gasdev_unique,Diphyes_Gasmat_unique,Diphyes_Gon_unique)))
        ),

zooid=c(rep("Pneumatophore",length(Agalma_Pne_unique)),
        rep("Nectophore developing",length(Agalma_necdev_unique)),
        rep("Gonodendron male",length(Agalma_Gonmal_unique)), 
        rep("Gonodendron female",length(Agalma_Gonfem_unique)),
        rep("Gastrozooid mature",length(Agalma_gasmat_unique)),
        rep("Palpon B mature",length(Agalma_PalponB_unique)),
        rep("Palpon mature",length(Agalma_Palpon_unique)), 
        rep("Gastrozooid white developing",length(Bargmannia_Gasdev_unique)),
        rep("Gastrozooid yellow developing",length(Bargmannia_Gasyeldev_unique)), 
        rep("Nectophore developing",length(Bargmannia_necdev_unique)),
        rep("Gastrozooid white mature",length(Bargmannia_Gaswhimat_unique)),
        rep("Gastrozooid yellow mature",length(Bargmannia_Gasyelmat_unique)),
        rep("Gonodendron male",length(Bargmannia_gonmal_unique)),
        rep("Pneumatophore",length(Bargmannia_Pne_unique)),
        rep("Nectophore developing", length(Nanomia_Necdev_unique)),
        rep("Gastrozooid mature",length(Nanomia_Gasmat_unique)),
        rep("Palpon mature",length(Nanomia_Palmat_unique)),
        rep("Gonodendron female",length(Nanomia_Gonfem_unique)),
        rep("Pneumatophore",length(Nanomia_Pne_unique)),
        rep("Gastrozooid developing",length(Nanomia_Gasdev_unique)),
        rep("Palpon developing",length(Nanomia_Paldev_unique)),
        rep("Pneumatophore",length(Frillagalma_Pne_unique)),
        rep("Bract developing",length(Frillagalma_Bradev_unique)),
        rep("Gastrozooid developing",length(Frillagalma_Gasdev_unique)),
        rep("Gastrozooid mature",length(Frillagalma_Gasmat_unique)),
        rep("Gonodendron female",length(Frillagalma_Gonfem_unique)),
        rep("Gonodendron male",length(Frillagalma_Gonmal_unique)),
        rep("Nectophore developing",length(Frillagalma_Necdev_unique)),
        rep("Palpon mature",length(Frillagalma_Palmat_unique)),
        rep("Pneumatophore",length(Apolemia_Pne_unique)),
        rep("Gastrozooid developing",length(Apolemia_Gasdev_unique)),
        rep("Gastrozooid developing",length(Physalia_Gasdev_unique)),
        rep("Gastrozooid mature",length(Physalia_Gasmat_unique)),
        rep("Tentacular palpon mature",length(Physalia_Tenpalmat_unique)),
        rep("Tentacular palpon developing",length(Physalia_Tenpaldev_unique)),
        rep("Tentacle developing",length(Physalia_Tendev_unique)),
        rep("Gastrozooid developing",length(Diphyes_Gasdev_unique)),
        rep("Gastrozooid developing",length(Diphyes_Gasmat_unique)),
        rep("Gonozooid",length(Diphyes_Gon_unique))
        )
  )

DE_zooids<-tibble(
  sequence_ids=c(Agalma_Pne,Agalma_necdev, Agalma_Gonmal,Agalma_Gonfem,Agalma_gasmat, Agalma_PalponB, Agalma_Palpon,Bargmannia_Gasdev,Bargmannia_Gasyeldev,Bargmannia_necdev,Bargmannia_Gaswhimat,Bargmannia_Gasyelmat,Bargmannia_Gonmal,Bargmannia_Pne,Nanomia_Necdev,Nanomia_Gasmat,Nanomia_Palmat,Nanomia_Gonfem,Nanomia_Pne,Nanomia_Gasdev,Nanomia_Paldev,Frillagalma_Pne,Frillagalma_Bradev,Frillagalma_Gasdev,Frillagalma_Gasmat,Frillagalma_Gonfem,Frillagalma_Gonmal,Frillagalma_Necdev,Frillagalma_Palmat,Apolemia_Pne,Apolemia_Gasdev, Physalia_Gasdev,Physalia_Gasmat,Physalia_Tenpalmat,Physalia_Tenpaldev,Physalia_Tendev,Diphyes_Gasdev,Diphyes_Gasmat,Diphyes_Gon),

species=c(rep("Agalma elegans",length(c(Agalma_Pne,Agalma_necdev, Agalma_Gonmal,Agalma_Gonfem,Agalma_gasmat, Agalma_PalponB, Agalma_Palpon))),
          rep("Bargmannia elongata",length(c(Bargmannia_Gasdev,Bargmannia_Gasyeldev,Bargmannia_necdev,Bargmannia_Gaswhimat,Bargmannia_Gasyelmat,Bargmannia_Gonmal,Bargmannia_Pne))),
        rep("Nanomia bijuga",length(c(Nanomia_Necdev,Nanomia_Gasmat,Nanomia_Palmat,Nanomia_Gonfem,Nanomia_Pne,Nanomia_Gasdev,Nanomia_Paldev))),
        rep("Frillagalma vityazi",length(c(Frillagalma_Pne,Frillagalma_Bradev,Frillagalma_Gasdev,Frillagalma_Gasmat,Frillagalma_Gonfem,Frillagalma_Gonmal,Frillagalma_Necdev,Frillagalma_Palmat))),
        rep("Apolemia lanosa",length(c(Apolemia_Pne,Apolemia_Gasdev))),
        rep("Physalia physalis",length(c(Physalia_Gasdev,Physalia_Gasmat,Physalia_Tenpalmat,Physalia_Tenpaldev,Physalia_Tendev))),
        rep("Diphyes dispar",length(c(Diphyes_Gasdev,Diphyes_Gasmat,Diphyes_Gon)))
        ),

zooid=c(rep("Pneumatophore",length(Agalma_Pne)),
        rep("Nectophore developing",length(Agalma_necdev)),
        rep("Gonodendron male",length(Agalma_Gonmal)), 
        rep("Gonodendron female",length(Agalma_Gonfem)),
        rep("Gastrozooid mature",length(Agalma_gasmat)),
        rep("Palpon B mature",length(Agalma_PalponB)),
        rep("Palpon mature",length(Agalma_Palpon)), 
        rep("Gastrozooid white developing",length(Bargmannia_Gasdev)),
        rep("Gastrozooid yellow developing",length(Bargmannia_Gasyeldev)), 
        rep("Nectophore developing",length(Bargmannia_necdev)),
        rep("Gastrozooid white mature",length(Bargmannia_Gaswhimat)),
        rep("Gastrozooid yellow mature",length(Bargmannia_Gasyelmat)),
        rep("Gonodendron male",length(Bargmannia_Gonmal)),
        rep("Pneumatophore",length(Bargmannia_Pne)),
        rep("Nectophore developing", length(Nanomia_Necdev)),
        rep("Gastrozooid mature",length(Nanomia_Gasmat)),
        rep("Palpon mature",length(Nanomia_Palmat)),
        rep("Gonodendron female",length(Nanomia_Gonfem)),
        rep("Pneumatophore",length(Nanomia_Pne)),
        rep("Gastrozooid developing",length(Nanomia_Gasdev)),
        rep("Palpon developing",length(Nanomia_Paldev)),
        rep("Pneumatophore",length(Frillagalma_Pne)),
        rep("Bract developing",length(Frillagalma_Bradev)),
        rep("Gastrozooid developing",length(Frillagalma_Gasdev)),
        rep("Gastrozooid mature",length(Frillagalma_Gasmat)),
        rep("Gonodendron female",length(Frillagalma_Gonfem)),
        rep("Gonodendron male",length(Frillagalma_Gonmal)),
        rep("Nectophore developing",length(Frillagalma_Necdev)),
        rep("Palpon mature",length(Frillagalma_Palmat)),
        rep("Pneumatophore",length(Apolemia_Pne)),
        rep("Gastrozooid developing",length(Apolemia_Gasdev)),
        rep("Gastrozooid developing",length(Physalia_Gasdev)),
        rep("Gastrozooid mature",length(Physalia_Gasmat)),
        rep("Tentacular palpon mature",length(Physalia_Tenpalmat)),
        rep("Tentacular palpon developing",length(Physalia_Tenpaldev)),
        rep("Tentacle developing",length(Physalia_Tendev)),
        rep("Gastrozooid developing",length(Diphyes_Gasdev)),
        rep("Gastrozooid developing",length(Diphyes_Gasmat)),
        rep("Gonozooid",length(Diphyes_Gon))
        )
  )

DE_zooids<-DE_zooids %>% dplyr::left_join(blast_lookup,by="sequence_ids") %>% dplyr::left_join(GO,by="sequence_ids") %>% dplyr::left_join(reduced_nodes, by="sequence_ids")

unique_zooids<-unique_zooids %>% dplyr::left_join(blast_lookup,by="sequence_ids") %>% dplyr::left_join(GO,by="sequence_ids") %>% dplyr::left_join(reduced_nodes, by="sequence_ids")

readr::write_csv(unique_zooids, "Supplementary_Data/Supplementaryfile1.csv", na = "NA")

readr::write_csv(DE_zooids, "Supplementary_Data/Supplementaryfile2.csv", na = "NA")

#### GO analyses -- commented out to speed up knit time

# Bargmannia_GO_Gasmat<- Go_lookup("Bargmannia",Bargmannia_Gaswhimat)
# Agalma_GO_Gasmat<- Go_lookup("Agalma",Agalma_gasmat)
# Nanomia_GO_Gasmat<- Go_lookup("Nanomia",Nanomia_Gasmat)
# Frillagalma_GO_Gasmat<- Go_lookup("Frillagalma",Frillagalma_Gasmat)
# Physalia_GO_Gasmat<- Go_lookup("Physalia",Physalia_Gasmat)
# 
# Bargmannia_GO_Gasdev<- Go_lookup("Bargmannia",Bargmannia_Gasdev)
# Nanomia_GO_Gasdev<- Go_lookup("Nanomia",Nanomia_Gasdev)
# Frillagalma_GO_Gasdev<- Go_lookup("Frillagalma",Frillagalma_Gasdev)
# Physalia_GO_Gasdev<- Go_lookup("Physalia",Physalia_Gasdev)
# 
# Bargmannia_GO_Gonmal<- Go_lookup("Bargmannia",Bargmannia_Gonmal)
# Frillagalma_GO_Gonmal<- Go_lookup("Frillagalma",Frillagalma_Gonmal)
# Agalma_GO_Gonmal<- Go_lookup("Agalma",Agalma_Gonmal)
# 
# Frillagalma_GO_Gonfem<- Go_lookup("Frillagalma",Frillagalma_Gonfem)
# Agalma_GO_Gonfem<- Go_lookup("Agalma",Agalma_Gonfem)
# Nanomia_GO_Gonfem<- Go_lookup("Nanomia",Nanomia_Gonfem)
# 
# Nanomia_GO_Palmat<- Go_lookup("Nanomia",Nanomia_Palmat)
# Frillagalma_GO_Palmat<- Go_lookup("Frillagalma",Frillagalma_Palmat)
# Agalma_GO_Palmat<- Go_lookup("Agalma", Agalma_Palpon)

# Nanomia_GO_Necdev<- Go_lookup("Nanomia",Nanomia_Necdev)
# Frillagalma_GO_Necdev<- Go_lookup("Frillagalma",Frillagalma_Necdev)
# Agalma_GO_Necdev<- Go_lookup("Agalma", Agalma_necdev)
# Bargmannia_GO_Necdev<-Go_lookup("Bargmannia", Bargmannia_necdev)
# 
# Diphyes_GO_Gon<- Go_lookup("Diphyes",Diphyes_Gon)

Bargmannia_GO_Gaswhimat<-Go_lookup("Bargmannia", Bargmannia_Gaswhimat)
Bargmannia_GO_Gasyelmat<-Go_lookup("Bargmannia", Bargmannia_Gasyelmat)




##identify putative transcription factors -- emphasis on putative. this will find all genes with DNA binding function

TF<-rep("N",nrow(unique_zooids))
TF[grep("GO:0003700",unique_zooids$GO_term)]<-"Y"

unique_zooids<-unique_zooids %>% dplyr::bind_cols(TF=TF)

TF2<-rep("N",nrow(DE_zooids))
TF2[grep("GO:0003700",DE_zooids$GO_term)]<-"Y"
DE_zooids<-DE_zooids %>%  dplyr::bind_cols(TF=TF2)

x<-DE_zooids %>% dplyr::mutate(zooid = stringr::str_replace(zooid,"Gastrozooid white mature","Gastrozooid mature")) %>% dplyr::mutate(zooid = stringr::str_replace(zooid,"Gastrozooid white developing","Gastrozooid developing")) %>% dplyr::mutate(blast_hit=stringr::str_replace(blast_hit,"\\w*\\|\\w*_\\w* ", ""))  %>% dplyr::mutate(blast_hit=stringr::str_replace(blast_hit,"\\{[\\|\\w\\:\\d]+\\}", "")) %>% dplyr::filter(species%in%c("Agalma elegans","Bargmannia elongata","Nanomia bijuga","Frillagalma vityazi"),zooid%in%c("Pneumatophore","Nectophore developing","Palpon mature","Gastrozooid mature","Gastrozooid developing","Gonodendron female","Gonodendron male"),TF=="Y") 

#a curated list based on the identified transcription factors

list<-readr::read_csv("transcriptionfactors")

x<-x[x$blast_hit%in%list$blast_hit,]

x<- x %>% dplyr::left_join(list)

x$short_blast_hit<-factor(x$short_blast_hit, levels=x$short_blast_hit[order(x$zooid)] %>% unique())

Agalma_TF<-x %>% dplyr::filter(species=="Agalma elegans") %>% .$short_blast_hit %>%  unique()
Bargmannia_TF<-x %>% dplyr::filter(species=="Bargmannia elongata") %>% .$short_blast_hit %>%  unique()
Frillagalma_TF<-x %>% dplyr::filter(species=="Frillagalma vityazi") %>% .$short_blast_hit %>%  unique()
Nanomia_TF<-x %>% dplyr::filter(species=="Nanomia bijuga") %>% .$short_blast_hit %>%  unique()

##check how many of these putative transcription factors are present in the original transcriptome (are differences between species a function of sequencing depth?)
check_hit<-function(object,item){
  check<-function(item,object){
  x<-object@blast_hit %>% grep(item,.)
  if(length(x>0)){
  return(TRUE)
  }
  else{
  return(FALSE)
  }
  }
  true_false<-lapply(item,check,object)
  true_false<-true_false %>% unlist()
  return(true_false)
}
y<-lapply(e,check_hit, unique(x$blast_hit))

```

We sequenced mRNA from microdissected zooids from seven different siphonophore species (at least two replicate colonies) and mapped these short-read libraries to previously published transcriptomes [@munro2018improved]. We collected RNA-seq data from, where possible, 5 different zooids and one specialized tissue, the pneumatophore, as well as unique zooids specific to *Agalma elegans* (B palpons), *Physalia physalis* (tentacular palpon), and *Bargmannia elongata* (yellow and white gastrozooids), and where possible developing and mature zooids (Table \@ref(tab:summarize-libraries)). Due to species availability, we were not able to sample more than one replicate for some zooids - single replicates were excluded from downstream differential gene expression analyses (see Fig. \@ref(fig:zooids-sampled)). 

The first component of variation that we assessed was among technical replicates. The technical replicates consist of re-sequenced developing nectophores and developing gastrozooids from the same *Frillagalma vityazi* individual that were spiked in across multiple lanes and runs. Lane and run effects have been proposed as major sources of technical variability in RNA-seq data that may confound observations of biological variation [@auer2010statistical; @mcintyre2011rna]. The differences between technical replicates (Fig. \@ref(fig:technical-replicate)) were found to be much smaller (0.39% variance of expression distance) than the differences between zooids (98.32% variance of expression distance). Differences among technical replicates of the same zooid were correlated with library size and run, not by lane.

The next component of variation we considered was biological variation among sampled colonies. Within species, the greatest variance was among zooids/tissues, as opposed to between biological replicates of the same zooid/tissue (Figs. \@ref(fig:pca-diphyes)--\@ref(fig:pca-physalia)), with some exceptions, such as a developing palpon replicate in *Nanomia bijuga* (Fig. \@ref(fig:pca-nanomia)), and a developing gastrozooid and male gonodendra in *Frillagalma vityazi* (Fig. \@ref(fig:pca-Frillagalma)). Specimens were collected in the wild at different depths and over different time periods, but despite these potentially confounding environmental factors, the major variation we observe is among zooids/specialized tissues within species.

(ref:dge-zooids) Differentially expressed genes in siphonophore zooids, with two species *Agalma elegans* and *Bargmannia elongata* shown here. Within the circles, the number of differentially expressed genes found uniquely within a particular zooid (bold) and the total number of differentially expressed genes identified in pairwise comparisons between zooids (square brackets). Connections between zooids show the number of highly differentially expressed genes shared among those zooids. *Due to the very large number of shared genes among Palpons and B Palpons, this number includes genes that are shared between the Palpon and B Palpon. 


```{r dge-zooids, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.7\\textheight', out.width='0.7\\textheight', fig.cap= '(ref:dge-zooids)', fig.align = 'center'}

knitr::include_graphics("figures/DE_all.pdf",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

```

We identified genes that are significantly upregulated in particular zooids, and in the pneumatophore, for each of the sampled species. We also identified genes that are significantly upregulated in only one zooid and not in any other within a given species (Fig. \@ref(fig:dge-zooids), \@ref(fig:unique-zooids)). The list of genes that were identified as significantly differentially expressed only in a given species and zooid, and the associated transcriptome sequence ids, top blast hits, GO terms, and gene phylogeny ids, are available as supplementary data (supplementary file 1). The full list of genes upregulated in particular zooids, based on pairwise differential expression analyses, are available in supplementary file 2. 

For each species and zooid, we found GO term enrichment for biological processes consistent with the functional specialization of the zooid. In gastrozooids that are solely responsible for feeding and digestion, for example, we found these zooids to be enriched for genes involved in proteolysis, chitin, glutathione and peptide catobolism, as well as metabolism of carbohydrates. Likewise for male gonodendra, we found GO term enrichment for biological processes such as sperm flagellum, mitotic cell cycle process, DNA replication. For female gonodendra, there was GO term enrichment for a number of biological processes including mitotic cell cycle process, DNA replication (in *Agalma elegans*), as well as a number of signalling pathways and developmental processes (in *Frillagalma vityazi*), likely reflecting maternal provisioning to the egg [REF].

Within the four best sampled species , we also identified `r x$short_blast_hit %>% unique() %>% length()` differentially expressed putative transcription factors (`r length(Bargmannia_TF)` out of `r y[[2]] %>% sum()` in *Bargmannia elongata*, `r length(Frillagalma_TF)` out of `r y[[4]] %>% sum` identified in *Frillagalma vityazi*, `r length(Agalma_TF)` out of `r y[[1]] %>% sum()` in *Agalma elegans* and `r length(Nanomia_TF)` out of `r y[[5]] %>% sum()` identified  in *Nanomia bijuga*). Many of these transcription factors are significantly differentially expressed in several zooids regardless of developmental stage (both mature and developing zooids), and a subset were found to be significantly differentially expressed only in particular zooids and developmental stages (\@ref(fig:transcription-factors)). 

```{r summarize-libraries, verbose= FALSE, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

summary<-lapply(e,function(object){
  x=data.frame(
    Species=object@species, 
    Genes_in_trees=length(rownames(object@x)[rownames(object@x) %in% reduced_nodes$sequence_ids]),
    Genes_trees=reduced_nodes$gene_tree[reduced_nodes$sequence_ids %in% rownames(object@x)] %>% unique() %>% length()
    )
  x$Species<-as.character(x$Species)
  return(x)})

summary<-summary %>% dplyr::bind_rows() %>% arrange(desc(Genes_in_trees))
names(summary)<-c("Species", "Gene trees","Unique gene trees")

expression_data<-lapply(dge, function(object){
if(object@ddsMF$treatment=="Tentacle developing"){
  x=data.frame(
    Genes= DESeq2::counts(object@ddsMF) %>% nrow(),
    Zooids=object@ddsMF$treatment[!object@ddsMF$treatment=="Tentacle developing"] %>% unique() %>% length()
  )
}
    else{
    x=data.frame(
    Genes= DESeq2::counts(object@ddsMF) %>% nrow(),
    Zooids=object@ddsMF$treatment %>% unique() %>% length()
  )
  return(x)
  }
})

expression_data<-expression_data %>% dplyr::bind_rows()

expression_data$Species<-c("Agalma elegans","Bargmannia elongata","Apolemia lanosa","Frillagalma vityazi","Nanomia bijuga","Physalia physalis","Diphyes dispar")

summary<- summary %>% dplyr::left_join(expression_data)

names(summary) <- c("Species","Number of Genes in Gene Trees","Total Number of Gene Trees","Number of Genes","Number of Zooids Sampled")
summary<- summary %>% select(Species,`Number of Genes`, `Number of Zooids Sampled`,`Number of Genes in Gene Trees`,`Total Number of Gene Trees`)

summary %>% dplyr::mutate(
    Species= cell_spec(Species, italic= TRUE)
) %>% kable(caption="Number of genes per  species (library size), number of sampled zooids (at least two  replicates, including developing and mature stages), total number of genes in gene trees (there can be multiple genes in a gene tree), and number of unique gene trees containing genes from this species.", caption.short ="Summary statistics for each of the sampled species, with details on the total number of genes in gene trees, and number of unique gene trees containing genes from this species.",booktabs=TRUE, align = c("l","c","c","c","c"), escape = FALSE) %>% kable_styling(full_width = F, font_size = 11) %>%
  kableExtra::kable_styling(latex_options = "hold_position") %>%
  column_spec(2:5, width = "5em")

```

### Unique species-specific zooids {-}

In siphonophores, there are several instances of lineage-specific zooid diversification events. We investigated gene expression patterns between the novel zooid type and the hypothesized ancestral type in three species. In *Bargmannia elongata* there are two morphologically distinct gastrozooids, that we termed "white" and "yellow" gastrozooids (Fig. \@ref(fig:novel-zooids)A and \@ref(fig:novel-zooids)B). The "yellow" gastrozooid is larger and darker and occurs as the 7th-10th gastrozooid on the stem [@dunn2005complex].  In the Portuguese man of war, *Physalia physalis*, the gastrozooid is unique compared to other gastrozooids in other species - it has a mouth, but no tentacle, and the basigaster region is greatly reduced [@Totton1960; @Mackie1960]. Meanwhile the tentacle is associated with another zooid, the tentacular palpon (Fig. \@ref(fig:novel-zooids)C) [@BARDI2007;@munro2019morphology; @Totton1960; @haeckel1888]. In *P. physalis*, both the gastrozooid and the tentacular palpon are considered to be subfunctionalized from an ancestral gastrozooid type [@munro2019morphology]. Finally, in *Agalma elegans*, there are thought to be at least two different palpon types: gastric palpons that arise at the base of the peduncle of the gastrozooid, and a palpon called the B-palpon (Fig. \@ref(fig:novel-zooids)D) [@dunn2006evolution]. The distinction between these two types of palpon is based on the location of these zooids - the gastrozooid is typically the last element of each cormidium, but based on the budding sequence, Dunn *et al* propose that the enlarged B-palpon is the last element in *A. elegans* [@dunn2006evolution]. Each of these cases represents a different type of novelty: in *Bargmannia elongata* the distinction between zooids was made based on size and color but not on obvious differences in function, in *P. physalis* the gastrozooids and tentacular palpons differ structurally and functionally, and finally in *A. elegans*, gastric palpons and B palpons differ only in colony location, development, and possibly size. 

A large number of genes were identified as being significantly differentially expressed between mature gastrozooids and the tentacular palpons in *Physalia physalis* (`r dge_sig[[6]]$TenpalmatGasmat %>% nrow()` genes in the mature tentacular palpon and `r dge_sig[[6]]$GasmatTenpalmat %>% nrow()` genes in the mature gastrozooid). A number of genes were found to be differentially expressed in the mature tentacular palpon relative to all other tissues, of which, `r Physalia_Tenpalmat_unique %>% length()` genes were found to have expression patterns that were not shared with the gastrozooid. In the gastrozooid `r Physalia_Gasmat_unique %>% length()` genes were found to have expression patterns that were not shared with the tentacular palpon. 

Between the white mature gastrozooid and the yellow mature gastrozooid in *Bargmannia elongata*, few significant expression differences were identified (`r dge_sig[[2]]$GaswhimatGasyelmat %>% nrow()` genes were up in "white" mature gastrozooids relative to `r dge_sig[[2]]$GasyelmatGaswhimat %>% nrow()` genes in "yellow" gastrozooids). Among genes that were found to be significantly differentially expressed in either "white" or "yellow" gastrozooids relative to all other tissues, `r Bargmannia_Gasyelmat_unique %>% length()` genes were unique to "yellow" gastrozooids and not found in "white" gastrozooids, and `r Bargmannia_Gaswhimat_unique %>% length()` genes were found in "white" gastrozooids and not found in "yellow" gastrozooids (Fig. \@ref(fig:dge-zooids), bottom panel). 

Finally, in *Agalma elegans* very few differentially expressed genes were identified between the B palpon and gastric palpons  (`r dge_sig[[1]]$PalBmatPalmat %>% nrow()` and `r dge_sig[[1]]$PalmatPalBmat %>% nrow()` genes respectively. All three genes have no significant blast hit and did not map to any gene trees). Genes were identified that are upregulated in B palpons relative to all other zooids (gastric palpons were excluded). Of this, `r Agalma_PalponB_unique %>% length()` genes were found to be differentially expressed in the B palpon. Most of these genes overlapped with those found to be differentially expressed in the gastric palpon relative to all other zooids (`r intersect(Agalma_PalponB_unique,Agalma_Palpon_unique) %>% length()` genes) (Fig. \@ref(fig:dge-zooids), top panel).

### Evolution of gene expression {-}

```{r plot_tree, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE}

change<-function(col_value,nhx){
  parents = nhx@phylo$edge[,1]
  children = nhx@phylo$edge[,2]
  edge_length=nhx@phylo$edge.length
  df = data.frame(
    parent=parents,
    node=children,
    edge_num=round((as.numeric( pull(nhx@data, col_value)[children] )-as.numeric( pull(nhx@data,col_value)[parents] ))/edge_length,2)
    )
  return(df)
}

plottreeexpression<-function(gene_tree,col_value){
  
nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==gene_tree][[1]]
    
edge<-change(col_value,nhx)

levels(nhx@data$Ev)<-c("D","S")

tips<-pull(nhx@data,col_value)

top_hit<-nhx@data$blast_hit %>% median(.,na.rm=TRUE)

p<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names ), hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,1))

p<-p+geom_text(aes(label=round(tips,3)),vjust=-1,hjust=1, size=3, col="Black")

p<- p %<+% edge + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3)  

p<- p+ggtitle(paste0("Most frequent blast hit=",top_hit))
return(p)
}

plotsimtreeexpression<-function(gene_tree,col_value){
  
nhx<-gene_trees_sim_calibrated_pruned[gene_trees_sim_digests==gene_tree][[1]]

blast_hit <- genes_to_tips %>% dplyr::filter(gene_tree==gene_tree) %>% .$blast_hit
    
edge<-change(col_value,nhx)

levels(nhx@data$Ev)<-c("D","S")

tips<-pull(nhx@data,col_value)

top_hit<-nhx@data$blast_hit %>% max(.,na.rm=TRUE)

p<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names ), hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,1))

p<-p+geom_text(aes(label=round(tips,3)),vjust=-1,hjust=1, size=3, col="Black")

p<- p %<+% edge + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3)  

p<- p+ggtitle(paste0("Most frequent blast hit=",top_hit))
return(p)
}


plotmultitree<-function(gene_tree,col_value){
  
nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==gene_tree][[1]]

levels(nhx@data$Ev)<-c("D","S")

tips<-c()
for(tip in 1:length(col_value)){
tips$col_value[[tip]]<-pull(nhx@data,col_value[[tip]])
}

p<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names ), hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,2))

p<-p+geom_text(aes(label=round(tips$col_value[[1]],3)),vjust=-4.5,hjust=1, size=3, col="Blue")

p<-p+geom_text(aes(label=round(tips$col_value[[2]],3)),vjust=-3.5,hjust=1, size=3, col="Red")

p<-p+geom_text(aes(label=round(tips$col_value[[3]],3)),vjust=-2.5,hjust=1, size=3, col="Dark Green")

p<-p+geom_text(aes(label=round(tips$col_value[[4]],3)),vjust=-1.5,hjust=1, size=3, col="Orange")

p<-p+geom_text(aes(label=round(tips$col_value[[5]],3)),vjust=1.5,hjust=1, size=3, col="Dark Red")

p<-p+geom_text(aes(label=round(tips$col_value[[6]],3)),vjust=2.5,hjust=1, size=3, col="Black")

p<-p+geom_text(aes(label=round(tips$col_value[[7]],3)),vjust=3.5,hjust=1, size=3, col="Purple")

p<-p+geom_text(aes(label=round(tips$col_value[[8]],3)),vjust=4.5,hjust=1, size=3, col="Dark Grey")

return(p)
}
```
In order to compare expression patterns for each zooid across siphonophore species, we used transcriptome and genome data from `r species_tree$tip.label %>% length()` cnidarian species to generate a total of `r gene_trees %>% length()` gene trees, of which `r gene_trees_ace_calibrated[!is.na(gene_trees_ace_calibrated)] %>% length()` gene trees passed filtering criteria. We used Orthofinder [@emms2019orthofinder] to reconcile the species tree and the gene trees and identify nodes that represent speciation and duplication events (Fig. \@ref(fig:methods-comp), step 1). Pruned and unpruned calibrated gene tree files are available as supplementary data. The number of genes represented in these gene trees is shown in table \@ref(tab:summarize-libraries). The internal nodes on these gene trees consist of `r nodes_ace %>% filter(is.na(species) & Ev=="S") %>% nrow()` speciation events and `r nodes_ace %>% filter(is.na(species) & Ev=="D") %>% nrow()` duplication events. Expression values, as transcripts per million 10K (TPM10K, see methods), were mapped to the tips of the gene trees (Fig. \@ref(fig:methods-comp), step 2), and maximum likelihood trait values were reconstructed at the nodes (Fig. \@ref(fig:methods-comp), step 3). Changes in expression across branches were calculated based on the difference between parent and child nodes, scaled by the branch length (Fig. \@ref(fig:methods-comp), step 4) [@dunn2018pairwise]. We focused on expression in a subset of zooids and tissues that are common across the sampled species: gastrozooids (developing and mature), nectophores (developing), palpons (mature), male and female gonodendra, and the pneumatophore. An exemplar gene tree, for Wnt family genes, with expression values from two zooids mapped at the nodes and tips is shown in figure \@ref(fig:wnt-plot).  

(ref:methods-comp) Methods used to identify changes in expression across branches in gene trees. In step 1, we have labelled each of the nodes in the siphonophore phylogeny, and identified equivalent speciation nodes across every gene tree (an exemplar is shown here). Step 2, we map expression values to the tips (TPM10K). Step 3, reconstruct ancestral trait expression values at all nodes where expression data is available. Step 4, we calculate scaled change in expression across branches and divide by the branch length. Branch length is calibrated to the species tree branch lengths. Step 5, we identify branches in gene trees that correspond to equivalent branches in the species tree. There may be more than one branch in a gene tree that corresponds to the same branch in the species tree.

```{r methods-comp, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.7\\textheight', out.width='0.7\\textheight', fig.cap='(ref:methods-comp)', fig.align = 'center'}

knitr::include_graphics("figures/Methods_figure.pdf")

```

We first considered the evolutionary change in expression along branches in the gene tree that correspond to branches in the species tree. These are identified as gene tree branches that have parent and child nodes that are both speciation events and that correspond to speciation events and branches in the species tree (Fig. \@ref(fig:methods-comp), step 5). This method enables the selection of specific branches within gene trees that are equivalent to branches within the species tree, and are thus comparable with one another across all gene trees. Unlike a strict 1:1 ortholog approach, this approach considers equivalent branches that are descended from speciation events, but that have more complex evolutionary histories. For example, due to deeper gene duplication events, gene trees often contain multiple branches that correspond to the same branch in the species tree (Fig. \@ref(fig:methods-comp), step 5). Our method allows us to consider all of these branches. Strict ortholog methods would discard some or all data impacted in this way by duplication events. Each speciation node in a gene tree is assigned an identifier that corresponds to the equivalent node in the species phylogeny. Each branch in the species tree is given a unique letter, as shown in figure \@ref(fig:changes), and the corresponding branches in gene trees are given the same letter. A simulated data set of random expression values was also obtained by using a Brownian motion model with empirically derived mean and standard-deviation values for each gene tree. Under this model, expression variance accumulates among linages on the gene tree as a function of time, and is used as a model of evolution under drift, as well as some forms of natural selection [@felsenstein1973maximum;@revell2008testing;@revell2008phylogenetic].

```{r changes, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=7.5, fig.height=7, fig.cap="Mean changes along branches across all gene trees that correspond to branches in the species tree, error bar is one standard deviation. Top panel: species phylogeny with branch IDs given as numbers. Lower panel: Distribution of changes along a branch in a gene tree, showing mean change in the empirical dataset and in the BM simulation. Branch refers to the branch ID in the species tree. Treatment type is coded in colour, Gasdev = Developing Gastrozooid, Gasmat= Mature gastrozooid, Gonfem= Female gonodendron, Gonmal= Male gonodendron, Necdev= Developing nectophore, Palmat = Mature palpon, Pne= pneumatophore.", fig.scap="Mean changes along branches across all gene trees that correspond to branches in the species tree, error bar is one standard deviation.", fig.align='center'}

layout<-rbind(c(1,1,NA),c(2,3,4))

grid.arrange(tree_reference,branch_change,sim_branch_change,leg, bottom="Branch",widths = c(2,2,0.5),layout_matrix = layout)

```

The distribution of expression changes along branches in gene trees that correspond to equivalent branches in the species tree are shown in figure \@ref(fig:changes). The variance structure of the empirical data matches that of the simulated data (Fig. \@ref(fig:variance-structure)), suggesting that the size of standard deviation of change around the mean that is observed in figure \@ref(fig:changes) is a function of the underlying structure of the tree and the ancestral trait reconstruction methods that were used. High variance is especially pronounced in branches in the gene tree that correspond to K and L that have a shared parent node and lead to sister taxa *Nanomia bijuga* and *Agalma elegans* (Fig. \@ref(fig:variance-structure)). These two taxa are the only sister tip values that were sampled in this study. The mean values of change along branches do, however, differ from the Brownian model (Fig. \@ref(fig:changes)). Using linear models we find that expression change along branches in gene trees that correspond to branches in the species tree can be explained by significant differences among zooids/the pneumatophore, branches, and the interaction of the two (p < 2.2e-16, p = 2.319e-13, p < 2.2e-16, respectively, two-tailed anova). The Brownian model, based on simulated values on the gene trees under BM (p=0.4438, p=0.9829, p=0.4694), and also based on random reshuffling of changes on branches (p=0.31218, p=0.06787, p=0.0.54698), show that expression change along branches in gene trees that correspond to branches in the species tree cannot be explained by significant differences among zooids/the pneumatophore, branches, and the interaction of the two. 

Global patterns of expression change across all branches and tissues/zooids provides perspective on potential linage-specific shifts in expression in each zooid/tissue. These results point to a subset of branches in gene trees for particular zooids with large increases or decreases in expression across the branch. These patterns were particularly strong for palpons, where on average changes in expression were highly negative across branches leading to the clade containing *Nanomia bijuga* and *Agalma elegans*, while on average changes in expression were highly positive across branches leading to *Frillagalma vityazi*. We identified `r edges_tidy %>% filter(S_child=="26" & change > 10 & treatment=="Palmat") %>% nrow()` gene tree branches in `r edges_tidy %>% filter(S_child=="26" & change > 10 & treatment=="Palmat") %>% .$gene_tree %>% unique() %>% length()`gene trees corresponding to branch J (branch leading to *Frillagalma vityazi*, fig \@ref(fig:changes)) with large (>10) positive changes in expression across the branch. Among these, we identified several gene families of interest that show large lineage specific increases in expression in Palpons, including genes involved in patterning and development such as Wnt receptor Frizzled-4, Wnt ligand Wnt2, and homeobox gene Otx-2. For the pneumatophore, we identified `r edges_tidy %>% filter(S_child=="26" & change > 10 & treatment=="Pne") %>% nrow()` gene tree branches corresponding to branch J in `r edges_tidy %>% filter(S_child=="26" & change > 10 & treatment=="Pne") %>% .$gene_tree %>% unique() %>% length()` gene trees with large (>10) positive changes in expression across the branch, including homeobox proteins SAX-1 and Hox-B8 - the latter gene showed low or no TPM10K expression in *Nanomia bijuga*, *Agalma elegans*, and *Apolemia lanosa* pneumatophores and high expression in *Frillagalma vityazi* pneumatophores. The full list of branches and changes in expression across the branch and tissue/zooid are available in supplementary file 3. 

```{r output_supp, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE}

new_edges <- edges_tidy %>% left_join(.,S_new, by=c("S_child"="S")) %>% dplyr::select(gene_tree, S_new, treatment, change, blast_hit) %>% dplyr::rename(.,c(Branch_ID=S_new,Zooid=treatment))

readr::write_csv(new_edges, "Supplementary_Data/Supplementaryfile3.csv", na = "NA")

#output trees - to be re run for any/all re-runs of prep files

# for(tree in 1:length(gene_trees_ace_calibrated_pruned)){
# if(!is.na(gene_trees_ace_calibrated_pruned[[tree]])){
# ape::write.tree(gene_trees_ace_calibrated_pruned[[tree]]@phylo,file=paste0("Gene_trees/Gene_trees_pruned/",gene_trees_ace_digests[[tree]],".txt"))
# }
# }
# 
# for(tree in 1:length(gene_trees_ace_calibrated)){
# if(!is.na(gene_trees_ace_calibrated[[tree]])){
# ape::write.tree(gene_trees_ace_calibrated[[tree]]@phylo,file=paste0("Gene_trees/Gene_trees/",gene_trees_ace_digests[[tree]],".txt"))
# }
# }

```

Next, we were interested in branch level patterns of expression, focusing on whether change in expression across the same branch in a gene tree is correlated between different zooids. We investigated the co-variance structure of changes across particular branches (that correspond to species tree branches) within gene trees among different zooids/the pneumatophore (Figs. \@ref(fig:covariances)). For change across the same branch within a gene tree that corresponds to a particular branch in the species tree, there is a positive correlation between pairwise comparisons of zooids/the pneumatophore. In particular, changes across all branches are highly correlated between developing and mature gastrozooids. Changes in expression across the same branch were also highly correlated between gastrozooids and palpons, especially so across branches corresponding to branch K (\@ref(fig:changes), top panel) in the species tree (branch leading to *Agalma elegans*).  Meanwhile, changes across all branches are much less correlated between any zooid/the pneumatophore and both the male or female gonodendra. These results suggest that changes in expression across gene tree branches are highly correlated among zooids - a large positive change in expression in one zooid across a gene tree branch that corresponds to branch L,  for example, will also typically see a large positive change in expression in another zooid across the same branch.

```{r covariances, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=7.5, fig.height=7, fig.cap="Siphonophore phylogeny and pairwise correlation of changes across a branch among different pairs of zooids/the pneumatophore. Correlation of changes from node to tip are shown in the plot to the right of the phylogeny. Changes between two nodes are shown adjacent to the child node. Color indicates correlation between 0.9 (red) and 0.5 (green). Treatment type is coded in colour along the x-axis, Gasdev = Developing Gastrozooid, Gasmat= Mature gastrozooid, Gonfem= Female gonodendron, Gonmal= Male gonodendron, Necdev= Developing nectophore, Palmat = Mature palpon, Pne= pneumatophore.", fig.align='center'}

knitr::include_graphics("Figures/Correlations.pdf")

```

We were then able to identify instances of branches in gene trees where change in expression across the branch is negatively or positively correlated among different gene trees. One such example of lineage and zooid specific changes in expression is in a putative hydrozoan specific actinoporin-like gene family, with closest blast similarity to Auger and Cone snail tereporin/coroporin toxins. In this toxin-like family, we were able to identify one clade (clade Y, Fig. \@ref(fig:venom-plot)) with three branches corresponding to branches K, L and I in the species tree. The branches that correspond to I and K, leading to a *Diphyes dispar* and *Agalma elegans* gene respectively, both have large increases in mature gastrozooid expression across the branch, while the branch that corresponds to branch L (leading to *Nanomia bijuga*) has a large decrease in expression across the branch in mature gastrozooids and an independent large increase in expression for the same branch in mature palpons. For all other zooids/tissues the expression at the tips is close to 0, with only small changes in expression across the branch. In *Nanomia bijuga* the gene 976500 in Clade Y is significantly differentially expressed in mature palpons as compared to mature gastrozooids (log2 Fold Change 6.83224, Bonferroni adjusted p-value 0.000567528). These results point towards a lineage and zooid specific change in expression.

This gene family shows a complex history of duplication events, with several deep duplication events prior to the divergence of siphonophores, and multiple species-specific duplication events. The exact function of these toxin-like genes is not known or characterized, however they appear to be found predominantly in gastrozooids and/or palpons. In the five duplicated *Physalia physalis* genes, expression is highest in mature gastrozooids relative to developing and mature tentacular palpons (the zooid that bears the tentacle), as well as the developing tentacle. The full phylogeny of all available hydrozoan genes is available in supplementary figure \@ref(fig:fullvenom-plot). The three *Clytia hemisphaerica* homologs are highly expressed specifically in the polyp head, and whole male and female medusa (blast search on Marine Invertebrate Models Database http://marimba-test.obs-vlfr.fr/, XLOC__029306, XLOC_013838, XLOC_029344).

(ref:venom-plot) Pruned maximum likelihood gene tree of a venom gene family found in siphonophore species with expression values (TPM10K) at the tips and nodes. Mature gastrozooid expression in orange (above node), palpon expression in yellow (below node). Changes in expression across the branch are found in black boxes along the branch, orange = mature gastrozooid, yellow = mature  palpon. Red circles are duplication nodes and blue circles are speciation nodes.

```{r venom-plot, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.height=15, fig.width=15, fig.cap='(ref:venom-plot)', fig.align = 'center'}

#pull gene tree digest

venom<-edges_tidy[grep("Tereporin-Ca1",edges_tidy$blast_hit),] %>% .$gene_tree %>% unique()

nhx_wider<-gene_trees_ace_calibrated[gene_trees_ace_digests==venom][[1]]
venom_p<-nhx_wider %>% ggtree()+geom_nodepoint(aes( color=Ev ))+geom_tiplab() +xlim(c(0,0.5))
nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==venom][[1]]
tips1<-pull(nhx@data,"Gasmat")
tips2<-pull(nhx@data,"Palmat")
p1<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names),pch=1, hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,0.5))
p1<-p1+geom_text(aes(label=round(tips1,2)),vjust=-1,hjust=1, size=3, col="#F97F06")
p1<-p1+geom_text(aes(label=round(tips2,2)),vjust=1.5,hjust=1, size=3, col="#F7D114")

edge <- change("Gasmat",nhx)
edge2<- change("Palmat",nhx)
p2<- p1 %<+% edge + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3, col="#F97F06")  
p3 <- p1 %<+% edge2 + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3, col="#F7D114")

##used the output from above to make the toxin plot

knitr::include_graphics("Figures/Toxin_plot.pdf",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

Physalia<-data.frame(treatment=e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@treatment, individual=e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@individual)

venom_Physalia <- e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@tpm[rownames(e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@tpm) %in% c("344760","344764","388102","388106","368190"),] %>% t %>% as.data.frame()

#plot of counts of all Physalia genes
#venom_physalia %>% bind_cols(.,Physalia) %>% gather(1:5,key=gene,value=count) %>% ggplot()+geom_point(aes(x=treatment,y=log(count+1), col=gene))

```

## Discussion {-}

Colonial animals, consisting of genetically identical modules that are physically and physiologically connected, can be found across the metazoan tree of life [@hiebert2020coloniality]. Functional specialization has evolved multiple times within colonial groups, and siphonophores in particular are suggested to have the greatest diversity of functionally specialized zooids [@Beklemishev1969;@hiebert2020coloniality]. Functionally specialized siphonophore zooids have been likened to animal organs, with each functionally specialized zooid possessing distinct cell-types and performing distinct roles within the colony [@mackie1963]. While this analogy works well for understanding the function of zooids within the colony as a whole, the analogy falls short in terms of explaining the evolutionary origin and development of these biological units: functionally specialized zooids are evolutionarily homologous to free living organisms, they are multicellular, possess distinct cell types, and show regional subfunctionalization [@church2015histology;@Mackie1960;@totton1965synopsis]. While the developmental mechanisms generating zooids are very different (they arise asexually from a growth zone), the evolutionary processes acting on zooids may be similar to those acting on other modular biological units such as cell type, tissue, and organ [@hiebert2020coloniality]. Understanding the cellular and molecular processes underlying the patterning and molecular function of functionally specialized cnidarian zooids remain an open biological question, and in recent years there has been a focus in particular on differential gene expression patterns found in different functionally specialized zooids [@sanders2014differential; @sanders2015interspecific; @Siebert:2011gv]. In this study, we found a large number of genes that are differentially expressed within different siphonophore zooids, reflecting the distinct anatomy and function of these zooids. We were also able to identify potential transcription factors that are significantly differentially expressed within particular zooids, with potential homologs found in several species, which may be interesting candidates for future study in siphonophores or related cnidarian colonial groups.     

Unique zooid types and zooid delimitation

Evolution of gene expression

### Comparison to other methods {-}

With the expansion of functional genomic tools, including RNA-seq and single cell sequencing methods, we are able to look not only at how genomic variation gives rise to phenotypic diversity in a single species or organism, but also at how functional genomic variation shapes phenotypic diversity across a number (>2) of closely and distantly related species to understand broader evolutionary patterns and processes [@Brawand:2011du; @barbosa2012evolutionary; @breschi2016gene; @clarke2017evolutionary; @doi:10.1093/gbe/evw155; @merkin2012evolutionary; @necsulea2014evolution; @perry2012comparative;  @Levin:2016bp; @sudmant2015meta; @ma2018comparative; @yang2013organ; @zhang2014comparative]. The methods developed here are broadly applicable to a range of functional genomic data, in addition to existing RNA-seq data sets in other species.

This approach presents several solutions to past limitations. We apply ancestral trait reconstruction methods to reconstruct expression values at nodes in gene trees and calculate changes across branches. This has several advantages -- we are able to take species relationships into account, and we are also able to overcome sampling issues at the tips, as expression values of different treatments can be reconstructed at deep internal nodes, even where there may be inconsistent sampling at the tips. We also propose a novel solution to identifying comparable genes across species. Where past approaches identify strict orthologs *before* conducting analyses, we use information from all genes within a gene tree regardless of their evolutionary history of duplication or speciation. Subsequently, we are then able to use information at the nodes to select branches in gene trees that correspond to particular branches in the species tree. This enables the selection of branches that are descended from speciation events, but that do not necessarily belong to gene trees where only single copy genes are found. This enables us to consider vastly more genes than we would be able to with a strict ortholog approach. Additionally, we are also able to compare expression following duplication and speciation events. Finally, by using gene trees, we are able to consider expression evolution within the context of the gene phylogeny, which may differ significantly from the species phylogeny. This enables us to consider the evolution of gene expression in context, considering not only the evolutionary relationships between genes, but also between species.

There are, however, limitations to this approach as applied here. As with all methods that rely on mapping to reference transcriptomes rather than genomes, this approach is limited by the quality of the reference transcriptomes. Not all reference transcriptomes were sequenced to equal depth among species, and this has important effects on the presence or absence of genes from particular species within the gene tree. This not only has an effect on the representation of expression values, but also impacts the power to investigate patterns of expression among branches within a gene tree. However, with genome sequencing becoming cheaper and more readily available, the widespread availability of reference genomes will help alleviate many of these issues. Reference genomes will also improve gene models, enabling the distinction of different alleles of the same gene from paralogous genes, this in turn will improve the quality of the gene trees. 

## Methods {-}

All scripts for the analyses are available in a git repository at https://github.com/dunnlab/Siphonophore_Expression. The most recent commit at the time of the analysis presented here was XXXX.

**Collecting** \newline
Specimens were collected in the north-eastern Pacific Ocean in Monterey Bay and, in the case of _Physalia physalis_ the Gulf of Mexico. Specimens were collected by remotely operated vehicle (ROV) or during blue-water SCUBA dives. *Physalia* specimens were collected by hand from the beach after being freshly washed on-shore by prevailing winds. Available physical vouchers have been deposited at the Peabody Museum of Natural History (Yale University), New Haven, CT. Specimens were relaxed using 7.5% MgCl~2~ hexahydrate in Milli-Q water at a ratio of 1/3 MgCl~2~ and 2/3 seawater. Zooids were subsequently dissected from the colony and flash frozen in liquid nitrogen. Colonies were cooled to collection temperatures (e.g 4 degrees C for deep sea species) while the dissections took place. Dissections took no longer than 15-20 minutes. In the case of large colonies, the stem was cut and only partial sections of the colony were placed under the microscope at a given time. Each replicate individual represents a genetically distinct colony from the same species. Replicate specimens were of an equivalent colony size, and zooid replicates were also equivalent sizes. Larger zooid types, such as gastrozooids, were sampled as a single zooid, but smaller zooids were pooled. Pooled zooids were of a comparable maturity and sampled from the same location in a single colony. 

**Sequencing** \newline
mRNA was extracted directly from tissue using Zymo Quick RNA MicroPrep (Zymo #R1050), including a DNase step, and subsequently prepared for sequencing using the Illumina TruSeq Stranded Library Prep Kit (Illumina, #RS-122-2101). 50 base-pair single-end libraries were all sequenced on the HiSeq 2500 sequencing platform. Three sequencing runs were conducted, representing three full flow cells. To avoid potential run/lane confounding effects, where possible, libraries of multiple zooids/tissue of a single individual in a species were barcoded and pooled in a single sequencing lane, and replicate lanes of zooids/tissue from different individuals of the same species were sequenced in separate runs. Additionally, two libraries were run as technical replicates across all runs and many lanes, for a total of 20 technical replicates. Sequences, along with sampling metadata, are publically available on NCBI under BioProject ID PRJNA540747.

**Analysis**

**_Sequencing & Differential gene expression_** \newline
Short read libraries were mapped to previously published transcriptomes (150bp paired end) [@munro2018improved] using Agalma v 2.0.0 [@Dunn:2013kw; @Guang202416], which uses a number of existing tools for transcript quantification, including RSEM (which uses Bowtie) [@langmead2009ultrafast;@li2011rsem]. Using the `agalmar` package (https://github.com/caseywdunn/agalmar), we filtered out genes that were flagged as being rRNA, and selected only protein coding genes. We also only considered genes that were greater than 0 in at least two libraries. Differential gene expression analyses, including normalization, were conducted in R, using the `DESeq2` package [@love2014moderated]. Libraries that were found to be outliers based on mean cook's distance were removed from the DESeq object and from downstream analyses and normalization. Testing for differential expression was conducted using the `results()` function in DESeq2. Genes were considered to be significantly differentially expressed if adjusted p-values (Bonferroni correction) were less than 0.05. Differential expression analyses were only conducted on zooids/tissue with two or more replicates.

GO annotations were retrieved for each of the reference translated transcriptomes [@munro2018improved] using the PANNZER2 web server [@toronen2018pannzer2].The PANNZER2 format was modified to match the gene2GO format required for the package `topGO` [@topGO]. Gene set enrichment analyses were carried out within species using the R package `GOseq` [@young2010gene], which takes gene length into account. Over and underrepresented categories were calculated using the Wallenius approximation, and p-values were adjusted using the Benjamini and Hochberg method. Categories with an adjusted p-value below 0.05 are considered enriched. Gene set enrichment analyses were also conducted at the gene tree level, considering representative GO terms for particular gene trees. Representative GO terms were selected based on the frequency of occurrence among genes in the gene tree. As gene lengths vary among species and genes in the gene tree, the `GOseq` approach could not be used, and `topGO` was used to detect GO terms that are enriched based on Fisher's exact test. This approach assumes that each gene tree has an equal probability of having genes shared among species that are detected as differentially expressed, however results may be biased by a number of factors, including mean gene length among genes in the gene tree [@young2010gene]. 

**_Phylogenetic analysis of gene expression_** \newline
Gene trees were generated using the transcriptomes and genomes from `r species_tree$tip.label %>% length()` species [@munro2018improved] using Agalma v 2.0.0 [@Dunn:2013kw; @Guang202416]. Following the `treefinder` step of Agalma v 2.0.0, amino acid data were exported and supplied to Orthofinder [@emms2019orthofinder] for simultaneous co-estimation of gene trees with the published maximum likelihood species tree [@munro2018improved]. Within Orthofinder, the selected multiple sequence alignment method was MAFFT [@katoh2013mafft] and maximum likelihood tree inference method was IQ-tree with the LG+F+R4 substitution model [@nguyen2015iq].

Phylogenetic analyses were conducted in R using `geiger`, `ape`, `phytools`, `Rphylopars`, and `hutan` [@ape2004; @doi:10.1093/sysbio/syv055; @goolsby2017rphylopars; @harmon2007geiger; @MEE3:MEE3169]. Phylogenetic trees were visualized in R using `ggtree` and `treeio` [@yu2017ggtree]. Linear models were constructed using lm(), and wilcox tests were carried out using the function wilcox.test() in base R. See Supplementary Information for R package version numbers.

Gene trees were filtered to exclude trees with a length threshold >2, a root depth >5, and that had more than 0.25 branches with a default length value (that are indicative of branch length=0). Gene trees nodes were annotated as speciation events or duplication events, based on assignment by Orthofinder [@emms2019orthofinder]. Speciation nodes were subsequently assigned a node ID equivalent to the species tree node, using species tip names from the gene tree to determine the most common recent ancestor in the species tree, using the `phytools` package. Due to the use of the species-overlap method by Orthofinder, some clades of single copy genes were assigned  as speciation events, although the topology is inconsistent with the species tree. To avoid time calibration issues, due to descendant nodes being assigned the same species node ID, descendant speciation nodes with the same species node ID were marked as null, and gene trees with nodes greater than 0.3 null nodes per internal node were excluded -- indicating widespread topological differences with the species tree. Tips without expression values were pruned out of the tree. Gene trees with fewer than three expression values at the tips were discarded, retaining only trees with three or more values. Additionally, only trees with one or more speciation events were retained, as speciation events are used for time calibrations. The gene trees were then time calibrated to the species tree using `chronos()` in the `ape` package, so that the branch lengths were scaled to the same equivalent length across all gene trees [@ape2004]. Some gene trees could not be calibrated against the node constraints from the species tree and were discarded. 

For comparitive expression analyses, expression values were normalized using a method we are calling transcripts per million 10K (TPM10K). For gene $i$ of a given species, TPM is typically calculated as [@li2009rna]:

$$ TPM_i = \frac{10^6 \times \theta_i}{\ell_i \times \displaystyle\sum_{i=1}^{n}\frac{\theta_j}{\ell_j}} $$

Where $\theta_i$ is the number of the mapped reads to gene $i$, $\ell_i$ is the effective length of the gene, and n is the number of genes in the reference. The intent of this measure is to make libraries comparable within a single species. The sum of TPM values within a library is $10^6$, and the mean is $\frac{10^6}{n}$. One implication of this is that TPM values are not directly comparable across species, since in practice $n$ differs across species. If this were not accounted for, then it could appear, for example, that genes all have lower expression in a species with a more complete reference transcriptome and higher $n$. To account for differences in means among species, we use a new measure, TPM10K, that accounts for differences in $n$:

$$TPM10K_i= \frac{TPM_i \times n}{10^4}$$

Where the sum of TPM10K values within a library is $10^2 \times n$ and the mean is $10^2$. By multiplying by $n$ we are able to account for different sequencing depths among species, and ensure a common mean. As $n$ is large, we divide by an arbitrary number (in this case $10^4$) in order to reduce the magnitude of the expression value.

We then took the mean TPM10K value for each gene across replicates of the same zooid/tissue within a species and applied a log transformation. Using gene trees with expression values for each gene within a species at the tips, maximum likelihood ancestral trait values were generated at the nodes using the anc.recon() function in `Rphylopars` assuming a Brownian model of evolution (Fig. \@ref(fig:methods-comp), step 2 & 3) [@goolsby2017rphylopars]. As not all zooids are present in all of the species, the trees were pruned down to the subset of tips with expression values for ancestral trait reconstructions. Node values were then added back to the unpruned tree with all of the reconstructed expression values. Change in expression was measured across a branch by taking the difference between a parent node and a child node, and then this difference is scaled by branch length (Fig. \@ref(fig:methods-comp), step 4).

## Acknowledgements {-}

This work was supported by the National Science Foundation (DEB-1256695 and the Waterman Award). CM was also supported in part by a RI-EPSCoR Fellowship, NSF EPS-1004057. Analyses were conducted with computational resources and services at the Center for Computation and Visualization at Brown University, supported in part by the NSF EPSCoR EPS-1004057 and the State of Rhode Island. Analyses were also conducted with computational resources and services XXXXXXX Yale CCV

\clearpage
## Supplementary data {-}

\setcounter{figure}{0}
\setcounter{table}{0}
\makeatletter
\renewcommand{\thefigure}{S\@arabic\c@figure}
\renewcommand{\thetable}{S\@arabic\c@table}
\makeatother

(ref:zooids-sampled) Phylogeny of the focal species sampled in this study, with details of the traits sampled for each of the species. Phylogeny modified from Munro et al. 2018, @munro2018improved. Black indicates that multiple replicates have been sampled, grey indicates that no or only one replicate has been sampled, and white indicates that this zooid/tissue is not present in this species. The category "unique zooid" indicates that a zooid type that is unique to this species was sampled.

(ref:zooids-sampledshort) Phylogeny of the focal species sampled in this study, with details of the traits sampled for each of the species.

```{r zooids-sampled, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.7\\textheight', out.width='0.7\\textheight', fig.cap= '(ref:zooids-sampled)', fig.scap='(ref:zooids-sampledshort)', fig.align = 'center'}

knitr::include_graphics("figures/DGE_figure.pdf",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

```

```{r PCA-new, echo=FALSE, message=F, warning=FALSE, verbose= FALSE}


diphyes_data <- DESeq2::plotPCA( dge$`K00162-189-HJTYGBBXX-7-NCAGTG`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_diph <- round( 100 * attr(diphyes_data, "percentVar") )

diphyes_pca <- ggplot(diphyes_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"= 15,"Gastrozooid mature"= 0, "Bract mature"= 7, "Gonophore"=3)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_diph[1],"% variance")) +  ylab(paste0("PC2:",percentVar_diph[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

physalia_data<-DESeq2::plotPCA(dge$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_phy <- round( 100 * attr(physalia_data, "percentVar") )

physalia_pca <- ggplot(physalia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8,"Tentacular palpon mature"=10,"Tentacular palpon developing"=16,"Gas gland"=9, "Tentacle developing"=23,"Tentacle mature"=5)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_phy[1],"% variance")) +  ylab(paste0("PC2:",percentVar_phy[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

apolemia_data<-DESeq2::plotPCA(dge$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_apol <- round( 100 * attr(apolemia_data, "percentVar") )

apolemia_pca <- ggplot(apolemia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_apol[1],"% variance")) +  ylab(paste0("PC2:",percentVar_apol[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

nanomia_data<-DESeq2::plotPCA(dge$`HWI-ST625-51-C02UNACXX-7-NANOMIA`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_nano <- round( 100 * attr(nanomia_data, "percentVar") )

nanomia_pca <- ggplot(nanomia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19,"Nectophore developing "=18, "Palpons mature"=2,"Palpons developing"=17,"Gonodendron male"=13,"Gonodendron female"=10)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_nano[1],"% variance")) +  ylab(paste0("PC2:",percentVar_nano[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

frillagalma_data<-DESeq2::plotPCA(dge$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_frill <- round( 100 * attr(frillagalma_data, "percentVar") )

frillagalma_pca <- ggplot(frillagalma_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19,"Palpons mature"=2,"Gonodendron male"=13,"Gonodendron female"=10,"Bract developing"= 7)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_frill[1],"% variance")) +  ylab(paste0("PC2:",percentVar_frill[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

bargmannia_data<-DESeq2::plotPCA(dge$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_barg <- round( 100 * attr(bargmannia_data, "percentVar") )

bargmannia_pca <- ggplot(bargmannia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid white mature"=0,"Gastrozooid yellow mature"=14,"Gastrozooid yellow developing"=12,"Pneumatophore"=8, "Nectophore developing"=19,"Gonodendron male"=13,"Bract developing"= 7)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_barg[1],"% variance")) +  ylab(paste0("PC2:",percentVar_barg[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

agalma_data<-DESeq2::plotPCA(dge$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_agalma <- round( 100 * attr(agalma_data, "percentVar") )

agalma_pca <- ggplot(agalma_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19,"Palpons mature"=2,"Palpons B mature"=6,"Gonodendron male"=13,"Gonodendron female"=10,"Bract developing"= 7)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_agalma[1],"% variance")) +  ylab(paste0("PC2:",percentVar_agalma[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))



```
(ref:technical-replicate) PCA of regularized log transformed expression counts of technical replicates of two different zooids in *Frillagalma vityazi* from different runs and lanes. Color indicates the run number, shape indicates the zooid, and size factor indicates number of genes sequenced.

(ref:technical-replicateshort)  PCA of regularized log transformed expression counts of technical replicates of two different zooids from different runs and lanes.

```{r technical-replicate, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.cap= '(ref:technical-replicate)', fig.scap='(ref:technical-replicateshort)', fig.align='center'}

knitr::include_graphics("Figures/technical-replicate.pdf",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

```

(ref:pca-diphyes) PCA of regularized log transformed expression counts of zooids/tissues in *Diphyes dispar*. Color indicates the replicate number, shape indicates the zooid.

(ref:pca-diphyesshort) PCA of regularized log transformed expression counts of zooids/tissues in *Diphyes dispar*.

```{r pca-diphyes, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-diphyes)', fig.scap='(ref:pca-diphyesshort)',fig.align='center'}

diphyes_pca

```

(ref:pca-apolemia) PCA of regularized log transformed expression counts of zooids/tissues in *Apolemia lanosa*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-apolemiashort) PCA of regularized log transformed expression counts of treatments in *Apolemia lanosa*.

```{r pca-apolemia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-apolemia)', fig.scap='(ref:pca-apolemiashort)',fig.align='center'}

apolemia_pca

```

(ref:pca-agalma) PCA of regularized log transformed expression counts of zooids/tissues in *Agalma elegans*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-agalmashort) PCA of regularized log transformed expression counts of treatments in *Agalma elegans*.

```{r pca-agalma, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-agalma)', fig.scap='(ref:pca-agalmashort)',fig.align='center'}

agalma_pca

```

(ref:pca-bargmannia) PCA of regularized log transformed expression counts of zooids/tissues in *Bargmannia elongata*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-bargmanniashort) PCA of regularized log transformed expression counts of treatments in *Bargmannia elongata*.

```{r pca-bargmannia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-bargmannia)', fig.scap='(ref:pca-bargmanniashort)',fig.align='center'}

bargmannia_pca

```

(ref:pca-Frillagalma) PCA of regularized log transformed expression counts of zooids/tissues in *Frillagalma vityazi*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-Frillagalmashort) PCA of regularized log transformed expression counts of treatments in *Frillagalma vityazi*.

```{r pca-Frillagalma, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-Frillagalma)', fig.scap='(ref:pca-Frillagalmashort)',fig.align='center'}

frillagalma_pca

```

(ref:pca-nanomia) PCA of regularized log transformed expression counts of zooids/tissues in *Nanomia bijuga*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-nanomiashort) PCA of regularized log transformed expression counts of treatments in *Nanomia bijuga*.

```{r pca-nanomia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-nanomia)', fig.scap='(ref:pca-nanomiashort)',fig.align='center'}

nanomia_pca

```

(ref:pca-physalia) PCA of regularized log transformed expression counts of zooids/tissues in *Physalia physalis*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-physaliashort) PCA of regularized log transformed expression counts of treatments in *Physalia physalis*.

```{r pca-physalia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-physalia)', fig.scap='(ref:pca-physaliashort)',fig.align='center'}

physalia_pca

```

(ref:unique-zooids) Number of genes identified as uniquely differentially expressed in zooids within different siphonophore species.  Unique in this case means that the genes are significantly upregulated in these zooids and are not found to be upregulated in any other zooid within the same species. Note that larger numbers of genes are identified in species where fewer zooids were sampled, this is likely due to sampling differences, as opposed to biological differences among species.

```{r unique-zooids, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.cap= '(ref:unique-zooids)', fig.align = 'center'}

unique_zooids %>% dplyr::group_by(zooid,species) %>% dplyr::summarize(number=n()) %>% ggplot()+ geom_point(aes(x=zooid, y=species, size=number),colour="#00BFC4", pch=19) + scale_size_area( max_size = 10 ) + theme(axis.text.x=element_text(angle = 30, hjust = 1))+labs(x="Zooids",y="Species")

```

(ref:transcription-factors) Putative transcription factors found to be significantly upregulated in zooids and the pneumatophore across four different siphonophore species. Transcription factor identity based on blast hit, GO DNA-binding transcription factor activity followed by manual curation. For *Bargmannia elongata* values for mature and developing gastrozooids and values for 'white' gastrozooids. 

```{r transcription-factors, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.height=12, fig.width=12, fig.cap= '(ref:transcription-factors)', fig.align = 'center'}

x %>% ggplot() + geom_point(aes(zooid, short_blast_hit,col=species),position=position_dodge2(width = 0.6)) +labs(y="Transcription factor", x="Zooid")+ theme(axis.text.x=element_text(angle = 30, hjust = 1))

```

(ref:novel-zooids) Unique siphonophore zooids that were sampled for differential gene expression. A. *Bargmannia elongata* with developing "yellow" gastrozooid surrounded by developing "white" gastrozooids. B.  *Bargmannia elongata* stem with mature "white" and "yellow" gastrozooids. C. Zooids in *Physalia physalis*, including multiple developing stages of gastrozooids, tentacular palpon and tentacle; the most mature form of either zooid is not shown. D. Stem of *Agalma elegans*, with gastric palpons, B-palpon and gastrozooid shown.

(ref:novel-zooidsshort) Unique siphonophore zooids that were sampled for differential gene expression.

```{r novel-zooids, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.5\\textheight', out.width='0.5\\textheight', fig.cap= '(ref:novel-zooids)', fig.scap='(ref:novel-zooidsshort)', fig.align = 'center'}

knitr::include_graphics("Figures/Zooid_type.jpg",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

```


(ref:wnt-plot) Top panel: Maximum likelihood gene tree of the Wnt family in Cnidaria, Wnt clades that include siphonophore sequences are highlighted. Bottom panel: Pruned gene tree of wnt genes in siphonophore species with expression values (TPM10K) at the tips and nodes. Developing gastrozooid expression in orange (above node), developing nectophore expression in green (below node). Red circles are duplication nodes and blue circles are speciation nodes.

```{r wnt-plot, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.height=15, fig.width=15, fig.cap='(ref:wnt-plot)', fig.align = 'center'}

#pull gene tree digest

wnt_digest<-edges_tidy[grep("WNT2B_CHICK Protein Wnt-2b",edges_tidy$blast_hit),] %>% .$gene_tree %>% unique()

nhx_wider<-gene_trees_ace_calibrated[gene_trees_ace_digests==wnt_digest][[1]]
p<-nhx_wider %>% ggtree()+geom_nodepoint(aes( color=Ev )) +xlim(c(0,1.5))
p<-p + geom_cladelabel(node=133, label="Wnt2", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=166, label="Wnt4", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=171, label="Wnt9/10", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=191, label="Wnt1", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=204, label="Wnt3", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=248, label="Wnt16", align=T, angle=270, hjust='center', offset=0.025, barsize=1) + theme(legend.position = "none") 

nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==wnt_digest][[1]]
tips1<-pull(nhx@data,"Gasdev")
tips2<-pull(nhx@data,"Necdev")
p1<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names),pch=1, hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,2))
p1<-p1+geom_text(aes(label=round(tips1,2)),vjust=-1,hjust=1, size=3, col="#F4A519")
p1<-p1+geom_text(aes(label=round(tips2,2)),vjust=1.5,hjust=1, size=3, col="#A7C136")
p1<-p1 + geom_cladelabel(node=23, label="Wnt2", align=T, angle=270, hjust='center', offset=0.5, barsize=1)
p1<-p1 + geom_cladelabel(node=31, label="Wnt9/10", align=T, angle=270, hjust='center', offset=0.5, barsize=1)
p1<-p1 + geom_cladelabel(node=33, label="Wnt1", align=T, angle=270, hjust='center', offset=0.5, barsize=1)
p1<-p1 + geom_cladelabel(node=34, label="Wnt3", align=T, angle=270, hjust='center', offset=0.5, barsize=1) + theme(legend.position = "none")

grid.arrange(p,p1)

```

```{r variance-structure, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=6.5, fig.height=6, fig.pos='h', out.extra = '', fig.cap="Variance of change across a branch plotted against the age of the parent node. Branch ID is coded in colour, values are separated by treatment. Gasdev = Developing Gastrozooid, Gasmat= Mature gastrozooid, Gonfem= Female gonodendron, Gonmal= Male gonodendron, Necdev= Developing nectophore, Palmat = Mature palpon, Pne= pneumatophore.", fig.scap="Variance of change across a branch plotted against the age of the parent node.", fig.align='center'}

grid.arrange(var_change_obs,legend,var_change_sim, bottom="Node age of parent", left="Variance of change",widths = c(2,0.25),layout_matrix = layout2)

```

(ref:fullvenom-plot) Maximum likelhood gene tree of the putative actinoporin-like toxin with all available species included. Red circles are duplication nodes and blue circles are speciation nodes.

```{r fullvenom-plot, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.height=15, fig.width=15, fig.cap='(ref:wnt-fullvenom)', fig.align = 'center'}

venom_p

```

\clearpage
### Software versions {-}

This manuscript was computed on `r format( Sys.time(), "%a %b %d %X %Y" )` with the following R package versions.

```{r session_summary, echo=FALSE, include=TRUE, comment=NA}
	sessionInfo()
```

\clearpage
## References {-}
