---
bibliography: references.bib
csl: genomebiology.csl
site: "bookdown::bookdown_site"
output:
  bookdown::pdf_document2:
    toc: FALSE
    fig_caption: yes
---

```{r preliminary-setup, include=FALSE}
	# Load libraries
	#biocLite libraries
  #install.packages("BiocManager")
 #BiocManager::install(version = "3.11")
  #BiocManager::install(c("ggtree","treeio","impute","DESeq2","topGO","goseq","GO.db","vsn"))
	library(ggtree)
  library(treeio)
  library(impute)
  library(DESeq2)
  library(topGO)
  library(goseq)
  library(GO.db)
  library(vsn)
	
	#devtools
  # library( devtools )
	library(hutan)
		# devtools::install_github("caseywdunn/hutan")
	library("agalmar")
    #devtools::install_github("caseywdunn/agalmar")
 
	#CRAN libraries
  library(ggplot2)
	library(bookdown)
  library(knitr)
	library(fields)
	library(picante)
	library(RColorBrewer)
	library(SparseM)
	library(tidyverse)
  library(digest)
  library(parallel)
  library(magrittr)
  library(PhylogeneticEM)
  library(GGally)
  library(geiger)
  library(gridExtra)
  library(emmeans)
  library(FSA)
  library(lme4)
  library(ggpubr)
  library(corrplot)
  library(kableExtra)
  library(ggpubr)
  library(ggrepel)
  library(gtools)


rm(list = ls())

  	## The minimum number of counts to pass gene sampling criteria
	min_count <- 1 
	
	## p value cutoff for evaluating differential expression significance
	p_value_threshold <- 0.05 
	
	#
	focal_species <- c( "Agalma elegans", "Nanomia bijuga", "Bargmannia elongata", "Frillagalma vityazi", "Diphyes dispar", "Physalia physalis", "Apolemia lanosa" )
	
## Gene tree parameters
	
	### The default branch length pace holder used in the gene tree inference software
	default_length_val <- 1e-06   
	
	### Exclude trees with branches that exceed length threshold
	edge_length_max_threshold <- 2

	### Exclude trees that exceed specified fraction of branches with default values
	fraction_default_max_threshold <- 0.25
	
	### The minimum number of tips with expression data for a tree to be considered 
	min_tips <- 3
	
	### The maximum root depth of gene trees, where the root of the species tree is 1
	### Gene trees that exceed this threshold are removed
	max_root_depth <- 5
	
	# Set system computational parameters
	cores <- detectCores() - 1
	if ( cores < 1 ) {
		cores <- 1
	}

	set.seed( 23456 )
	
	cores <- detectCores() - 1

```

```{r load-data, include=FALSE}
#Generated by ancestral_trait_recon.R

  load("manuscript_checkpoint_ace.RData")
  

	source( "functions.R" )
```


```{r blast by gene_tree,verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}
#this gives you gene trees with the top blast hit for that gene tree

genes_to_tips =
		nodes_ace %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,blast_lookup, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( blast_hit =names(which.max(table(blast_hit))) ) %>% # A convoluted way to get the most frequent blast hit for each gene
	  dplyr::mutate( blast_hit = stringr::str_replace(blast_hit, "swissprot|", ""))

```

```{r subtree, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

phyldog_species_numbered_tree_original <-phyldog_species_numbered_tree	
	
	# Need to get the phyldog node numbers for tips from elsewhere
	node_map = 
		nodes_raw %>% 
		dplyr::select( S, species ) %>% 
		na.omit() %>% 
		dplyr::distinct()
	
	# Reorder to match tree tips
	node_map = 
		node_map[ match( phyldog_species_numbered_tree$tip.label, node_map$species ), ]
	
#	tiplabels( node_map$S )
	
#	kable(node_map, row.names=FALSE)
	
		observed_phyldog_nodes = 
		unique( c( edges_ace$S_parent, edges_ace$S_child ) )
	
	phyldog_species_numbered_tree = 
		ape::drop.tip( 
			phyldog_species_numbered_tree, 
			which( ! node_map$S %in% observed_phyldog_nodes )
		)
	
	node_map = node_map[ node_map$S %in% observed_phyldog_nodes, ]
	
	# make sure order matches tree tips
	node_map = 
		node_map[ match( phyldog_species_numbered_tree$tip.label, node_map$species ), ]

	# Create a vector of phyldog node labels in the order of the ape nodes
	S = c( node_map$S, phyldog_species_numbered_tree$node.label ) 
	
		##Have a human readable name for each of the branches instead of the numbers given by S
S_new<-data.frame( S_new=c("F","G","H","I","J","K","L","X","A","B","C","D","E" ),S=S)

	

```

```{r match edges, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE }
  edges_ace_original<- edges_ace

	# Only retain edges where both parent and child is a speciation event
	edges_ace %<>% dplyr::filter( (Ev_parent=="S") & (Ev_child=="S") )
	#dim(edges_ace)
	
	# Create an edge matrix that is in terms of phyldog node numbers rather than ape node numbers
	phyldog_edges = S[ as.vector(phyldog_species_numbered_tree$edge) ]
	dim(phyldog_edges) = dim( phyldog_species_numbered_tree$edge )
	
	# For each row in edges_ace, identify the expected parent in the species tree for S_child
	expected_parent = phyldog_edges[ match(edges_ace$S_child, phyldog_edges[,2]), 1 ]
	
	# Filter to retain only the edges where S_parent matches the expected parent of S_child in the species tree
	
	edges_ace %<>% dplyr::filter( (S_parent==expected_parent) )
	#dim(edges_ace)
	
	# Create a tidy data frame of changes in expression
	edges_tidy = edges_ace %>% dplyr::select( gene_tree, S_child, length, node_depth_child, node_age_child, node_age_parent, node_parent, node_child, terminal )
	edges_tidy = cbind( edges_tidy, edges_ace[, grepl("scaled", names(edges_ace))] )
	colnames(edges_tidy) = colnames(edges_tidy) %>% sub( "_scaled", "", .)
	edges_tidy %<>% tidyr::gather( names(edges_tidy)[10:21], key = "treatment", value = "change" )
	
	# Change child node to factor
	edges_tidy %<>% dplyr::mutate( S_child = as.factor(S_child) )
	
	# Remove tau for now since it is on a different scale
	edges_tidy<-edges_tidy %>% select(-tau)
	
	# Remove rows with missing values
	edges_tidy %<>% na.omit()
	
	# Add blast reports
	edges_tidy %<>%
		dplyr::left_join( genes_to_tips, key="gene_tree")
	
	edges_tidy$change_abs = abs( edges_tidy$change )
	
	#remove zooid values from branches where there shouldn't be any (due to topological issues)
	
	edges_tidy<-edges_tidy %>% group_by(S_child,treatment) %>% filter(n()>300) %>% ungroup()
	
	#add child values
	
	child<-edges_ace %>% select(gene_tree, node_parent, node_child, PalmatGasmat_child, GasdevGasmat_child, NecdevGasmat_child, PneGasmat_child)

  edges_tidy<- edges_tidy %>% left_join(child)
  
  edges_tidy<- edges_tidy %>% left_join(S_new, by=c("S_child"="S"))
	
```

```{r match edges sim, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

edges_sim_original<-edges_sim

genes_to_tips_sim =
		nodes_sim %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,blast_lookup, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( blast_hit =names(which.max(table(blast_hit))) ) %>% # A convoluted way to get the most frequent blast hit for each gene
		dplyr::mutate( blast_hit = stringr::str_replace(blast_hit, "swissprot|", ""))

	# Only retain edges where both parent and child is a speciation event
	edges_sim %<>% dplyr::filter( (Ev_parent=="S") & (Ev_child=="S") )
	#dim(edges_sim)
	
	# Create an edge matrix that is in terms of phyldog node numbers rather than ape node numbers
	phyldog_edges = S[ as.vector(phyldog_species_numbered_tree$edge) ]
	dim(phyldog_edges) = dim( phyldog_species_numbered_tree$edge )
	
	# For each row in edges_sim, identify the expected parent in the species tree for S_child
	expected_parent = phyldog_edges[ match(edges_sim$S_child, phyldog_edges[,2]), 1 ]
	
	# Filter to retain only the edges where S_parent matches the expected parent of S_child in the species tree
	
	edges_sim %<>% dplyr::filter( S_parent==expected_parent) 
	#dim(edges_sim)
	
	# Create a tidy data frame of changes in expression
	edges_sim_tidy = edges_sim %>% dplyr::select( gene_tree, S_child, length, node_depth_child, node_age_child, node_age_parent, node_parent, node_child, terminal )
	edges_sim_tidy = cbind( edges_sim_tidy, edges_sim[, grepl("scaled", names(edges_sim))] )
	colnames(edges_sim_tidy) = colnames(edges_sim_tidy) %>% sub( "_scaled", "", .)
	edges_sim_tidy %<>% tidyr::gather( names(edges_sim_tidy)[10:21], key = "treatment", value = "change" )
	
	# Change child node to factor
	edges_sim_tidy %<>% dplyr::mutate( S_child = as.factor(S_child) )
	
	# Remove tau for now since it is on a different scale
  edges_sim_tidy<-edges_sim_tidy %>% select(-tau)
	
	# Remove rows with missing values
	edges_sim_tidy %<>% na.omit()
	
	# Add blast reports
	edges_sim_tidy %<>%
		dplyr::left_join( genes_to_tips_sim, key="gene_tree")
	
	edges_sim_tidy$change_abs = abs( edges_sim_tidy$change )
	
	edges_sim_tidy<-edges_sim_tidy %>% group_by(S_child,treatment) %>% filter(n()>300) %>% ungroup()
	
	digest_sim<-lapply(gene_trees_sim_calibrated_pruned,digest::digest) %>% unlist()


```

```{r go-annots , verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

parseGO<-function(GOannot){
 GO<-readr::read_tsv(GOannot, col_names=TRUE) 
 GO_bygene<-strsplit(GO$GO_term,";")
 names(GO_bygene)<-GO$sequence_id
 return(GO_bygene)
}

getannots<-function(GobyGo){
  x<-list()
  for(i in 1:length(GobyGo)){
  x[[i]]<-data.frame(go=rep(names(GobyGo)[[i]],length(GobyGo[[i]])),gene=GobyGo[[i]])
  x[[i]]$go<-as.character(x[[i]]$go)
  x[[i]]$gene<-as.character(x[[i]]$gene)
  }
  x<-x %>% dplyr::bind_rows()
  return(x)
}

Agalma_GO<- parseGO(GOannot="GO_terms/Agalma_GO_anno.txt")
Agalma_GO<-Agalma_GO[names(Agalma_GO) %in% rownames(e$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`@x)]
Agalma_annot <- getannots(topGO::inverseList(Agalma_GO))
Nanomia_GO<-parseGO(GOannot="GO_terms/Nanomia_GO_anno.txt")
Nanomia_GO<-Nanomia_GO[names(Nanomia_GO) %in% rownames(e$`HWI-ST625-51-C02UNACXX-7-NANOMIA`@x)]
Nanomia_annot <- getannots(topGO::inverseList(Nanomia_GO))
Bargmannia_GO<-parseGO(GOannot="GO_terms/Bargmannia_GO_anno.txt")
Bargmannia_GO<-Bargmannia_GO[names(Bargmannia_GO) %in% rownames(e$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`@x)]
Bargmannia_annot <- getannots(topGO::inverseList(Bargmannia_GO))
Frillagalma_GO<-parseGO(GOannot="GO_terms/Frillagalma_GO_anno.txt")
Frillagalma_GO<-Frillagalma_GO[names(Frillagalma_GO) %in% rownames(e$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`@x)]
Frillagalma_annot <- getannots(topGO::inverseList(Frillagalma_GO))
Diphyes_GO<-parseGO(GOannot="GO_terms/Diphyes_GO_anno.txt")
Diphyes_GO<-Diphyes_GO[names(Diphyes_GO) %in% rownames(e$`K00162-189-HJTYGBBXX-7-NCAGTG`@x)]
Diphyes_annot <- getannots(topGO::inverseList(Diphyes_GO))
Physalia_GO<-parseGO(GOannot="GO_terms/Physalia_GO_anno.txt")
Physalia_GO<-Physalia_GO[names(Physalia_GO) %in% rownames(e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@x)]
Physalia_annot <- getannots(topGO::inverseList(Physalia_GO))
Apolemia_GO<-parseGO(GOannot="GO_terms/Apolemia_GO_anno.txt")
Apolemia_GO<-Apolemia_GO[names(Apolemia_GO) %in% rownames(e$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`@x)]
Apolemia_annot <- getannots(topGO::inverseList(Apolemia_GO))
GO<- readr::read_tsv("GO_terms/AllGOcat.txt", col_names=TRUE)

reduced_nodes<-nodes_ace[,c("gene_tree","sequence_ids")]

GO_to_tips =
		nodes_ace %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,GO, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( GO =names(which.max(table(GO_term))) ) 

GO_by_tree<-strsplit(GO_to_tips$GO,";")
names(GO_by_tree)<-GO_to_tips$gene_tree

readable_term<-function(golist){
x1<-lapply(golist, Term)
if(length(x1)>5){
  x1<-x1[1:5] %>% paste(collapse=", ")
}
else{
x1<-x1 %>% paste(collapse=", ")
}
return(x1)
}

genes_to_tips =
		nodes_ace %>%
		dplyr::select( gene_tree, sequence_ids ) %>%
    dplyr::left_join(.,blast_lookup, by="sequence_ids") %>%
		na.omit() %>%
		dplyr::select( -sequence_ids ) %>%
		dplyr::arrange(gene_tree) %>%
		dplyr::group_by(gene_tree) %>%
		dplyr::summarise( blast_hit =names(which.max(table(blast_hit))) ) %>% # A convoluted way to get the most frequent blast hit for each gene
		dplyr::mutate( blast_hit = stringr::str_replace(blast_hit, "swissprot|", ""))

	lengthfunction<-function(object,GOlist){
	length<-as.integer(object@lengths)
	names(length)<-rownames(object@x)
	length<-length[names(length) %in% names(GOlist)]
	return(length)
	}
  
	Physalia_length<-lengthfunction(e$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`,Physalia_GO)
	Nanomia_length<-lengthfunction(e$`HWI-ST625-51-C02UNACXX-7-NANOMIA`,Nanomia_GO)
	Diphyes_length<-lengthfunction(e$`K00162-189-HJTYGBBXX-7-NCAGTG`,Diphyes_GO)
	Apolemia_length<-lengthfunction(e$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`,Apolemia_GO)
	Agalma_length<-lengthfunction(e$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`,Agalma_GO)
	Bargmannia_length<-lengthfunction(e$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`,Bargmannia_GO)
	Frillagalma_length<-lengthfunction(e$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`,Frillagalma_GO)
```

```{r linear models, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

edges_tidy_original<-edges_tidy
edges_sim_tidy_original<-edges_sim_tidy

Sum_sim= edges_sim_tidy %>% filter(treatment %in% c("PalmatGasmat","GasdevGasmat","NecdevGasmat","PneGasmat")) %>% dplyr::select(change,gene_tree,S_child,treatment) %>% Summarize(change ~ S_child*treatment,  data=.,digits=3)

Sum = edges_tidy %>% filter(treatment %in% c("PalmatGasmat","GasdevGasmat","NecdevGasmat","PneGasmat")) %>% dplyr::select(change,gene_tree,S_child,treatment) %>% Summarize(change ~ S_child*treatment,  data=.,digits=3)

#add more memorable branch names with letters
Sum = data.frame(S_new = S_new$S_new[match(Sum$S_child, S_new$S)], S_child=Sum$S_child) %>% dplyr::bind_cols(Sum, .)

Sum_sim = data.frame(S_new = S_new$S_new[match(Sum_sim$S_child, S_new$S)], S_child=Sum_sim$S_child) %>% dplyr::bind_cols(Sum_sim, .)

pd = position_dodge(.2)

branch_change<- ggplot(Sum, aes(x = S_new,
                y = mean,
                color = treatment)) +
    geom_errorbar(aes(ymin = mean - sd,
                      ymax = mean + sd),
                   width=.2, size=0.7, position=pd) +
    geom_point(shape=15, size=4, position=pd) +
    theme_bw() +
    theme(axis.title = element_text(face = "plain", size=12)) +
    ylab("Mean change along branch") + xlab("") + ggtitle("Empirical") +theme(plot.title = element_text(hjust = 0.5)) +ylim(c(-8,8)) + scale_color_manual(values=c("#F9B326","#AEC64A","#F7D114","#7ECBE0"))

leg<- get_legend(branch_change) 

branch_change<-branch_change + guides(color=FALSE)

sim_branch_change<- ggplot(Sum_sim, aes(x = S_new,
                y = mean,
                color = treatment)) +
    geom_errorbar(aes(ymin = mean - sd,
                      ymax = mean + sd),
                   width=.2, size=0.7, position=pd) +
    geom_point(shape=15, size=4, position=pd) +
    theme_bw() +
    theme(axis.title = element_text(face = "plain", size=12)) +
    ylab("") + ggtitle("Simulated") + xlab("") + guides(color=FALSE) + theme(plot.title = element_text(hjust = 0.5)) +ylim(c(-8,8)) + scale_color_manual(values=c("#F9B326","#AEC64A","#F7D114","#7ECBE0"))

tree_reference<- ggtree(phyldog_species_numbered_tree, branch.length="none") + geom_tiplab(aes(label=label), fontface='italic')+xlim(-6,15) +geom_label(aes(x=branch,label=S_new$S_new), size=2, label.size = 0.1)

layout<-rbind(c(1,1,NA),c(2,3,4))

change_figure<- grid.arrange(tree_reference,branch_change,sim_branch_change,leg, bottom="Branch",widths = c(2,2,0.5),layout_matrix = layout)


#the vast majority of branches show small or no change in expression along a branch - conversely, a handful of gene trees show large changes along a branch

var_change_obs_data<-edges_tidy %>% filter(treatment %in% c("PalmatGasmat","GasdevGasmat","NecdevGasmat","PneGasmat")) %>% group_by(S_child,treatment) %>% summarise(mean_abs_change=mean(change_abs), var_abs_change=var(change_abs),mean_change=mean(change),var_change=var(change),node_age_parent=node_age_parent[1])

##add human readable branch data
var_change_obs_data<-data.frame(S_new = S_new$S_new[match(var_change_obs_data$S_child, S_new$S)], S_child=var_change_obs_data$S_child) %>% bind_cols(var_change_obs_data, .)

var_change_obs<-var_change_obs_data%>% ggplot()+geom_point(aes(node_age_parent,var_abs_change,col=S_new)) + facet_wrap(~treatment) + xlab(" ") + ylab("")+ggtitle(label="Empirical") + theme(plot.title = element_text(hjust = 0.5))+ guides(col=guide_legend(title="Branch"))

legend<- get_legend(var_change_obs) 

var_change_obs<-var_change_obs + guides(color=FALSE)

var_change_sim_data<- edges_sim_tidy %>% filter(treatment %in% c("PalmatGasmat","GasdevGasmat","NecdevGasmat","PneGasmat")) %>% group_by(S_child,treatment) %>% summarise(mean_abs_change=mean(change_abs), var_abs_change=var(change_abs),mean_change=mean(change),var_change=var(change),node_age_parent=node_age_parent[1])
  
##add human readable branch data
var_change_sim_data<-data.frame(S_new = S_new$S_new[match(var_change_sim_data$S_child, S_new$S)], S_child=var_change_sim_data$S_child) %>% bind_cols(var_change_sim_data, .)
    
var_change_sim<-var_change_sim_data %>% ggplot()+geom_point(aes(node_age_parent,var_abs_change,col=S_new)) + facet_wrap(~treatment) + xlab(" ") + ylab(" ")+ggtitle(label="Simulated") + theme(plot.title = element_text(hjust = 0.5))+ guides(color=FALSE)

layout2<-rbind(c(1,2),c(3,2))



### now look at change data w/o ratios

Sum_sim2= edges_sim_tidy %>% filter(treatment %in% c("Palmat", "Gasmat", "Gasdev","Necdev","Pne")) %>% dplyr::select(change,gene_tree,S_child,treatment) %>% Summarize(change ~ S_child*treatment,  data=.,digits=3)

Sum2 = edges_tidy %>% filter(treatment %in% c("Palmat", "Gasmat", "Gasdev","Necdev","Pne")) %>% dplyr::select(change,gene_tree,S_child,treatment) %>% Summarize(change ~ S_child*treatment,  data=.,digits=3)

#add more memorable branch names with letters
Sum2 = data.frame(S_new = S_new$S_new[match(Sum2$S_child, S_new$S)], S_child=Sum2$S_child) %>% dplyr::bind_cols(Sum2, .)

Sum_sim2 = data.frame(S_new = S_new$S_new[match(Sum_sim2$S_child, S_new$S)], S_child=Sum_sim2$S_child) %>% dplyr::bind_cols(Sum_sim2, .)

pd = position_dodge(.2)

branch_change2<- ggplot(Sum2, aes(x = S_new,
                y = mean,
                color = treatment)) +
    geom_errorbar(aes(ymin = mean - sd,
                      ymax = mean + sd),
                   width=.2, size=0.7, position=pd) +
    geom_point(shape=15, size=4, position=pd) +
    theme_bw() +
    theme(axis.title = element_text(face = "plain", size=12)) +
    ylab("Mean change along branch") + xlab("") + ggtitle("Empirical") +theme(plot.title = element_text(hjust = 0.5)) +ylim(c(-18,18)) + scale_color_manual(values=c("#F9B326","#F9C153","#AEC64A","#F7D114","#7ECBE0"))

leg2<- get_legend(branch_change2) 

branch_change2<-branch_change2 + guides(color=FALSE)

sim_branch_change2<- ggplot(Sum_sim2, aes(x = S_new,
                y = mean,
                color = treatment)) +
    geom_errorbar(aes(ymin = mean - sd,
                      ymax = mean + sd),
                   width=.2, size=0.7, position=pd) +
    geom_point(shape=15, size=4, position=pd) +
    theme_bw() +
    theme(axis.title = element_text(face = "plain", size=12)) +
    ylab("") + ggtitle("Simulated") + xlab("") + guides(color=FALSE) + theme(plot.title = element_text(hjust = 0.5)) +ylim(c(-18,18)) + scale_color_manual(values=c("#F9B326","#F9C153","#AEC64A","#F7D114","#7ECBE0"))

tree_reference<- ggtree(phyldog_species_numbered_tree, branch.length="none") + geom_tiplab(aes(label=label), fontface='italic')+xlim(-6,15) +geom_label(aes(x=branch,label=S_new$S_new), size=2, label.size = 0.1)

layout<-rbind(c(1,1,NA),c(2,3,4))

change_figure2<- grid.arrange(tree_reference,branch_change2,sim_branch_change2,leg2, bottom="Branch",widths = c(2,2,0.5),layout_matrix = layout)

var_change_obs_data_noratio<-edges_tidy %>% filter(treatment %in% c("Palmat","Gasdev","Necdev","Pne","Gasmat","Gonmal","Gonfem")) %>% group_by(S_child,treatment) %>% summarise(mean_abs_change=mean(change_abs), var_abs_change=var(change_abs),mean_change=mean(change),var_change=var(change),node_age_parent=node_age_parent[1])

##add human readable branch data
var_change_obs_data_noratio<-data.frame(S_new = S_new$S_new[match(var_change_obs_data_noratio$S_child, S_new$S)], S_child=var_change_obs_data_noratio$S_child) %>% bind_cols(var_change_obs_data_noratio, .)

var_change_obs_noratio<-var_change_obs_data_noratio %>% ggplot()+geom_point(aes(node_age_parent,var_abs_change,col=S_new)) + facet_wrap(~treatment) + xlab(" ") + ylab("")+ggtitle(label="Empirical") + theme(plot.title = element_text(hjust = 0.5))+ guides(col=guide_legend(title="Branch"))

legend2<- get_legend(var_change_obs_noratio) 

var_change_obs_noratio<-var_change_obs_noratio + guides(color=FALSE)


var_change_sim_data_noratio<- edges_sim_tidy %>% filter(treatment %in% c("Palmat","Gasdev","Necdev","Pne","Gasmat","Gonmal","Gonfem")) %>% group_by(S_child,treatment) %>% summarise(mean_abs_change=mean(change_abs), var_abs_change=var(change_abs),mean_change=mean(change),var_change=var(change),node_age_parent=node_age_parent[1])
  
##add human readable branch data
var_change_sim_data_noratio<-data.frame(S_new = S_new$S_new[match(var_change_sim_data_noratio$S_child, S_new$S)], S_child=var_change_sim_data_noratio$S_child) %>% bind_cols(var_change_sim_data_noratio, .)
    
var_change_sim_noratio<-var_change_sim_data_noratio %>% ggplot()+geom_point(aes(node_age_parent,var_abs_change,col=S_new)) + facet_wrap(~treatment) + xlab(" ") + ylab(" ")+ggtitle(label="Simulated") + theme(plot.title = element_text(hjust = 0.5))+ guides(color=FALSE)

layout3<-rbind(c(1,2),c(3,2))

var_change_plot<- grid.arrange(var_change_obs_noratio,legend2,var_change_sim_noratio, bottom="Node age of parent", left="Variance of change",widths = c(2,0.25),layout_matrix = layout3)

###

edges_tidy_directional <- edges_tidy %>% filter(treatment%in%c("GasdevGasmat", "PneGasmat", "NecdevGasmat", "PalmatGasmat")) %>% mutate(change_direction= case_when(change<2 & change >-2 ~ paste0("Neutral"), change>=2 ~ paste0("Positive"), change<=-2 ~ paste0("Negative"))) %>% select(gene_tree, S_child, S_new, change, treatment, change_direction, blast_hit)

child2<-edges_ace %>% select(gene_tree, node_parent, node_child, ND_parent, ND_child)

edges_tidy_directional_export <- edges_tidy %>% filter(treatment%in%c("GasdevGasmat", "PneGasmat", "NecdevGasmat", "PalmatGasmat")) %>% mutate(change_direction= case_when(change<2 & change >-2 ~ paste0("Neutral"), change>=2 ~ paste0("Positive"), change<=-2 ~ paste0("Negative"))) %>% left_join(child2) %>%  select(gene_tree, S_child, node_parent, node_child, ND_parent, ND_child, S_new, change, treatment, change_direction, blast_hit)

readr::write_csv(edges_tidy_directional_export, "Supplementary_Data/Supplementaryfile6.csv", na = "NA") 

#output trees - to be re run for any/all re-runs of prep files

 # for(tree in 1:length(gene_trees_ace_calibrated_pruned)){
 # if(!is.na(gene_trees_ace_calibrated_pruned[[tree]])){
 # ape::write.tree(gene_trees_ace_calibrated_pruned[[tree]]@phylo,file=paste0("Gene_trees/Gene_trees_pruned/",gene_trees_ace_digests[[tree]],".txt"))
 # }
 # }
 # 
 # for(tree in 1:length(gene_trees_ace_calibrated)){
 # if(!is.na(gene_trees_ace_calibrated[[tree]])){
 # ape::write.tree(gene_trees_ace_calibrated[[tree]]@phylo,file=paste0("Gene_trees/Gene_trees/",gene_trees_ace_digests[[tree]],".txt"))
 # }
 # }

```



# Evolution of gene expression across species and specialized zooids in Siphonophora {-}
Catriona Munro^1,2^,  Stefan Siebert^3^, Mark Howison^4^, Felipe Zapata^5^, Casey W. Dunn^6^
\newline
\newline
^1^ Department of Ecology and Evolutionary Biology, Brown University, Providence, RI 02912, USA \newline
^2^ Current address: Collège de France, PSL Research University, CNRS, Inserm, Center for Interdisciplinary Research in Biology, 75005 Paris, France \newline
^3^ XXXXXX Insert Stefan address XXXXXXXXXXX \newline
^4^ Research Improving People's Lives (RIPL), Providence, RI, USA \newline
^5^ Department of Ecology and Evolutionary Biology, University of California Los Angeles, Los Angeles, CA 90095, USA \newline
^6^ Department of Ecology and Evolutionary Biology, Yale University, New Haven, CT 06520, USA



## Abstract {-}

**Background**

Siphonophores are highly complex colonial organisms, consisting of asexually-produced bodies (called zooids) that share a common gastrovascular cavity but that are functionally specialized to specific tasks, including feeding, swimming, and reproducing. Though this extreme functional specialization has captivated biologists for generations, the genomic underpinnings of this phenomenon remain poorly understood. We use RNA-seq to investigate gene expression patterns in five zooids, and one specialized tissue (the pneumatophore, a gas filled float), across seven siphonophore species. To overcome the analytical challenges of studying the evolution of gene expression across genes and species, we introduce a novel expression metric and two new analytical approaches.

**Results**

Within species, we identified hundreds of zooid-specific genes, as well as a number of putative transcription factors showing higher abundance in particular zooids and developmental stages. In species with unique zooid types, we found significant differences in expression profiles between zooids with different function (i.e., feeding vs. tentacle in Physalia physalis), but no differences between distinct zooids with similar functions (e.g., feeding zooids in Bargmannia elongata and Agalma elegans). Across all species, we found that gene expression patterns tended to be largely consistent among species rather than zooids, and found evidence of large lineage specific shifts in gene expression.

**Conclusions**

Our findings show that there are distinct expression patterns defining each functionally specialized zooid or tissue. In species-specific instances of zooid diversification, we could distinguish novel zooids with distinct morphological and functional differences, but were unable to gain insights into differences between zooids with small differences in morphology or colony placement. Using comparative approaches, we also identified shared zooid-specific gene expression patterns across species, and within a given zooid type, identified instances of clade/species-specific expression patterns. 

## Introduction {-}

Colonial animals, consisting of genetically identical bodies that are physically and physiologically connected, can be found across the metazoan tree of life [@hiebert2020coloniality]. Functional specialization of such bodies has evolved multiple times within colonial animals, with siphonophores in particular showing the greatest diversity of functionally specialized bodies [@Beklemishev1969]. Siphonophores are highly complex, colonial “superorganisms” consisting of asexually produced bodies (termed zooids) that are homologous to solitary free-living polyps and medusae (the typical body forms in Cnidaria), but that share a common gastrovascular cavity [@dunn2006evolution;@mackie1963; @mackie1986aggregates;@mackie1987siphonophore;@totton1965synopsis]. The extreme specialization of siphonophore zooids has been of central interest to zoologists since the 19th century, in part because these zooids are highly interdependent [@Beklemishev1969; @mackie1963]. Siphonophore zooids have been likened to animal organs, with each functionally specialized zooid performing distinct roles within the colony [@mackie1963]. While this analogy works well for understanding the function of zooids within the colony as a whole, the analogy falls short in terms of explaining the evolutionary origin and development of these biological units: functionally specialized zooids are evolutionarily homologous to free living organisms, they are multicellular, possess distinct zooid specific cell types, and show regional subfunctionalization [@church2015histology;@Mackie1960;@totton1965synopsis]. While the developmental mechanisms generating zooids are very different in different clades of colonial animals [@carre1967; @carre1969etude; @carre1991complete; @dunn2006evolution; @siebert2015stem], the evolutionary processes acting on zooids may be similar to those acting on other modular biological units such as cell type, tissue, and organ [@hiebert2020coloniality]. The cellular and molecular processes underlying the patterning and molecular function of functionally specialized cnidarian zooids remain an open biological question. In recent years there has been a focus in particular on differential gene expression patterns found in different functionally specialized zooids [@sanders2014differential; @sanders2015interspecific; @Siebert:2011gv]. 

Efforts to investigate the functional specialization of siphonophores have been limited, in part because there have been few detailed investigations of zooid structure. In the last half century, the microanatomy of siphonophore zooids and tissues has been investigated in only a handful of siphonophore species [@BARDI2007; @carre1969etude;@church2015histology; @Mackie1960]. This leaves many unknowns about how zooid structure and function differ across zooid types and species. Recent *in situ* gene expression analyses in siphonophores have described where a small number of pre-selected genes are expressed at high spatial resolution [@church2015histology; @Siebert:2011gv;@siebert2015stem], but these methods are limited since they require a large number of specimens per gene and siphonophores are relatively difficult to collect. RNA-seq analyses of hand-dissected specimens [@macrander2015rna;@sanders2014differential;@sanders2015interspecific;@Siebert:2011gv], in contrast, can describe the expression of a very large number of genes at low spatial resolution. The fact that so much data is obtained from each specimen is particularly advantageous in difficult-to-collect organisms like siphonophores. An earlier RNA-seq study of gene expression in two zooid types in a single siphonophore species showed the potential of this method to better understand differences between zooids [@Siebert:2011gv]. 

(ref:zooids-definition) Schematic of a siphonophore colony, zooid organization, and zooid function. Here, we illustrate the siphonophore *Nanomia bijuga*, with highlighted zooids and pneumatophore, and explanations of known function. NGZ - nectosomal growth zone. SGZ - siphosomal growth zone. Diagram by Freya Goetz (http://commons.wikimedia.org/wiki/File:Nanomia_bijuga_whole_animal_and_growth_zones.svg)

```{r zooids-definition, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.5\\textheight', out.width='0.5\\textheight', fig.cap='(ref:zooids-definition)', fig.align = 'center'}

knitr::include_graphics("Figures/Siphonophoresare.pdf")

```

In this study we use RNA-seq data investigate the functional specialization and evolution of zooids across siphonophore species. To carry out these analyses, we also address three key challenges to study the evolution of gene expression in a comparative framework [@dunn2013phylogenetic]. First, we introduce a new metric to normalize gene expression which is valid for comparison across species. A common metric used to quantify gene expression is Transcripts Per Million (TPM) [@li2011rsem; @wagner2012measurement]. TPM is a relative measure of expression which depends on the number of genes present in a reference. Hence, TPM values are a valid metric when comparing libraries within a single species, but are not directly comparable across species when the references are incomplete, or genes have been gained and lost in the course of evolution. To address this issue, we introduce Transcripts Per Million 10K (TPM10K), a metric which normalizes TPM to account for different sequencing depths among species (see Methods for details). 

Second, we further account for gene and species effects to compare gene expression across genes and species. Normalized read counts produced by RNA-seq experiments are proportional to gene expression, but they are also impacted by factors that can vary across genes and species such as gene sequence and length. Therefore, to estimate and compare expected gene counts it is necessary to model such factors with unknown species and gene-specific counting-efficiency coefficients [@dunn2013phylogenetic]. A direct comparison of expected gene counts among species, however, can be misleading if differences in counts are simply due to differences in counting efficiency and not due to differences in expression. To address this issue, we use ratios of expected counts to compare expression across genes and species. Using ratios of counts, we are able to eliminate species- and gene- specific counting efficiency within species, as the unknown counting efficiency factor is in both the numerator and the denominator, and is thus removed prior to comparisons among species [@dunn2013phylogenetic]. 

Third, we use a novel approach to investigate the evolution of gene expression on gene trees with complex evolutionary histories. Most evolutionary studies of gene expression focus exclusively on comparing expression values of strict orthologs (i.e., gene lineages related only by speciation events) which are shared across all species. We call this type of classical analysis Strict Ortholog Filtering of Trees (SOFT). This limits analyses to the subset of genes that have no evidence of duplication. This is an all or nothing approach, where gene trees are retained if they have one sequence for each species, and discarded if there are multiple sequences in any of the species. Because this is usually such a small fraction of genes, traditional SOFT analyses discard a large fraction of the data, including many gene families that may be of interest to the investigator. 

Because the history of genes is characterized by more complex scenarios involving gene duplication and speciation, we introduce a new method that takes advantage of this rich history to examine the evolution of gene expression across more genes and species. We call this method Species Tree Edge Extraction and Layover (STEEL) (Fig. \@ref(fig:methods-comp)), as we map expression data to gene phylogenies and identify equivalent branches in the species tree which are descended from speciation events in order to make valid comparisons across species. While SOFT is a gene tree filtering approach, where entire trees are discarded due to even a single duplication event, STEEL is a branch filtering approach. This means that STEEL retains many informative branches from trees that would be removed by SOFT, and preserves many more evolutionary comparisons for analyses.

Briefly, the principles of STEEL are as follows: first we identify speciation and duplication events in a given gene tree, specifically labelling nodes that correspond to speciation events in the species tree (Fig. \@ref(fig:methods-comp), step 1). Next, we map gene expression values to the tips of the gene tree (Fig. \@ref(fig:methods-comp), step 2), and use phylogenetic methods to reconstruct expression values at the internal nodes (Fig. \@ref(fig:methods-comp), step 3). Expression values are mapped and reconstructed for each zooid/tissue separately, with the assumption that the structure is homologous across species. Then, we calculate scaled expression values across branches, by subtracting the expression value at the child node from the parent node, and dividing by branch length (branch lengths are calibrated against branches in the species tree) (Fig. \@ref(fig:methods-comp), step 4). Finally, we identify branches in the gene trees that correspond to equivalent branches in the species tree (hereafter species-equivalent branches) (Fig. \@ref(fig:methods-comp), step 5). Species-equivalent branches are gene tree branches with parent and child nodes that are both speciation events and that correspond to branches in the species tree. Each branch in the species tree is given a unique identifier (i.e., a letter) and the species-equivalent branches in gene trees are given the same identifier (Fig. \@ref(fig:methods-comp), step 5). This method enables the selection of branches within gene trees that are equivalent to branches within the species tree, and are thus comparable with one another across all gene trees. Unlike the SOFT approach, this approach considers equivalent branches that are descended from speciation events, but that have more complex evolutionary histories. For example, due to deeper gene duplication events, gene trees often contain multiple branches that correspond to the same branch in the species tree (Fig. \@ref(fig:methods-comp), step 5). Our method allows us to consider all of these branches. 

Using the methods described above, combined with classical differential gene expression analysis, we conducted differential expression analyses among siphonophore zooids, and one specialized tissue (the pneumatophore, a gas filled float), in seven siphonophore species to address three questions. First, we analyzed differences in gene expression within species to identify patterns that define particular zooid types, and gain insight into the genes that may be playing a role in the structure, maintenance, and functioning of zooids. Second, we examined whether gene expression patterns could distinguish novel, distinct species-specific zooid types within particular species. Third we compared gene expression across the seven siphonophore species, to identify putative clade/lineage or zooid specific expression patterns. 

(ref:methods-comp) New phylogenetic approach to identify evolutionary changes in gene expression, called Species Tree Edge Extraction and Layover (STEEL). Step 1, we label each of the nodes in the species tree, and identify equivalent speciation nodes across every gene tree (an exemplar is shown here). Step 2, we map expression values (TPM10K) to the tips (expression values are mapped and reconstructed for each homologous zooid separately). Step 3, we reconstruct ancestral trait expression values at all internal nodes where expression data are available. Step 4, we calculate scaled change in gene expression (child node expression - parent node expression / branch length). Branch length is calibrated to the species tree branch lengths. Step 5, we identify branches in gene trees that correspond to equivalent branches in the species tree. There may be more than one branch in a gene tree that corresponds to the same branch in the species tree.

```{r methods-comp, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.7\\textheight', out.width='0.7\\textheight', fig.cap='(ref:methods-comp)', fig.align = 'center'}

knitr::include_graphics("figures/Methods_figure.pdf")

```

## Results {-}

### Within species analyses: which genes are specific to particular zooid types? {-}

```{r differentially-expressed, verbose= FALSE, include=F, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

top_res<- function(pair,object){
  
treat1=pair[1]
treat2=pair[2]
  
if( (length(unique(object@ddsMF@colData$treatment)[unique(object@ddsMF@colData$treatment) %in% c(treat1,treat2) ==TRUE])==2)&((length(object@ddsMF@colData$treatment[object@ddsMF@colData$treatment==treat1])>=2)&(length(object@ddsMF@colData$treatment[object@ddsMF@colData$treatment==treat2])>=2))){

res<-DESeq2::results(object@ddsMF,contrast=c( "treatment", treat1, treat2 ), pAdjustMethod="bonferroni", alpha=0.05 )

resOrdered <- res[order(res$padj),] 

resSig <- subset(resOrdered, padj < 0.05) %>% as.data.frame(.)

resSig <- subset(resSig,log2FoldChange > 0 )

resSig<-tibble::rownames_to_column(resSig, var="sequence_ids") 
resSig$sequence_ids<-as.numeric(resSig$sequence_ids)

resSig<-dplyr::left_join(resSig,reduced_nodes, by="sequence_ids")

resSig<-dplyr::left_join(resSig,blast_lookup, by="sequence_ids")

resSig<-dplyr::left_join(resSig,GO,by="sequence_ids")

return(resSig)
}
}


significant_de_genes<-function(object){
all_pairs_dge<- object@ddsMF$treatment %>% unique() %>% as.character() %>% gtools::permutations(n=length(.), r=2,v=., repeats.allowed = TRUE) %>% as.data.frame(stringsAsFactors=FALSE) 

all_pairs_dge<- all_pairs_dge %>% split(., seq(nrow(.))) %>% unname() %>% lapply(.,as.character)

x<-lapply(all_pairs_dge, top_res, object)

short_pairs_function<- function(pairs){
      pair_short<-pairs%>%
      stringr::str_split(.," ")%>%
      unlist() %>%
      stringr::str_sub(.,1,3) %>%
      stringr::str_c(.,collapse="")
      return(pair_short)
}

names(x)<-lapply(all_pairs_dge,short_pairs_function) %>% unlist()

return(x)
}

dge_sig<-lapply(dge, significant_de_genes)

#Agalma

##This is clunky, but due to the different sampling between species I have to do this manually

Agalma_necdev<-c( dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevPalmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$NecdevGonfem$sequence_ids) %>% unique()

Agalma_Palpon<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalmatPne$sequence_ids) %>% unique()

Agalma_PalponB<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PalBmatPne$sequence_ids) %>% unique()

Agalma_gasmat<-c( dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GasmatPalmat$sequence_ids) %>% unique()

Agalma_Gonfem<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonfemPalmat$sequence_ids) %>% unique()

Agalma_Gonmal<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalPne$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$GonmalPalmat$sequence_ids) %>% unique()

Agalma_Pne<-c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneGonfem$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneNecdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PneGonmal$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`$PnePalmat$sequence_ids) %>% unique()

Agalma_Pne_unique<-Agalma_Pne[!Agalma_Pne %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_gasmat,Agalma_Gonfem,Agalma_necdev, Agalma_Gonmal)]

Agalma_necdev_unique<-Agalma_necdev[!Agalma_necdev %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_Gonfem,Agalma_Gonmal,Agalma_gasmat, Agalma_Pne)]

Agalma_Gonmal_unique<-Agalma_Gonmal[!Agalma_Gonmal %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_gasmat,Agalma_Gonfem,Agalma_necdev, Agalma_Pne)]

Agalma_Gonfem_unique<-Agalma_Gonfem[!Agalma_Gonfem %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_gasmat,Agalma_Gonmal,Agalma_necdev, Agalma_Pne)]

Agalma_gasmat_unique<-Agalma_gasmat[!Agalma_gasmat %in% c(Agalma_Palpon,Agalma_PalponB,Agalma_Gonfem,Agalma_Gonmal,Agalma_necdev, Agalma_Pne)]

Agalma_PalponB_unique<-Agalma_PalponB[!Agalma_PalponB %in% c(Agalma_necdev,Agalma_Gonfem,Agalma_Gonmal,Agalma_gasmat, Agalma_Pne)]

Agalma_Palpon_unique<-Agalma_Palpon[!Agalma_Palpon %in% c(Agalma_necdev,Agalma_Gonfem,Agalma_Gonmal,Agalma_gasmat, Agalma_Pne)]

## Bargmannia

Bargmannia_necdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGasyelmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevGonmal$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$NecdevPne$sequence_ids) %>% unique

Bargmannia_Gaswhimat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGasyelmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatGonmal$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GaswhimatPne$sequence_ids) %>% unique

Bargmannia_Gasyelmat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatGonmal$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyelmatPne$sequence_ids) %>% unique

Bargmannia_Gonmal<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GonmalPne$sequence_ids) %>% unique

Bargmannia_Pne<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$PneGonmal$sequence_ids) %>% unique

Bargmannia_Gasyeldev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevPne$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasyeldevGonmal$sequence_ids) %>% unique

Bargmannia_Gasdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGasyeldev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGaswhimat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevPne$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGasyelmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`$GasdevGonmal$sequence_ids) %>% unique


Bargmannia_Gasdev_unique<-Bargmannia_Gasdev[!Bargmannia_Gasdev %in% c(Bargmannia_Gasyeldev,Bargmannia_Pne,Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_Gaswhimat,Bargmannia_necdev)]

Bargmannia_Gasyeldev_unique<-Bargmannia_Gasyeldev[!Bargmannia_Gasyeldev %in% c(Bargmannia_Gasdev,Bargmannia_Pne,Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_Gaswhimat,Bargmannia_necdev)]

Bargmannia_necdev_unique<-Bargmannia_necdev[!Bargmannia_necdev %in% c(Bargmannia_Pne,Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_Gaswhimat)]

#exclude developing gastrozooids due to significant overlap (note I will include it for the developing zooid because I want to know genes involved specifically in development of this zooid)

Bargmannia_Gaswhimat_unique<-Bargmannia_Gaswhimat[!Bargmannia_Gaswhimat %in% c(Bargmannia_Pne,Bargmannia_Gonmal, Bargmannia_necdev,Bargmannia_Gasyelmat)]

Bargmannia_Gasyelmat_unique<-Bargmannia_Gasyelmat[!Bargmannia_Gasyelmat %in% c(Bargmannia_Pne,Bargmannia_Gonmal, Bargmannia_necdev,Bargmannia_Gaswhimat)]

Bargmannia_Gasmat_unique<-Bargmannia_Gaswhimat[!Bargmannia_Gaswhimat %in% c(Bargmannia_Pne,Bargmannia_Gonmal, Bargmannia_necdev)]

Bargmannia_gonmal_unique<-Bargmannia_Gonmal[!Bargmannia_Gonmal %in% c(Bargmannia_Pne,Bargmannia_Gasyelmat, Bargmannia_necdev,Bargmannia_Gaswhimat)]

Bargmannia_Pne_unique<-Bargmannia_Pne[!Bargmannia_Pne %in% c(Bargmannia_Gonmal,Bargmannia_Gasyelmat, Bargmannia_necdev,Bargmannia_Gaswhimat)]

##Nanomia

Nanomia_Necdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevGasmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevPalmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$NecdevPne$sequence_ids) %>% unique

Nanomia_Gasmat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatPalmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasmatPne$sequence_id) %>% unique

Nanomia_Gasdev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevGasmat$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevPalmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GasdevPne$sequence_id) %>% unique

Nanomia_Palmat<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PalmatPne$sequence_id) %>% unique

Nanomia_Paldev<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevGonfem$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PaldevPne$sequence_id) %>% unique

Nanomia_Gonfem<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemPaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$GonfemPne$sequence_id) %>% unique

Nanomia_Pne<-c( dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneNecdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneGasdev$sequence_ids, dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PnePaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PnePaldev$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneGasmat$sequence_ids,
dge_sig$`HWI-ST625-51-C02UNACXX-7-NANOMIA`$PneGonfem$sequence_id) %>% unique

Nanomia_Necdev_unique<-Nanomia_Necdev[!Nanomia_Necdev %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Paldev,Nanomia_Palmat,Nanomia_Gasdev,Nanomia_Gasmat)]

#exclude developing gastrozooids due to significant overlap (note I will include it for the developing zooid because I want to know genes involved specifically in development of this zooid)
Nanomia_Gasmat_unique<-Nanomia_Gasmat[!Nanomia_Gasmat %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Paldev,Nanomia_Palmat,Nanomia_Necdev)]

#exclude developing palpons for the same reason (note I will include it for the developing zooid because I want to know genes involved specifically in development of this zooid)
Nanomia_Palmat_unique<-Nanomia_Palmat[!Nanomia_Palmat %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev)]

Nanomia_Gonfem_unique<-Nanomia_Gonfem[!Nanomia_Gonfem %in% c(Nanomia_Pne,Nanomia_Palmat,Nanomia_Paldev, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev)]

Nanomia_Pne_unique<-Nanomia_Pne[!Nanomia_Pne %in% c(Nanomia_Gonfem,Nanomia_Palmat,Nanomia_Paldev, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev)]

Nanomia_Gasdev_unique<-Nanomia_Gasdev[!Nanomia_Gasdev %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Paldev,Nanomia_Palmat,Nanomia_Necdev, Nanomia_Gasmat)]

Nanomia_Paldev_unique<-Nanomia_Paldev[!Nanomia_Paldev %in% c(Nanomia_Pne,Nanomia_Gonfem, Nanomia_Gasmat,Nanomia_Gasdev,Nanomia_Necdev, Nanomia_Palmat)]

### 


Frillagalma_Gasdev<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevBradev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasdevPne$sequence_ids) %>% unique()

Frillagalma_Gasmat<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatBradev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GasmatPne$sequence_ids) %>% unique()

Frillagalma_Bradev<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$BradevPne$sequence_ids) %>% unique()

Frillagalma_Gonmal<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonmalPne$sequence_ids) %>% unique()

Frillagalma_Gonfem<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$GonfemPne$sequence_ids) %>% unique()

Frillagalma_Necdev<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevPalmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$NecdevPne$sequence_ids) %>% unique()

Frillagalma_Palmat<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PalmatPne$sequence_ids) %>% unique()

Frillagalma_Pne<-c(dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGasmat$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGasdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGonmal$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneGonfem$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PneNecdev$sequence_ids,
                      dge_sig$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`$PnePalmat$sequence_ids) %>% unique()

Frillagalma_Pne_unique<- Frillagalma_Pne[!Frillagalma_Pne%in% c(Frillagalma_Bradev, Frillagalma_Gasdev,Frillagalma_Gasmat, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Bradev_unique<- Frillagalma_Bradev[!Frillagalma_Bradev%in% c(Frillagalma_Pne, Frillagalma_Gasdev,Frillagalma_Gasmat, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gasdev_unique<- Frillagalma_Gasdev[!Frillagalma_Gasdev%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gasmat_unique<- Frillagalma_Gasmat[!Frillagalma_Gasmat%in% c(Frillagalma_Pne, Frillagalma_Bradev, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gonfem_unique<- Frillagalma_Gonfem[!Frillagalma_Gonfem%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonmal, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Gonmal_unique<- Frillagalma_Gonmal[!Frillagalma_Gonmal%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonfem, Frillagalma_Necdev, Frillagalma_Palmat)]

Frillagalma_Necdev_unique<- Frillagalma_Necdev[!Frillagalma_Necdev%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Palmat)]

Frillagalma_Palmat_unique<- Frillagalma_Palmat[!Frillagalma_Palmat%in% c(Frillagalma_Pne, Frillagalma_Bradev,Frillagalma_Gasmat, Frillagalma_Gasdev, Frillagalma_Gonfem, Frillagalma_Gonmal, Frillagalma_Necdev)]

### Apolemia

Apolemia_Pne <- dge_sig$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`$GasdevPne$sequence_ids 
Apolemia_Gasdev <- dge_sig$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`$PneGasdev$sequence_ids

Apolemia_Pne_unique<-Apolemia_Pne[!Apolemia_Pne %in% Apolemia_Gasdev]
Apolemia_Gasdev_unique<-Apolemia_Gasdev[!Apolemia_Gasdev %in% Apolemia_Pne]

### Physalia

Physalia_Gasdev <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevTendev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasdevTenpalmat$sequence_ids) %>% unique()

Physalia_Gasmat <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatTendev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$GasmatTenpalmat$sequence_ids) %>% unique()

Physalia_Tendev <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TendevTenpalmat$sequence_ids) %>% unique()

Physalia_Tenpalmat <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatTenpaldev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpalmatTendev$sequence_ids) %>% unique()

Physalia_Tenpaldev <- c(dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevGasmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevGasdev$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevTenpalmat$sequence_ids, dge_sig$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`$TenpaldevTendev$sequence_ids) %>% unique()

Physalia_Gasdev_unique <- Physalia_Gasdev[!Physalia_Gasdev %in% c(Physalia_Gasmat,Physalia_Tenpalmat, Physalia_Tenpaldev, Physalia_Tendev)]

Physalia_Gasmat_unique <- Physalia_Gasmat[!Physalia_Gasmat %in% c(Physalia_Tenpalmat, Physalia_Tenpaldev, Physalia_Tendev)]

Physalia_Tenpalmat_unique <- Physalia_Tenpalmat[!Physalia_Tenpalmat %in% c(Physalia_Gasmat,Physalia_Gasdev, Physalia_Tendev)]

Physalia_Tenpaldev_unique <- Physalia_Tenpaldev[!Physalia_Tenpaldev %in% c(Physalia_Gasmat,Physalia_Gasdev, Physalia_Tendev, Physalia_Tenpalmat)]

Physalia_Tendev_unique <- Physalia_Tendev[!Physalia_Tendev %in% c(Physalia_Gasmat,Physalia_Gasdev, Physalia_Tenpaldev, Physalia_Tenpalmat)]

### Diphyes

Diphyes_Gasdev <- c(dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasdevGasmat$sequence_ids, dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasdevGon$sequence_ids) %>% unique()

Diphyes_Gasmat <- c(dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasmatGasdev$sequence_ids, dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GasmatGon$sequence_ids) %>% unique()

Diphyes_Gon <- c(dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GonGasdev$sequence_ids, dge_sig$`K00162-189-HJTYGBBXX-7-NCAGTG`$GonGasmat$sequence_ids) %>% unique()

Diphyes_Gasdev_unique<-Diphyes_Gasdev[!Diphyes_Gasdev %in% c(Diphyes_Gasmat, Diphyes_Gon)]

Diphyes_Gasmat_unique<-Diphyes_Gasmat[!Diphyes_Gasmat %in% Diphyes_Gon]

Diphyes_Gon_unique<-Diphyes_Gon[!Diphyes_Gon %in% c(Diphyes_Gasmat, Diphyes_Gasdev)]


## Now put it all into one tidy dataframe

unique_zooids<-tibble(
  sequence_ids=c(Agalma_Pne_unique,Agalma_necdev_unique, Agalma_Gonmal_unique,Agalma_Gonfem_unique,Agalma_gasmat_unique, Agalma_PalponB_unique, Agalma_Palpon_unique,Bargmannia_Gasdev_unique,Bargmannia_Gasyeldev_unique,Bargmannia_necdev_unique,Bargmannia_Gaswhimat_unique,Bargmannia_Gasyelmat_unique,Bargmannia_gonmal_unique,Bargmannia_Pne_unique,Nanomia_Necdev_unique,Nanomia_Gasmat_unique,Nanomia_Palmat_unique,Nanomia_Gonfem_unique,Nanomia_Pne_unique,Nanomia_Gasdev_unique,Nanomia_Paldev_unique,Frillagalma_Pne_unique,Frillagalma_Bradev_unique,Frillagalma_Gasdev_unique,Frillagalma_Gasmat_unique,Frillagalma_Gonfem_unique,Frillagalma_Gonmal_unique,Frillagalma_Necdev_unique,Frillagalma_Palmat_unique,Apolemia_Pne_unique,Apolemia_Gasdev_unique, Physalia_Gasdev_unique,Physalia_Gasmat_unique,Physalia_Tenpalmat_unique,Physalia_Tenpaldev_unique,Physalia_Tendev_unique,Diphyes_Gasdev_unique,Diphyes_Gasmat_unique,Diphyes_Gon_unique),

species=c(rep("Agalma elegans",length(c(Agalma_Pne_unique,Agalma_necdev_unique, Agalma_Gonmal_unique,Agalma_Gonfem_unique,Agalma_gasmat_unique, Agalma_PalponB_unique, Agalma_Palpon_unique))),
          rep("Bargmannia elongata",length(c(Bargmannia_Gasdev_unique,Bargmannia_Gasyeldev_unique,Bargmannia_necdev_unique,Bargmannia_Gaswhimat_unique,Bargmannia_Gasyelmat_unique,Bargmannia_gonmal_unique,Bargmannia_Pne_unique))),
        rep("Nanomia bijuga",length(c(Nanomia_Necdev_unique,Nanomia_Gasmat_unique,Nanomia_Palmat_unique,Nanomia_Gonfem_unique,Nanomia_Pne_unique,Nanomia_Gasdev_unique,Nanomia_Paldev_unique))),
        rep("Frillagalma vityazi",length(c(Frillagalma_Pne_unique,Frillagalma_Bradev_unique,Frillagalma_Gasdev_unique,Frillagalma_Gasmat_unique,Frillagalma_Gonfem_unique,Frillagalma_Gonmal_unique,Frillagalma_Necdev_unique,Frillagalma_Palmat_unique))),
        rep("Apolemia lanosa",length(c(Apolemia_Pne_unique,Apolemia_Gasdev_unique))),
        rep("Physalia physalis",length(c(Physalia_Gasdev_unique,Physalia_Gasmat_unique,Physalia_Tenpalmat_unique,Physalia_Tenpaldev_unique,Physalia_Tendev_unique))),
        rep("Diphyes dispar",length(c(Diphyes_Gasdev_unique,Diphyes_Gasmat_unique,Diphyes_Gon_unique)))
        ),

zooid=c(rep("Pneumatophore",length(Agalma_Pne_unique)),
        rep("Nectophore developing",length(Agalma_necdev_unique)),
        rep("Gonodendron male",length(Agalma_Gonmal_unique)), 
        rep("Gonodendron female",length(Agalma_Gonfem_unique)),
        rep("Gastrozooid mature",length(Agalma_gasmat_unique)),
        rep("Palpon B mature",length(Agalma_PalponB_unique)),
        rep("Palpon mature",length(Agalma_Palpon_unique)), 
        rep("Gastrozooid white developing",length(Bargmannia_Gasdev_unique)),
        rep("Gastrozooid yellow developing",length(Bargmannia_Gasyeldev_unique)), 
        rep("Nectophore developing",length(Bargmannia_necdev_unique)),
        rep("Gastrozooid white mature",length(Bargmannia_Gaswhimat_unique)),
        rep("Gastrozooid yellow mature",length(Bargmannia_Gasyelmat_unique)),
        rep("Gonodendron male",length(Bargmannia_gonmal_unique)),
        rep("Pneumatophore",length(Bargmannia_Pne_unique)),
        rep("Nectophore developing", length(Nanomia_Necdev_unique)),
        rep("Gastrozooid mature",length(Nanomia_Gasmat_unique)),
        rep("Palpon mature",length(Nanomia_Palmat_unique)),
        rep("Gonodendron female",length(Nanomia_Gonfem_unique)),
        rep("Pneumatophore",length(Nanomia_Pne_unique)),
        rep("Gastrozooid developing",length(Nanomia_Gasdev_unique)),
        rep("Palpon developing",length(Nanomia_Paldev_unique)),
        rep("Pneumatophore",length(Frillagalma_Pne_unique)),
        rep("Bract developing",length(Frillagalma_Bradev_unique)),
        rep("Gastrozooid developing",length(Frillagalma_Gasdev_unique)),
        rep("Gastrozooid mature",length(Frillagalma_Gasmat_unique)),
        rep("Gonodendron female",length(Frillagalma_Gonfem_unique)),
        rep("Gonodendron male",length(Frillagalma_Gonmal_unique)),
        rep("Nectophore developing",length(Frillagalma_Necdev_unique)),
        rep("Palpon mature",length(Frillagalma_Palmat_unique)),
        rep("Pneumatophore",length(Apolemia_Pne_unique)),
        rep("Gastrozooid developing",length(Apolemia_Gasdev_unique)),
        rep("Gastrozooid developing",length(Physalia_Gasdev_unique)),
        rep("Gastrozooid mature",length(Physalia_Gasmat_unique)),
        rep("Tentacular palpon mature",length(Physalia_Tenpalmat_unique)),
        rep("Tentacular palpon developing",length(Physalia_Tenpaldev_unique)),
        rep("Tentacle developing",length(Physalia_Tendev_unique)),
        rep("Gastrozooid developing",length(Diphyes_Gasdev_unique)),
        rep("Gastrozooid developing",length(Diphyes_Gasmat_unique)),
        rep("Gonozooid",length(Diphyes_Gon_unique))
        )
  )

DE_zooids<-tibble(
  sequence_ids=c(Agalma_Pne,Agalma_necdev, Agalma_Gonmal,Agalma_Gonfem,Agalma_gasmat, Agalma_PalponB, Agalma_Palpon,Bargmannia_Gasdev,Bargmannia_Gasyeldev,Bargmannia_necdev,Bargmannia_Gaswhimat,Bargmannia_Gasyelmat,Bargmannia_Gonmal,Bargmannia_Pne,Nanomia_Necdev,Nanomia_Gasmat,Nanomia_Palmat,Nanomia_Gonfem,Nanomia_Pne,Nanomia_Gasdev,Nanomia_Paldev,Frillagalma_Pne,Frillagalma_Bradev,Frillagalma_Gasdev,Frillagalma_Gasmat,Frillagalma_Gonfem,Frillagalma_Gonmal,Frillagalma_Necdev,Frillagalma_Palmat,Apolemia_Pne,Apolemia_Gasdev, Physalia_Gasdev,Physalia_Gasmat,Physalia_Tenpalmat,Physalia_Tenpaldev,Physalia_Tendev,Diphyes_Gasdev,Diphyes_Gasmat,Diphyes_Gon),

species=c(rep("Agalma elegans",length(c(Agalma_Pne,Agalma_necdev, Agalma_Gonmal,Agalma_Gonfem,Agalma_gasmat, Agalma_PalponB, Agalma_Palpon))),
          rep("Bargmannia elongata",length(c(Bargmannia_Gasdev,Bargmannia_Gasyeldev,Bargmannia_necdev,Bargmannia_Gaswhimat,Bargmannia_Gasyelmat,Bargmannia_Gonmal,Bargmannia_Pne))),
        rep("Nanomia bijuga",length(c(Nanomia_Necdev,Nanomia_Gasmat,Nanomia_Palmat,Nanomia_Gonfem,Nanomia_Pne,Nanomia_Gasdev,Nanomia_Paldev))),
        rep("Frillagalma vityazi",length(c(Frillagalma_Pne,Frillagalma_Bradev,Frillagalma_Gasdev,Frillagalma_Gasmat,Frillagalma_Gonfem,Frillagalma_Gonmal,Frillagalma_Necdev,Frillagalma_Palmat))),
        rep("Apolemia lanosa",length(c(Apolemia_Pne,Apolemia_Gasdev))),
        rep("Physalia physalis",length(c(Physalia_Gasdev,Physalia_Gasmat,Physalia_Tenpalmat,Physalia_Tenpaldev,Physalia_Tendev))),
        rep("Diphyes dispar",length(c(Diphyes_Gasdev,Diphyes_Gasmat,Diphyes_Gon)))
        ),

zooid=c(rep("Pneumatophore",length(Agalma_Pne)),
        rep("Nectophore developing",length(Agalma_necdev)),
        rep("Gonodendron male",length(Agalma_Gonmal)), 
        rep("Gonodendron female",length(Agalma_Gonfem)),
        rep("Gastrozooid mature",length(Agalma_gasmat)),
        rep("Palpon B mature",length(Agalma_PalponB)),
        rep("Palpon mature",length(Agalma_Palpon)), 
        rep("Gastrozooid white developing",length(Bargmannia_Gasdev)),
        rep("Gastrozooid yellow developing",length(Bargmannia_Gasyeldev)), 
        rep("Nectophore developing",length(Bargmannia_necdev)),
        rep("Gastrozooid white mature",length(Bargmannia_Gaswhimat)),
        rep("Gastrozooid yellow mature",length(Bargmannia_Gasyelmat)),
        rep("Gonodendron male",length(Bargmannia_Gonmal)),
        rep("Pneumatophore",length(Bargmannia_Pne)),
        rep("Nectophore developing", length(Nanomia_Necdev)),
        rep("Gastrozooid mature",length(Nanomia_Gasmat)),
        rep("Palpon mature",length(Nanomia_Palmat)),
        rep("Gonodendron female",length(Nanomia_Gonfem)),
        rep("Pneumatophore",length(Nanomia_Pne)),
        rep("Gastrozooid developing",length(Nanomia_Gasdev)),
        rep("Palpon developing",length(Nanomia_Paldev)),
        rep("Pneumatophore",length(Frillagalma_Pne)),
        rep("Bract developing",length(Frillagalma_Bradev)),
        rep("Gastrozooid developing",length(Frillagalma_Gasdev)),
        rep("Gastrozooid mature",length(Frillagalma_Gasmat)),
        rep("Gonodendron female",length(Frillagalma_Gonfem)),
        rep("Gonodendron male",length(Frillagalma_Gonmal)),
        rep("Nectophore developing",length(Frillagalma_Necdev)),
        rep("Palpon mature",length(Frillagalma_Palmat)),
        rep("Pneumatophore",length(Apolemia_Pne)),
        rep("Gastrozooid developing",length(Apolemia_Gasdev)),
        rep("Gastrozooid developing",length(Physalia_Gasdev)),
        rep("Gastrozooid mature",length(Physalia_Gasmat)),
        rep("Tentacular palpon mature",length(Physalia_Tenpalmat)),
        rep("Tentacular palpon developing",length(Physalia_Tenpaldev)),
        rep("Tentacle developing",length(Physalia_Tendev)),
        rep("Gastrozooid developing",length(Diphyes_Gasdev)),
        rep("Gastrozooid developing",length(Diphyes_Gasmat)),
        rep("Gonozooid",length(Diphyes_Gon))
        )
  )

DE_zooids<-DE_zooids %>% dplyr::left_join(blast_lookup,by="sequence_ids") %>% dplyr::left_join(GO,by="sequence_ids") %>% dplyr::left_join(reduced_nodes, by="sequence_ids")

unique_zooids<-unique_zooids %>% dplyr::left_join(blast_lookup,by="sequence_ids") %>% dplyr::left_join(GO,by="sequence_ids") %>% dplyr::left_join(reduced_nodes, by="sequence_ids")

readr::write_csv(DE_zooids, "Supplementary_Data/Supplementaryfile1.csv", na = "NA")

readr::write_csv(unique_zooids, "Supplementary_Data/Supplementaryfile2.csv", na = "NA")

#### GO analyses -- commented out to speed up knit time

# Bargmannia_GO_Gasmat<- Go_lookup("Bargmannia",Bargmannia_Gaswhimat)
# Agalma_GO_Gasmat<- Go_lookup("Agalma",Agalma_gasmat)
# Nanomia_GO_Gasmat<- Go_lookup("Nanomia",Nanomia_Gasmat)
# Frillagalma_GO_Gasmat<- Go_lookup("Frillagalma",Frillagalma_Gasmat)
# Physalia_GO_Gasmat<- Go_lookup("Physalia",Physalia_Gasmat)
# Physalia_GO_Tenpalmat<- Go_lookup("Physalia",Physalia_Tenpalmat)
# 
# Bargmannia_GO_Gasdev<- Go_lookup("Bargmannia",Bargmannia_Gasdev)
# Nanomia_GO_Gasdev<- Go_lookup("Nanomia",Nanomia_Gasdev)
# Frillagalma_GO_Gasdev<- Go_lookup("Frillagalma",Frillagalma_Gasdev)
# Physalia_GO_Gasdev<- Go_lookup("Physalia",Physalia_Gasdev)
# 
# Bargmannia_GO_Gonmal<- Go_lookup("Bargmannia",Bargmannia_Gonmal)
# Frillagalma_GO_Gonmal<- Go_lookup("Frillagalma",Frillagalma_Gonmal)
# Agalma_GO_Gonmal<- Go_lookup("Agalma",Agalma_Gonmal)
# 
# Frillagalma_GO_Gonfem<- Go_lookup("Frillagalma",Frillagalma_Gonfem)
# Agalma_GO_Gonfem<- Go_lookup("Agalma",Agalma_Gonfem)
# Nanomia_GO_Gonfem<- Go_lookup("Nanomia",Nanomia_Gonfem)
# 
# Nanomia_GO_Palmat<- Go_lookup("Nanomia",Nanomia_Palmat)
# Frillagalma_GO_Palmat<- Go_lookup("Frillagalma",Frillagalma_Palmat)
# Agalma_GO_Palmat<- Go_lookup("Agalma", Agalma_Palpon)

# Nanomia_GO_Necdev<- Go_lookup("Nanomia",Nanomia_Necdev)
# Frillagalma_GO_Necdev<- Go_lookup("Frillagalma",Frillagalma_Necdev)
# Agalma_GO_Necdev<- Go_lookup("Agalma", Agalma_necdev)
# Bargmannia_GO_Necdev<-Go_lookup("Bargmannia", Bargmannia_necdev)
# 
# Diphyes_GO_Gon<- Go_lookup("Diphyes",Diphyes_Gon)

#Bargmannia_GO_Gaswhimat<-Go_lookup("Bargmannia", Bargmannia_Gaswhimat)
#Bargmannia_GO_Gasyelmat<-Go_lookup("Bargmannia", Bargmannia_Gasyelmat)

#Bargmannia_GO_Gasyelmatunique<-Go_lookup("Bargmannia", Bargmannia_Gasyelmat_unique)
#Bargmannia_GO_Gaswhimatunique<-Go_lookup("Bargmannia", Bargmannia_Gaswhimat_unique)

##identify putative transcription factors -- emphasis on putative. this will find all genes with DNA binding function

TF<-rep("N",nrow(unique_zooids))
TF[grep("GO:0003700",unique_zooids$GO_term)]<-"Y"

unique_zooids<-unique_zooids %>% dplyr::bind_cols(TF=TF)

TF2<-rep("N",nrow(DE_zooids))
TF2[grep("GO:0003700",DE_zooids$GO_term)]<-"Y"
DE_zooids<-DE_zooids %>%  dplyr::bind_cols(TF=TF2)

x<-DE_zooids %>% dplyr::mutate(zooid = stringr::str_replace(zooid,"Gastrozooid white mature","Gastrozooid mature")) %>% dplyr::mutate(zooid = stringr::str_replace(zooid,"Gastrozooid white developing","Gastrozooid developing")) %>% dplyr::mutate(blast_hit=stringr::str_replace(blast_hit,"\\w*\\|\\w*_\\w* ", ""))  %>% dplyr::mutate(blast_hit=stringr::str_replace(blast_hit,"\\{[\\|\\w\\:\\d]+\\}", "")) %>% dplyr::filter(species%in%c("Agalma elegans","Bargmannia elongata","Nanomia bijuga","Frillagalma vityazi"),zooid%in%c("Pneumatophore","Nectophore developing","Palpon mature","Gastrozooid mature","Gastrozooid developing","Gonodendron female","Gonodendron male"),TF=="Y") 

#a curated list based on the identified transcription factors

list<-readr::read_csv("transcriptionfactors")

x<-x[x$blast_hit%in%list$blast_hit,]

x<- x %>% dplyr::left_join(list)

x$short_blast_hit<-factor(x$short_blast_hit, levels=x$short_blast_hit[order(x$zooid)] %>% unique())

Agalma_TF<-x %>% dplyr::filter(species=="Agalma elegans") %>% .$short_blast_hit %>%  unique()
Bargmannia_TF<-x %>% dplyr::filter(species=="Bargmannia elongata") %>% .$short_blast_hit %>%  unique()
Frillagalma_TF<-x %>% dplyr::filter(species=="Frillagalma vityazi") %>% .$short_blast_hit %>%  unique()
Nanomia_TF<-x %>% dplyr::filter(species=="Nanomia bijuga") %>% .$short_blast_hit %>%  unique()

##check how many of these putative transcription factors are present in the original transcriptome (are differences between species a function of sequencing depth?)
check_hit<-function(object,item){
  check<-function(item,object){
  x<-object@blast_hit %>% grep(item,.)
  if(length(x>0)){
  return(TRUE)
  }
  else{
  return(FALSE)
  }
  }
  true_false<-lapply(item,check,object)
  true_false<-true_false %>% unlist()
  return(true_false)
}

y<-lapply(e,check_hit, unique(x$blast_hit))


```

We sequenced mRNA from microdissected zooids from seven different siphonophore species (at least two replicate colonies) and mapped these short-read libraries to previously published transcriptomes [@munro2018improved]. We collected RNA-seq data from, where possible, 5 different zooids and one specialized tissue, the pneumatophore, as well as unique zooids specific to *Agalma elegans* (B palpons), *Physalia physalis* (tentacular palpon), and *Bargmannia elongata* (yellow and white gastrozooids), and where possible developing and mature zooids (Table \@ref(tab:summarize-libraries)). Due to species availability, we were not able to sample more than one replicate for some zooids - single replicates were excluded from downstream differential gene expression analyses (see Fig. \@ref(fig:zooids-sampled)). 

The first component of variation that we assessed was among technical replicates. The technical replicates consist of re-sequenced developing nectophores and developing gastrozooids from the same *Frillagalma vityazi* individual that were spiked in across multiple lanes and runs. Lane and run effects have been proposed as major sources of technical variability in RNA-seq data that may confound observations of biological variation [@auer2010statistical; @mcintyre2011rna]. The differences between technical replicates (Fig. \@ref(fig:technical-replicate)) were much smaller (0.39% variance of expression distance) than the differences between zooids (98.32% variance of expression distance). Differences among technical replicates of the same zooid were correlated with library size and run, not by lane.

The second component of variation we considered was biological variation among sampled colonies. Specimens were collected in the wild at different depths and over different time periods, but despite these environmental factors there was remarkably little variation among sampled colonies. Some samples did show greater variation across replicates, such as a developing palpon replicate in *Nanomia bijuga* (Fig. \@ref(fig:pca-nanomia)), and a developing gastrozooid and male gonodendra in *Frillagalma vityazi* (Fig. \@ref(fig:pca-Frillagalma)). 

The third component of variation we considered was among zooids/specialized tissues within species. This was the greatest component of variation (Fig. \@ref(fig:pca-diphyes)--\@ref(fig:pca-physalia)). We identified genes that show higher transcript abundance in particular zooids, and in the pneumatophore, for each of the sampled species, based on pairwise comparisons (supplementary file 1). From this, we identified a subset of genes that have higher transcript abundance in only one zooid but not in any other zooid within a given species (Fig. \@ref(fig:unique-zooids), supplementary file 2).

For each species and zooid, we found GO term enrichment for biological processes consistent with the functional specialization of the zooid. In gastrozooids that are solely responsible for feeding and digestion, for example, we found these zooids to be enriched for genes involved in chitin, glutathione, and peptide catabolism, proteolysis, as well as metabolism of carbohydrates. Likewise for male gonodendra, we found GO term enrichment for biological processes such as sperm flagellum, mitotic cell cycle process, DNA replication. For female gonodendra, there was GO term enrichment for a number of biological processes including mitotic cell cycle process, DNA replication (in *Agalma elegans*), as well as a number of signalling pathways and developmental processes (in *Frillagalma vityazi*).

Within the four best sampled species, we also identified `r x$short_blast_hit %>% unique() %>% length()` higher abundance putative transcription factors (`r length(Bargmannia_TF)` out of `r y[[2]] %>% sum()` in *Bargmannia elongata*, `r length(Frillagalma_TF)` out of `r y[[4]] %>% sum` identified in *Frillagalma vityazi*, `r length(Agalma_TF)` out of `r y[[1]] %>% sum()` in *Agalma elegans* and `r length(Nanomia_TF)` out of `r y[[5]] %>% sum()` identified  in *Nanomia bijuga*). Many of these transcription factors have higher expression in several zooids regardless of developmental stage (both mature and developing zooids), and a subset have higher expression only in particular zooids and developmental stages (Fig. \@ref(fig:transcription-factors)). 

```{r summarize-libraries, verbose= FALSE, message = FALSE, echo=FALSE, comment=NA, warning=FALSE}

summary<-lapply(e,function(object){
  x=data.frame(
    Species=object@species, 
    Genes_in_trees=length(rownames(object@x)[rownames(object@x) %in% reduced_nodes$sequence_ids]),
    Genes_trees=reduced_nodes$gene_tree[reduced_nodes$sequence_ids %in% rownames(object@x)] %>% unique() %>% length()
    )
  x$Species<-as.character(x$Species)
  return(x)})

summary<-summary %>% dplyr::bind_rows() %>% arrange(desc(Genes_in_trees))
names(summary)<-c("Species", "Gene trees","Unique gene trees")

expression_data<-lapply(dge, function(object){
if(object@ddsMF$treatment=="Tentacle developing"){
  x=data.frame(
    Genes= DESeq2::counts(object@ddsMF) %>% nrow(),
    Zooids=object@ddsMF$treatment[!object@ddsMF$treatment=="Tentacle developing"] %>% unique() %>% length()
  )
}
    else{
    x=data.frame(
    Genes= DESeq2::counts(object@ddsMF) %>% nrow(),
    Zooids=object@ddsMF$treatment %>% unique() %>% length()
  )
  return(x)
  }
})

expression_data<-expression_data %>% dplyr::bind_rows()

expression_data$Species<-c("Agalma elegans","Bargmannia elongata","Apolemia lanosa","Frillagalma vityazi","Nanomia bijuga","Physalia physalis","Diphyes dispar")

summary<- summary %>% dplyr::left_join(expression_data)

names(summary) <- c("Species","Number of Genes in Gene Trees","Total Number of Gene Trees","Number of Genes","Number of Zooids Sampled")
summary<- summary %>% select(Species,`Number of Genes`, `Number of Zooids Sampled`,`Number of Genes in Gene Trees`,`Total Number of Gene Trees`)

summary %>% dplyr::mutate(
    Species= cell_spec(Species, italic= TRUE)
) %>% kable(caption="Number of genes per  species (library size), number of sampled zooids (at least two  replicates each, including developing and mature stages), total number of genes in gene trees (there can be multiple genes in a gene tree), and number of unique gene trees containing genes from this species.", caption.short ="Summary statistics for each of the sampled species, with details on the total number of genes in gene trees, and number of unique gene trees containing genes from this species.",booktabs=TRUE, align = c("l","c","c","c","c"), escape = FALSE) %>% kable_styling(full_width = F, font_size = 11) %>%
  kableExtra::kable_styling(latex_options = "hold_position") %>%
  column_spec(2:5, width = "5em")

```

### Novel species-specific zooids: can expression patterns distinguish distinct zooid types? {-}

In siphonophores, there are several instances of lineage-specific zooid diversification events. We investigated gene expression patterns between the novel zooid type and the hypothesized most closely related zooid type in three species. In *Bargmannia elongata* there are two morphologically distinct gastrozooids, that we termed "white" and "yellow" gastrozooids (Fig. \@ref(fig:novel-zooids)A and \@ref(fig:novel-zooids)B). The "yellow" gastrozooid is larger and darker and occurs as the 7th-10th gastrozooid on the stem [@dunn2005complex].  In the Portuguese man of war, *Physalia physalis*, the gastrozooid is unique compared to other gastrozooids in other species - it has a mouth, but no tentacle, and the basigaster region is greatly reduced [@Totton1960; @Mackie1960]. Meanwhile the tentacle is associated with another zooid, the tentacular palpon (Fig. \@ref(fig:novel-zooids)C) [@BARDI2007;@munro2019morphology; @Totton1960; @haeckel1888]. In *P. physalis*, both the gastrozooid and the tentacular palpon are considered to be subfunctionalized from an ancestral gastrozooid type [@munro2019morphology]. Finally, in *Agalma elegans*, there are thought to be at least two different palpon types: gastric palpons that arise at the base of the peduncle of the gastrozooid, and a palpon called the B-palpon (Fig. \@ref(fig:novel-zooids)D) [@dunn2006evolution]. The distinction between these two types of palpon is based on the location of these zooids - the gastrozooid is typically the last element of each cormidium, but based on the budding sequence, Dunn *et al* propose that the enlarged B-palpon is the last element in *A. elegans* [@dunn2006evolution]. Each of these cases represents a different type of novelty: in *Bargmannia elongata* the distinction between zooids was made based on size and color but not on obvious differences in function, in *P. physalis* the gastrozooids and tentacular palpons differ structurally and functionally, and finally in *A. elegans*, gastric palpons and B palpons differ only in colony location, development, and possibly size. 

In *Physalia physalis* `r dge_sig[[6]]$TenpalmatGasmat %>% nrow()` genes showed higher transcript abundance in the mature tentacular palpon, compared to `r dge_sig[[6]]$GasmatTenpalmat %>% nrow()` genes in the mature gastrozooid. A number of genes had higher transcript abundance in the mature tentacular palpon relative to all other tissues, of which, `r Physalia_Tenpalmat_unique %>% length()` genes were upregulated relative to other tissues that were not shared with the gastrozooid. In the gastrozooid `r Physalia_Gasmat_unique %>% length()` genes were upregulated relative to other tissues that were not shared with the tentacular palpon. A number of genes upregulated in the tentacular palpon are uncharacterized, however we identified 46 putative toxins in the tentacular palpon, including hemostasis interfering and platelet aggregation activating toxins, phospholipases, serineproteases, hydrolases, metalloendoproteases, calglandulin-like genes, and a neurotoxin. By contrast, in the gastrozooid, we found 59 upregulated putative toxins, including pore-forming Conoporin-Cn1-like and Tereporin-Ca1-like toxins, multiple neurotoxins, hydrolases, serine proteases, toxins likely involved in the promotion of blood coagulation and inhibition of platelet aggregation. Reflecting the role of the gastrozooid in digestion, we also find upregulation of digestive enzymes, including chymotrypsin-like genes.   

Between the white mature gastrozooid and the yellow mature gastrozooid in *Bargmannia elongata*, few upregulated genes were identified (`r dge_sig[[2]]$GaswhimatGasyelmat %>% nrow()` genes were up in "white" mature gastrozooids relative to `r dge_sig[[2]]$GasyelmatGaswhimat %>% nrow()` genes in "yellow" gastrozooids). Among genes that were significantly upregulated in either "white" or "yellow" gastrozooids relative to all other tissues, `r Bargmannia_Gasyelmat_unique %>% length()` genes were unique to "yellow" gastrozooids and not found in "white" gastrozooids, and `r Bargmannia_Gaswhimat_unique %>% length()` genes were found in "white" gastrozooids and not found in "yellow" gastrozooids. 

Finally, in *Agalma elegans* very few significantly upregulated genes were identified between the B palpon and gastric palpons  (`r dge_sig[[1]]$PalBmatPalmat %>% nrow()` had higher transcript abundance in B palpons and `r dge_sig[[1]]$PalmatPalBmat %>% nrow()` had higher transcript abundance in gastric palpons. All three genes have no significant blast hit and did not map to any gene trees). Genes were identified that are upregulated in B palpons relative to all other zooids (gastric palpons were excluded). Of this, `r Agalma_PalponB_unique %>% length()` genes were differentially expressed in the B palpons. Most of these genes overlapped with those differentially expressed in the gastric palpons relative to all other zooids (`r intersect(Agalma_PalponB_unique,Agalma_Palpon_unique) %>% length()` genes).

### Comparative gene expression across species: can we identify lineage and zooid specific expression patterns? {-}

**Strict Ortholog Filtering of Trees** 

```{r soft, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE}
Orthologs2<-read_tsv("strict_orthologs/Aug31/Single_Copy_Orthologue_Sequences/Strict_orthologs.fa")
Orthologs2<-Orthologs2 %>% mutate(Gene=as.character(Gene))

expression_pull<-function(object){
x<-Fulldataframe %>% dplyr::select(c(sequence_ids, Gasdev, Palmat,Gasmat, Necdev, Pne, Gonmal, Gonfem ))
subset<-Orthologs2[Orthologs2$Species %in% object@species,]
tpm<-object@tpm[rownames(object@tpm) %in% subset$Gene,] %>% as.data.frame()
treatment<-object@treatment %>% stringr::str_replace_all(.," ",".")
species<-object@species %>% stringr::str_replace_all(.," ",".")
col<-paste(treatment,object@individual, sep="_")
col<-paste(rep(species,length(treatment)),col,sep="_")
colnames(tpm)<-col
tpm<- tpm %>% rownames_to_column(., var="Gene")
tpm<-tpm %>%  left_join(Orthologs2)
tpm <- tpm %>% dplyr::select(.,-c(Species,Gene))
return(tpm)
}


expression_pull<-function(object){
x<-Fulldataframe %>% dplyr::select(c(sequence_ids, Gasdev, Palmat,Gasmat, Necdev, Pne))
subset<-Orthologs2[Orthologs2$Species %in% object@species,]
tpm<-x[x$sequence_ids %in% subset$Gene,]
#remove na columns
tpm<-tpm %>% select_if(~sum(!is.na(.)) > 0)
species<-object@species %>% stringr::str_replace_all(.," ",".")
tpm<- tpm %>% dplyr::rename_at(vars(-sequence_ids), ~ paste(species,., sep="_"))
tpm<- tpm %>% dplyr::rename(Gene=sequence_ids) %>% mutate(Gene = as.character(Gene))
tpm<-tpm %>% left_join(subset)
tpm <- tpm %>% dplyr::select(.,-c(Species,Gene))
return(tpm)
}

test<-lapply(e[c(1,2,4,5)], expression_pull)
test<-purrr::reduce(test,left_join)

rownames(test)<-test$Orthogroup
test<- test %>% na.omit()
test <-test %>% dplyr::select(-"Orthogroup")

compute_cv <- function(x) sd(x) / mean(x)
cv <- apply(test, 1, compute_cv)
df<-data.frame(cv=cv)
keep_test<-test
test<- test %>% rownames_to_column(., var="Orthogroup")
#summary(test)
test<-test %>% bind_cols(df) %>% arrange(desc(cv))
pcatest<-test  %>% dplyr::select(-c(cv,Orthogroup))

pca_data<-prcomp(t(pcatest))
pca_data_perc<-round(100*pca_data$sdev^2/sum(pca_data$sdev^2),1)
test_names<-colnames(pcatest) %>% str_split("_")
species<-lapply(test_names, `[[`, 1) %>% unlist() %>% str_replace("\\."," ")
treatment<-lapply(test_names, `[[`, 2) %>% unlist() 
#replicate<-lapply(test_names, `[[`, 3) %>% unlist()
df_pca_data <- data.frame(PC1 = pca_data$x[,1], PC2 = pca_data$x[,2], species=species, treatment=treatment)
plot_pca_ortholog<-df_pca_data %>% ggplot()+geom_point(aes(PC1,PC2, col=treatment, shape=species)) + 
  scale_shape_manual(values=c("Agalma elegans"=15,"Nanomia bijuga"=0,"Diphyes dispar"=8,"Physalia physalis"=10,"Frillagalma vityazi"=16,"Bargmannia elongata"=9, "Apolemia lanosa"=23)) +
  xlab(paste0("PC1: ",pca_data_perc[1],"% variance")) +  ylab(paste0("PC2:",pca_data_perc[2],"% variance")) +
  scale_color_manual(values=c("Gonfem"="#BFA7CE","Gonmal"="#F2B6BC","Necdev"="#B9C977","Palmat"="#F4DF1C","Gasdev"="#F78508", "Gasmat"="#F9C153", "Pne"="#8DCCE2"))


### calculating proportion of variance based on linear models

m=keep_test

# Code modified from Breschi et al. Genome Biology 2016;  https://github.com/abreschi/Rscripts/blob/master/anova.R

res = t(sapply(1:nrow(m), 
               function(i) {
                 mm = suppressMessages(reshape2::melt(m[i,]))
                 tmp = tibble(value=mm$value, species,treatment);
                 #		print(tmp)
                 aov_res = anova(lm(value~species+treatment, tmp));
                 return(unlist(aov_res[,-1]))
               }
))

tmp = tibble(value=reshape2::melt(m[1,])$value, species,treatment);
aov_res = anova(lm(value~species+treatment, tmp));
labels = apply(expand.grid(rownames(aov_res), c("SS", "MeanSq", "F", "pvalue")), 1, paste, collapse="_") 

res = data.frame(res)
colnames(res) <- labels


# Adjust pvalue
for (i in grep("pvalue", colnames(res))) {
  adj_header = paste(colnames(res)[i], "adj", sep=".")
  res[,adj_header] = p.adjust(res[,i], method="BH")
}

res = sapply(res, round, 4)
rownames(res) <- rownames(m)

# If the variance is zero set the results to NA
res[apply(m, 1, var)==0,] <- NA

res<- as.data.frame(res) %>% rownames_to_column(., var="Orthogroup")

stvariance<-as_tibble(res) %>% dplyr::select(c(Orthogroup,species_SS,treatment_SS,Residuals_SS))

stvariance<-stvariance %>% rowwise() %>% mutate(species_var=species_SS/(species_SS+treatment_SS+Residuals_SS),treatment_var=treatment_SS/(species_SS+treatment_SS+Residuals_SS))
stvariance<-stvariance %>% rowwise() %>% mutate(speciesdom=(species_var+treatment_var)>=0.75 && species_var>=2*treatment_var)
stvariance<-stvariance %>% rowwise() %>% mutate( treatmentdom= (species_var+treatment_var)>=0.75 && treatment_var>+2*species_var)

stvariance<-stvariance %>% rowwise() %>% mutate(class= case_when(speciesdom ==TRUE & treatmentdom ==FALSE ~ "SVG",treatmentdom ==TRUE & speciesdom ==FALSE ~"TVG",treatmentdom ==FALSE & speciesdom ==FALSE ~"other"))

group_colours<-c(other="grey",SVG="#F8766D",TVG="#00BFC4")
Variance_explained<-stvariance %>% ggplot()+geom_point(aes(species_var,treatment_var,col=class))+ scale_color_manual(values=group_colours) +xlab("Proportion of variance across species") + ylab("Proportion of variance across tissue")

Absolute_variance<-stvariance %>% ggplot()+geom_point(aes(species_SS,treatment_SS,col=class))+ scale_color_manual(values=group_colours) +xlab("Species Sum of Squares") + ylab("Tissue Sum of Squares")

#Add blast_hit

Orthologs_blast<-Orthologs2 %>% mutate(Gene=as.numeric(Gene))
Orthologs_blast<-Orthologs_blast %>%  left_join(blast_lookup, by =c("Gene"="sequence_ids")) %>% left_join(GO, by=c("Gene"="sequence_ids")) 



Orthologs_blast<-Orthologs_blast %>% group_by(Orthogroup) %>% summarize(Common_blast=max(blast_hit), Common_GO=max(GO_term, na.rm=TRUE))

GO_by_ortholog<-strsplit(Orthologs_blast$Common_GO,";")
names(GO_by_ortholog)<-Orthologs_blast$Orthogroup

ortholog_tree_names<-names(GO_by_ortholog)

fishertest<-function(ortholog, GO_ontology){

geneList <- factor(as.integer(ortholog_tree_names %in% unique(ortholog)))
names(geneList) <- ortholog_tree_names

GOdata <- new("topGOdata", ontology = GO_ontology, allGenes = geneList, annot = annFUN.gene2GO, gene2GO = GO_by_ortholog, nodeSize = 10)

resultFisher<-runTest(GOdata, algorithm = "classic", statistic = "fisher")
allRes <- GenTable(GOdata, classicFisher = resultFisher, ranksOf = "classicFisher", topNodes = 20)

return(allRes)
}

stvariance<-stvariance %>% left_join(Orthologs_blast)

m<- m %>% tibble::rownames_to_column(., var="Orthogroup")
m3<-stvariance %>% select(Orthogroup,class,Common_blast)
m<-m %>% left_join(m3)
m<-reshape2::melt(m) %>% rename(expression=value)
m<-m %>% separate(variable, c("species","tissue"),sep="_") 


#stvariance %>% filter(treatmentdom==TRUE) %>% .$Orthogroup %>% fishertest(.,"BP")

##Dominated by recombination -- especially gonozooids driving major tissue differences? Also - numbers of genes considered here are very small, hence weak p values

#stvariance %>% filter(speciesdom==TRUE) %>% .$Orthogroup %>% fishertest(.,"BP")

expression_pull2<-function(object){
x<-Fulldataframe %>% dplyr::select(c(sequence_ids, Gasdev, Palmat,Gasmat, Necdev, Pne))
subset<-Orthologs2[Orthologs2$Species %in% object@species,]
tpm<-x[x$sequence_ids %in% subset$Gene,]
tpm<-tpm %>% mutate(GasdevGasmat=(Gasdev+1)/(Gasmat+1),PalmatGasmat=(Palmat+1)/(Gasmat+1), PneGasmat=(Pne+1)/(Gasmat+1), NecdevGasmat=(Necdev+1)/(Gasmat+1))
tpm<-tpm %>% select(-c(Gasdev, Gasmat,Palmat, Necdev, Pne))
#remove na columns
tpm<-tpm %>% select_if(~sum(!is.na(.)) > 0)
species<-object@species %>% stringr::str_replace_all(.," ",".")
tpm<- tpm %>% dplyr::rename_at(vars(-sequence_ids), ~ paste(species,., sep="_"))
tpm<- tpm %>% dplyr::rename(Gene=sequence_ids) %>% mutate(Gene = as.character(Gene))
tpm<-tpm %>% left_join(subset)
tpm <- tpm %>% dplyr::select(.,-c(Species,Gene))
return(tpm)
}

test2<-lapply(e[c(1,2,4,5)], expression_pull2)
test2<-purrr::reduce(test2,left_join)

rownames(test2)<-test2$Orthogroup
test2<- test2 %>% na.omit()
test2 <-test2 %>% dplyr::select(-"Orthogroup")

compute_cv <- function(x) sd(x) / mean(x)
cv <- apply(test2, 1, compute_cv)
df<-data.frame(cv=cv)
keep_test2<-test2
test2<- test2 %>% rownames_to_column(., var="Orthogroup")

test2<-test2 %>% bind_cols(df) %>% arrange(desc(cv))

pcatest2<-test2  %>% dplyr::select(-c(cv,Orthogroup))

pca_data2<-prcomp(t(pcatest2))
pca_data_perc2<-round(100*pca_data2$sdev^2/sum(pca_data2$sdev^2),1)
test_names2<-colnames(pcatest2) %>% str_split("_")
species2<-lapply(test_names2, `[[`, 1) %>% unlist() %>% str_replace("\\."," ")
treatment2<-lapply(test_names2, `[[`, 2) %>% unlist() 
#replicate<-lapply(test_names, `[[`, 3) %>% unlist()
df_pca_data2 <- data.frame(PC1 = pca_data2$x[,1], PC2 = pca_data2$x[,2], species=species2, treatment=treatment2)
plot_pca_ortholog2<-df_pca_data2 %>% ggplot()+geom_point(aes(PC1,PC2, col=treatment, shape=species)) + 
  scale_shape_manual(values=c("Agalma elegans"=15,"Nanomia bijuga"=0,"Frillagalma vityazi"=16,"Bargmannia elongata"=9)) +
  xlab(paste0("PC1: ",pca_data_perc[1],"% variance")) +  ylab(paste0("PC2:",pca_data_perc[2],"% variance")) +
  scale_color_manual(values=c("GasdevGasmat"="#F78508","NecdevGasmat"="#B9C977","PalmatGasmat"="#F4DF1C","PneGasmat"="#8DCCE2"))

### calculating proportion of variance based on linear models

m2=keep_test2

# Code modified from Breschi et al. Genome Biology 2016;  https://github.com/abreschi/Rscripts/blob/master/anova.R

res2 = t(sapply(1:nrow(m2), 
               function(i) {
                 mm = suppressMessages(reshape2::melt(m2[i,]))
                 tmp = tibble(value=mm$value, species2,treatment2);
                 #		print(tmp)
                 aov_res = anova(lm(value~species2+treatment2, tmp));
                 return(unlist(aov_res[,-1]))
               }
))

tmp2 = tibble(value=reshape2::melt(m2[1,])$value, species2,treatment2);
aov_res2 = anova(lm(value~species2+treatment2, tmp2));
labels2 = apply(expand.grid(rownames(aov_res2), c("SS", "MeanSq", "F", "pvalue")), 1, paste, collapse="_") 

res2 = data.frame(res2)
colnames(res2) <- labels2


# Adjust pvalue
for (i in grep("pvalue", colnames(res2))) {
  adj_header = paste(colnames(res2)[i], "adj", sep=".")
  res[,adj_header] = p.adjust(res2[,i], method="BH")
}

res2 = sapply(res2, round, 4)
rownames(res2) <- rownames(m2)

# If the variance is zero set the results to NA
res2[apply(m2, 1, var)==0,] <- NA

res2<- as.data.frame(res2) %>% rownames_to_column(., var="Orthogroup")

stvariance2<-as_tibble(res2) %>% dplyr::select(c(Orthogroup,species2_SS,treatment2_SS,Residuals_SS))

stvariance2<-stvariance2 %>% rowwise() %>% mutate(species_var=species2_SS/(species2_SS+treatment2_SS+Residuals_SS),treatment_var=treatment2_SS/(species2_SS+treatment2_SS+Residuals_SS))
stvariance2<-stvariance2 %>% rowwise() %>% mutate(speciesdom=(species_var+treatment_var)>=0.75 && species_var>=2*treatment_var)
stvariance2<-stvariance2 %>% rowwise() %>% mutate( treatmentdom= (species_var+treatment_var)>=0.75 && treatment_var>+2*species_var)

stvariance2<-stvariance2 %>% rowwise() %>% mutate(class= case_when(speciesdom ==TRUE & treatmentdom ==FALSE ~ "SVG",treatmentdom ==TRUE & speciesdom ==FALSE ~"TVG",treatmentdom ==FALSE & speciesdom ==FALSE ~"other"))

group_colours<-c(other="grey",SVG="#F8766D",TVG="#00BFC4")
Variance_explained2<-stvariance2 %>% ggplot()+geom_point(aes(species_var,treatment_var,col=class))+ scale_color_manual(values=group_colours) +xlab("Proportion of variance across species") + ylab("Proportion of variance across tissue")

Absolute_variance2<-stvariance2 %>% ggplot()+geom_point(aes(species2_SS,treatment2_SS,col=class))+ scale_color_manual(values=group_colours) +xlab("Species Sum of Squares") + ylab("Tissue Sum of Squares")

stvariance2<-stvariance2 %>% left_join(Orthologs_blast)

m2<- m2 %>% tibble::rownames_to_column(., var="Orthogroup")
m4<-stvariance2 %>% select(Orthogroup,class,Common_blast)
m2<-m2 %>% left_join(m4)

m2_keep <- m2
m2<-reshape2::melt(m2) %>% rename(expression=value)
m2<-m2 %>% separate(variable, c("species","tissue"),sep="_")

m2<-m2 %>% mutate(species=str_replace(species,"\\."," ")) %>% left_join(Orthologs2, by=c("Orthogroup","species"="Species"))

genes_to_trees<-nodes_ace %>% select(gene_tree, sequence_ids) %>% mutate(Gene=as.character(sequence_ids)) %>% select(-sequence_ids)
genes_to_trees<-genes_to_trees %>% na.omit()

species_variable<-m2 %>% filter(class=="SVG") %>% left_join(genes_to_trees)
species_variable_subset<-species_variable %>% filter(gene_tree!="NA")

treatment_variable<-m2 %>% filter(class=="TVG") %>% left_join(genes_to_trees)
treatment_variable_subset<-treatment_variable %>% filter(gene_tree!="NA")

#stvariance2 %>% filter(class=="SVG") %>% .$Orthogroup %>% fishertest(.,"BP")

#stvariance2 %>% filter(class=="TVG") %>% .$Orthogroup %>% fishertest(.,"BP")

Orthologs2<-Orthologs2 %>% left_join(genes_to_trees) 
Orthologs_tree<-Orthologs2 %>% na.omit %>% group_by(Orthogroup) %>% summarise(gene_tree = paste(unique(gene_tree), collapse=", "))

readr::write_csv(treatment_variable, "Supplementary_Data/Supplementaryfile4.csv", na = "NA")
readr::write_csv(species_variable, "Supplementary_Data/Supplementaryfile5.csv", na = "NA")

```

(ref:SOFT-plot) Results of Strict Ortholog Filtering of Trees analyses. Top row: results using TPM10K expression values; Bottom row: results using TPM10K expression ratios. For both analyses of expression variance explained by species and tissue (right-hand column), we identified highly species-variable genes (SVG, red) and highly tissue-variable genes (TVG, blue). In SVGs, the proportion of variance explained by species is two times greater than that explained by tissues; while in TVGs the proportion of variance explained by tissues is two times greater than that explained by species. All SVGs and TVGs are genes for which both species and tissue explain greater than 75% of variance. A. PCA plot of expression values (TPM10K) of `r res %>% nrow()` strict ortholog genes, showing a species dominated clustering pattern. B.  The proportion of expression variance explained by species (x-axis) versus the proportion of expression variance explained by tissues (y-axis), using TPM10K expression values. C. PCA plot of expression ratios of `r res2 %>% nrow()` strict ortholog genes, showing a tissue dominated clustering pattern.  D. The proportion of expression variance explained by species (x-axis) versus the proportion of expression variance explained by tissues (y-axis), using expression ratios. Abbreviations: Gasdev = developing gastrozooid, Necdev = developing nectophore, Palmat = mature palpon, Pne = pneumatophore, Gasmat = mature gastrozooid.

```{r SOFT-plot, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=7.5, fig.height=7, fig.cap='(ref:SOFT-plot)', fig.align='center'}

### Files used to make figure in illustrator:

#pdf(file = "figures/plot_pca_ortholog2.pdf", width=6, height=4)
#plot_pca_ortholog2
#dev.off()
#pdf(file = "figures/plot_pca_ortholog.pdf", width=6, height=4)
#plot_pca_ortholog
#dev.off()
#pdf(file = "figures/Variance_explained.pdf", width=7, height=7)
#Variance_explained
#dev.off()
#pdf(file = "figures/Variance_explained2.pdf", width=7, height=7)
#Variance_explained2
#dev.off()

knitr::include_graphics("figures/SOFT_analyses.pdf",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))


```

Most comparative studies of gene expression focus exclusively on strict 1:1 orthologs, requiring an assessment of orthology across all species. We call this type of analysis Strict Ortholog Filtering of Trees (SOFT), as it limits analyses to a subset of genes with very specific evolutionary histories [e.g @Brawand:2011du; @cardoso2019gene; @Levin:2016bp; @yang2013organ]. Following this approach, we used Orthofinder 2 (v2.4.0) [@emms2019orthofinder] to identify a set of `r Orthologs2$Orthogroup %>% unique() %>% length()` strict orthologs for four of the best sampled species (*Agalma elegans*, *Bargmannia elongata*, *Frillagalma vityazi*, and *Nanomia bijuga*), of which `r stvariance %>% nrow()` orthologs had expression values in four zooids/tissues: gastrozooids (developing and mature), nectophores (developing), palpons (mature), and the pneumatophore. Using TPM10K alone, we found that orthologs clustered largely by species rather than zooid/ tissue (Fig. \@ref(fig:SOFT-plot)A). Following Breschi et al. @breschi2016gene (see methods), we used linear models to identify the proportion of variance that could be explained by species or zooid/tissue (Fig. \@ref(fig:SOFT-plot)B), finding out of `r stvariance  %>% nrow()` orthologs, `r stvariance %>% filter(class=="SVG") %>% nrow()` species-variable genes (SVG), and `r stvariance %>% filter(class=="TVG") %>% nrow()` zooid/tissue-variable genes (TVG). 

The strong species-dominated clustering observed when comparing expression values directly may be due to differences in counting efficiency between species, especially as we used reference transcriptomes [@dunn2013phylogenetic]. To account for this, we used ratios of expression values, using TMP10K expression values of the most commonly sampled zooid -- mature gastrozooid -- as the denominator [@dunn2013phylogenetic]. Using this approach, we found that orthologs clustered largely by zooid/tissue rather than species (Fig. \@ref(fig:SOFT-plot)C). Using linear models, we identified, out of `r stvariance2  %>% nrow()` orthologs, `r stvariance2 %>% filter(class=="SVG") %>% nrow()` species-variable genes (SVG), and `r stvariance2 %>% filter(class=="TVG") %>% nrow()` tissue-variable genes (TVG) (Fig. \@ref(fig:SOFT-plot)D). GO terms enriched among TVG included embryonic morphogenesis, embryo development, cartilage development, and a number of metabolic and biosynthetic processes. Meanwhile, SVG were enriched in GO terms such as cellular response to stress, DNA repair and cell cycle processes. A list of TVG and SVG can be found in supplementary files 4 and 5 respectively.  

**Species Tree Edge Extraction and Layover** 

```{r plot_tree, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE}

change<-function(col_value1, col_value2,nhx){
  parents = nhx@phylo$edge[,1]
  children = nhx@phylo$edge[,2]
  edge_length=nhx@phylo$edge.length
  df = data.frame(
    parent=parents,
    node=children,
    edge_num=round((((pull(nhx@data, col_value1)[children] +1 )/(pull(nhx@data, col_value2)[children]+1))-( (pull(nhx@data,col_value1)[parents]+1)/(pull(nhx@data,col_value2)[parents] +1)))/edge_length,2)
    )
  return(df)
}

plottreeexpression<-function(gene_tree_value,col_value){
  
nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==gene_tree_value][[1]]
    
edge<-change(col_value,nhx)

levels(nhx@data$Ev)<-c("D","S")

tips<-pull(nhx@data,col_value)

top_hit<-genes_to_tips %>% filter(gene_tree==gene_tree_value) %>% .$blast_hit

p<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names ), hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,1))

p<-p+geom_text(aes(label=round(tips,3)),vjust=-1,hjust=1, size=3, col="Black")

p<- p %<+% edge + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3)  

p<- p+ggtitle(paste0("Most frequent blast hit=",top_hit))
return(p)
}

plotsimtreeexpression<-function(gene_tree,col_value){
  
nhx<-gene_trees_sim_calibrated_pruned[gene_trees_sim_digests==gene_tree][[1]]

blast_hit <- genes_to_tips %>% dplyr::filter(gene_tree==gene_tree) %>% .$blast_hit
    
edge<-change(col_value,nhx)

levels(nhx@data$Ev)<-c("D","S")

tips<-pull(nhx@data,col_value)

top_hit<-nhx@data$blast_hit %>% max(.,na.rm=TRUE)

p<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names ), hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,1))

p<-p+geom_text(aes(label=round(tips,3)),vjust=-1,hjust=1, size=3, col="Black")

p<- p %<+% edge + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3)  

p<- p+ggtitle(paste0("Most frequent blast hit=",top_hit))
return(p)
}


plotmultitree<-function(gene_tree,col_value){
  
nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==gene_tree][[1]]

levels(nhx@data$Ev)<-c("D","S")

tips<-c()
for(tip in 1:length(col_value)){
tips$col_value[[tip]]<-pull(nhx@data,col_value[[tip]])
}

p<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names ), hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,2))

p<-p+geom_text(aes(label=round(tips$col_value[[1]],3)),vjust=-4.5,hjust=1, size=3, col="Blue")

p<-p+geom_text(aes(label=round(tips$col_value[[2]],3)),vjust=-3.5,hjust=1, size=3, col="Red")

p<-p+geom_text(aes(label=round(tips$col_value[[3]],3)),vjust=-2.5,hjust=1, size=3, col="Dark Green")

p<-p+geom_text(aes(label=round(tips$col_value[[4]],3)),vjust=-1.5,hjust=1, size=3, col="Orange")

p<-p+geom_text(aes(label=round(tips$col_value[[5]],3)),vjust=1.5,hjust=1, size=3, col="Dark Red")

p<-p+geom_text(aes(label=round(tips$col_value[[6]],3)),vjust=2.5,hjust=1, size=3, col="Black")

p<-p+geom_text(aes(label=round(tips$col_value[[7]],3)),vjust=3.5,hjust=1, size=3, col="Purple")

p<-p+geom_text(aes(label=round(tips$col_value[[8]],3)),vjust=4.5,hjust=1, size=3, col="Dark Grey")

return(p)
}
```

For our Species Tree Edge Extraction and Layover (STEEL) analyses (see Fig. \@ref(fig:methods-comp) and methods), we used transcriptome and genome data from `r species_tree$tip.label %>% length()` cnidarian species to generate a total of `r gene_trees %>% length()` gene trees, of which `r gene_trees_ace_calibrated[!is.na(gene_trees_ace_calibrated)] %>% length()` gene trees passed filtering criteria (see Methods). We used Orthofinder 2 [@emms2019orthofinder] to reconcile the species tree and the gene trees and annotate each gene tree node as either a speciation or duplication events. The number of genes represented in these gene trees is shown in Table \@ref(tab:summarize-libraries). The internal nodes on these gene trees consist of `r nodes_ace %>% filter(is.na(species) & Ev=="S") %>% nrow()` speciation events and `r nodes_ace %>% filter(is.na(species) & Ev=="D") %>% nrow()` duplication events. Expression values (TPM10K) were mapped to the tips of the gene trees for each zooid/tissue separately, and ancestral values were inferred at internal nodes with maximum likelihood. We focused on expression in a subset of zooids and tissues that are common across the sampled species: gastrozooids (developing and mature), nectophores (developing), palpons (mature), and the pneumatophore. The distribution of expression changes along species-equivalent branches are shown in figure \@ref(fig:changes-branches). 

As with the SOFT analyses, we investigated the impact of counting efficiency on our comparative analyses. We used the TPM10K values at the tips and nodes to calculate expression ratios and calculated changes in expression ratios across a branch for pairs of tissues, using the mature gastrozooid values as the denominator. The distribution of expression ratio changes along species-equivalent branches (branches in the gene tree that correspond to equivalent branches in the species tree, and are descended from speciation events) are shown in figure \@ref(fig:changes-branches2). We found that the variance of change across a given species-equivalent branch is considerably higher for raw TPM10K values as compared with ratios of expression (Fig.\@ref(fig:variance-structure), Fig.  \@ref(fig:variance-structure2)). We also identified the number of gene tree branches that had a negative (change <= -2), positive (change >= 2) or neutral (change > -2 and < 2) change (Fig.\@ref(fig:neg-pos)). For each species-equivalent branch, we found more negative and positive branches than neutral branches when we used TPM10K values, while for expression ratios, we found that the majority of branches show no change across the branch. This suggests that as for the SOFT analyses, counting efficiency has a large impact on expression change and in turn leads to large gene/branch specific signal. For expression ratios, these results indicate that the vast majority of branches show close to 0 (neutral) change across the branch, suggesting that for closely related genes, expression ratios tend not to differ. 

(ref:neg-pos) Number of branches with negative, positive or neutral changes across species-equivalent branches. Top panel: species phylogeny with branch IDs given as letters. Bottom left panel: changes of TPM10K expression across each of the species-equivalent branches (IDs correspond to branch IDs shown in the species tree); bottom right panel: changes of expression ratios across species equivalent branches. Neutral change (green) is defined as <= 2 and >= -2, negative change (red) is defined as <-2, and positive change (blue) is defined as > 2. These units correspond to TPM10K ratios/substitutions per site. GasdevGasmat = developing gastrozooid / mature gastrozooid, NecdevGasmat = developing nectophore / mature gastrozooid, PalmatGasmat = mature palpon / mature gastrozooid, PneGasmat = Pneumatophore / mature gastrozooid. 

```{r neg-pos, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=8, fig.height=10, fig.cap='(ref:neg-pos)', fig.align='center'}

Changes_raw<-edges_tidy %>% filter(treatment %in% c("Gasdev", "Gasmat", "Pne", "Necdev", "Palmat")) %>% group_by(S_new, treatment) %>% dplyr::summarise(Neutral=sum(change<2 & change >-2), Positive=sum(change>=2), Negative=sum(change<=-2)) %>% gather(c(Neutral,Positive,Negative), key = "value", value = "number") %>% ggplot()+geom_point(aes(treatment, number, col=value))+facet_wrap(~S_new) + xlab("") + ylab("Number of branches")+ggtitle(label="TPM10K expression values") + theme(plot.title = element_text(hjust = 0.5)) + theme(axis.text.x=element_text(angle = 30, hjust = 1)) + ylim(0,1500)

Changes_ratio<-edges_tidy %>% filter(treatment%in%c("GasdevGasmat", "PneGasmat", "NecdevGasmat", "PalmatGasmat")) %>% group_by(S_new, treatment) %>% dplyr::summarise(Neutral=sum(change<2 & change >-2), Positive=sum(change>=2), Negative=sum(change<=-2)) %>% gather(c(Neutral,Positive,Negative), key = "value", value = "number") %>% ggplot()+geom_point(aes(treatment, number, col=value))+facet_wrap(~S_new) + xlab(" ") + ylab("")+ggtitle(label="Expression ratios") + theme(plot.title = element_text(hjust = 0.5)) + theme(axis.text.x=element_text(angle = 30, hjust = 1)) 

legend4<- get_legend(Changes_raw) 

Changes_raw<- Changes_raw +guides(color=FALSE)
Changes_ratio<-Changes_ratio +guides(color=FALSE)

layout4<-rbind(c(1,1,NA),c(2,3,4))

grid.arrange(tree_reference, Changes_raw,Changes_ratio,legend4, bottom="Tissue",widths = c(2,2,0.5),layout_matrix = layout4)

```

Out of a total of `r edges_tidy_directional %>% .$gene_tree %>% unique %>% length()` final gene trees considered in these analyses, `r edges_tidy_directional %>% filter(change_direction=="Neutral") %>% .$gene_tree %>% unique %>% length()` gene trees contained branches with neutral changes, `r edges_tidy_directional %>% filter(change_direction=="Positive") %>% .$gene_tree %>% unique %>% length()` gene trees contained branches with positive changes, and `r edges_tidy_directional %>% filter(change_direction=="Negative") %>% .$gene_tree %>% unique %>% length()` gene trees contained branches with negative changes. Expression ratio changes across species-equivalent branches from all gene trees is available in supplementary file 6 (note: in this file 'blast hit' is the most frequent blast hit for the gene tree, gene identity should still be confirmed for each gene, particularly in large gene trees).

The vast majority of gene trees contain species-equivalent branches with neutral changes, including a number of transcription factors and morphogenic signalling pathway genes. For example, in ratios of developing and mature gastrozooids, among the relevant species-equivalent branches in the Wnt gene phylogeny, changes are neutral across Wnt3 and Wnt2 branches, with the exception of a slight positive change across the branch leading to a *Diphyes dispar* Wnt2 paralog, suggesting a higher relative expression of this gene in the developing gastrozooid of *D. dispar* (Fig. \@ref(fig:wntplot)). For all zooid ratios, we find very consistent Wnt3 expression patterns with neutral or very small positive or negative changes in expression ratio across the branches. Indeed, across all cnidarians examined, Wnt3 has consistent localised expression at the oral pole, likely playing a role in axis formation and maintenance [@bagaeva2019wnt; @guder2006wnt; @hensel2014lineage; @hobmayer2000wnt; @momose2008maternally; @nawrocki2013expression; @sanders2015patterns]. However, for Pneumatophore/Gastrozooid ratios, some large changes are seen across branches K and L, leading to a *Agalma elegans* and *Nanomia bijuga* Wnt2 paralog respectively, likewise for Palpon/Gastrozooid a large change is seen across branch J, leading to a *Frillagalma vityazi* Wnt2 paralog. In *Nematostella vectensis* and *Hydractinia echinata*, Wnt2 is expressed in the middle of the polyp [@hensel2014lineage; @kusserow2005unexpected]. Without more detailed spatial expression patterns, it is difficult to know whether these differences in expression between species reflect species-specific differences in axial patterning and morphogenesis, such as an expansion or reduction of the expression domain. 

We also looked at gene trees with species-relative branches that have very large changes ( 5 < change < -5), representing putative lineage-specific expression patterns. In Frizzled5/8, Wnt4, and homeobox (putatively Hox-B8) gene-trees, for example, we found very large positive changes in ratios of developing gastroozoid/mature gastrozooids across the same species-equivalent branches, J (leading to *Frillagalma vityazi*) and D (leading to Euphysonectae, the clade comprising *A. elegans*, *N. bijuga*, and *F. vityazi*). In Frizzled5/8 for developing nectophores/gastrozooids we also see a large positive change across branch L (leading to *Nanomia bijuga*) and J (leading to *Frillagalma vityazi*) and negative change across branch K (leading to *Agalma elegans*). In ratios of developing nectophores/gastrozooids, we also found very large positive changes in branch J for Wnt-7b-like, Wnt-4, (putatively) Hox-B8 and NKX1-2. Likewise in pneumatophore/gastrozooid ratios, we also find large positive changes in branch J and L for Frizzled5/8, and a negative change across branch K. Across J, we also see positive changes in pneumatophore/gastrozooid in Hox-B8 and NKX1-2. Finally, for branch K, we see large positive changes in Wnt4 and Wnt7b-like. Many of these genes have been shown to have specific expression domains patterning cnidarian bodies [@hensel2014lineage; @kusserow2005unexpected; @ryan2007pre; @sinigaglia2013bilaterian], and further detailed expression analyses of these genes and others within the zooid are required to determine whether the patterns identified here reflect differences in expression domain between species and zooid. 

## Discussion {-}

**Evolution of gene expression in Siphonophores** 

In this study, we used RNA-seq to investigate the functional specialization and evolution of zooids across siphonophore species. In our analysis of differential expression within species, we found a large number of differentially expressed genes across siphonophore zooids, reflecting the distinct anatomy and function of these zooids. In addition, we identified potential transcription factors that are significantly differentially expressed within particular zooids, with potential homologs found in several species, which are interesting candidates for future study in siphonophores or related cnidarian colonial groups. Using these within-species differential gene expression analyses, we identified distinct expression patterns that may be used to define particular zooid types. 

We also explored gene expression patterns in three zooid types that are unique to three species, *Physalia physalis*, *Agalma elegans*, and *Bargmannia elongata*. In siphonophores, different zooid types are typically defined based on morphological and functional differences/similarities, and on the location of the zooid within the colony (which is determined in most species by the asexual budding process in the growth zone, which gives rise to a repeating pattern of zooids along the stem) [@dunn2005complex; @dunn2006evolution; @mackie1987siphonophore; @totton1965synopsis]. Based on both morphology and development, *Physalia physalis* tentacular palpons are considered to be a distinct and unique zooid type [@munro2019morphology], and the differential gene expression data indicate that there is a clear morphological and functional difference between gastrozooids and tentacular palpons. We also identified several putative toxin genes with distinct expression profiles between these two zooids, which matches tissue-specific venom observed in other cnidarian species [@doi:10.1093/gbe/evw155; @ames2016new; @10.1093/gbe/evab081]. 

For "yellow" and "white" gastrozooids in *Bargmannia elongata*, the differential expression data point to few morphological or functional differences between these two zooids. Through pairwise differential expression patterns between "yellow" or "white" gastrozooids and other zooids within the colony we were able to identify hundreds of genes that are significantly differentiated in one zooid and not the other. This suggests that these two gastrozooids may be distinct zooid types, although they are functionally very similar to one another. Sequencing at greater depth, and also functional work within this species may help clarify the nature of these differences. By contrast, there isn't strong evidence with our data that the B palpon and gastric palpons in *Agalma elegans* are sufficiently different to constitute a novel zooid type. These findings suggest that location within the colony is not necessarily sufficient to designate a novel zooid type. 

With our SOFT analyses we found that the vast majority of identified orthologs were neither exclusively species-variable or tissue-variable, however we were nevertheless able to identify a subset of either tissue- or species- variable genes. As has been observed for vertebrate organs, we find based on GO terms that identified species-variable genes tend to play a role in basic cellular functions (so-called 'housekeeping genes'), as compared to tissue-variable genes [@@breschi2016gene]. 

What can we learn from the STEEL analyses, using ratios of expression data? It's important to stress that changes in expression ratios cannot tell us about changes in expression *magnitude*. A tissue-variable gene that is either highly or lowly expressed in, for example, nectophores relative to gastrozooids in all species, may show largely neutral changes across all species-equivalent branches. The sign (positive or negative) of change provides an indication of whether expression in the child node is relatively higher in the denominator or the numerator, relative to the parent node. We find overall that most expression ratios show very little change across species-equivalent gene tree branches -- suggesting that gene expression patterns tend to be largely consistent among species. Positive or negative changes across branches, especially very large changes, across branches, represent putative linage specific shifts in expression. Whether these reflect distinct morphological or functional changes in the tissues of a particular species or clade requires further validation -- a difficulty in this system, where the species are difficult to collect and maintain in the lab.

**Challenges and solutions in the analysis of the evolution of gene expression** 

In this study we used phylogenetic comparative methods to investigate gene expression patterns within homologous zooids across species. To do this, we implemented a new approach called STEEL. Using gene trees, we considered expression evolution within the context of the gene phylogeny, which may differ significantly from the species phylogeny. Where traditional approaches identify strict ortholog genes *before* conducting analyses, we use information from all gene copies within a gene tree regardless of their evolutionary history of duplication or speciation. By filtering our data by branches, we consider more genes than we would be able to with a strict ortholog approach, and additionally, we investigate expression patterns at both the tips and internal nodes of gene phylogenies. Although we focus here on expression following speciation events, it is also possible to compare expression following duplication and speciation events.

Using ancestral trait reconstruction, we also overcame sampling issues at the tips of the gene and species phylogeny, as expression values of different treatments can be reconstructed at deep internal nodes, even where there may be inconsistent sampling at the tips. That is, even if expression values are missing for a given tissue and gene in a gene phylogeny, we are still able to examine expression patterns within this gene tree. By contrast, in the SOFT analyses, incomplete sampling in the expression matrix leads to the elimination of the ortholog from the analysis. 

As with all methods that rely on mapping to reference transcriptomes rather than genomes, this approach is limited by the quality of the reference transcriptomes. Ratios of expression helped significantly to improve issues of differences in count efficiency among species [@dunn2013phylogenetic]. However, reference transcriptome quality also has an impact on the quality of the gene trees used for expression mapping. Not all reference transcriptomes were sequenced to equal depth among species, and this has important effects on the presence or absence of genes from particular species within the gene tree. This not only has an effect on the representation of expression values, but also impacts the power to investigate patterns of expression among branches within a gene tree. With genome sequencing becoming cheaper and more readily available, the widespread availability of reference genomes will help alleviate many of these issues. Reference genomes will also improve gene models, enabling the distinction of different alleles of the same gene from duplicated gene copies, this in turn will improve the quality of the gene trees. However, gene loss will nevertheless present a challenge to these analyses.

**Perspectives** 

With the expansion of functional genomic tools, including RNA-seq and single cell sequencing methods, there is considerable interest in looking not only at how genomic variation gives rise to phenotypic diversity in a single species or organism, but also at how functional genomic variation shapes phenotypic diversity across multiple closely and distantly related species to understand broader evolutionary patterns and processes [@Brawand:2011du; @barbosa2012evolutionary; @breschi2016gene; @cardoso2019gene; @clarke2017evolutionary; @doi:10.1093/gbe/evw155; @merkin2012evolutionary; @necsulea2014evolution; @perry2012comparative;  @Levin:2016bp; @sudmant2015meta; @ma2018comparative; @yang2013organ; @zhang2014comparative; @10.1093/molbev/msz212; @fukushima2020amalgamated]. Many of these analyses have the goal of identifying shared expression patterns among modular biological units (cell type, tissue, organ, zooid) across species, in order to identify commonalities in expression patterns among species. This is of particular interest for medically orientated fields interested in understanding the extent to which expression results can be extrapolated from one model organism to another. Within-species differential gene expression analyses, SOFT and STEEL analyses can all be useful tools for exploring common gene expression patterns among species. Another goal is to identify expression patterns in a particular biological unit that are unique to a particular species or even clade. For these questions in particular, as long as homologous tissues are available across species, STEEL analyses provide an opportunity to investigate lineage-specific expression patterns, especially where additional trait values may be mapped to gene or species phylogenies.

## Methods {-}

All scripts for the analyses are available in a git repository at https://github.com/dunnlab/Siphonophore_Expression. The most recent commit at the time of the analysis presented here was XXXX.

**Collecting** \newline
Specimens were collected in the north-eastern Pacific Ocean in Monterey Bay and, in the case of _Physalia physalis_ the Gulf of Mexico. Specimens were collected by remotely operated vehicle (ROV) or during blue-water SCUBA dives. *Physalia* specimens were collected by hand from the beach after being freshly washed on-shore by prevailing winds. Available physical vouchers have been deposited at the Peabody Museum of Natural History (Yale University), New Haven, CT. Specimens were relaxed using 7.5% MgCl~2~ hexahydrate in Milli-Q water at a ratio of 1/3 MgCl~2~ and 2/3 seawater. Zooids were subsequently dissected from the colony and flash frozen in liquid nitrogen. Colonies were cooled to collection temperatures (e.g 4 degrees C for deep sea species) while the dissections took place. Dissections took no longer than 15-20 minutes. In the case of large colonies, the stem was cut and only partial sections of the colony were placed under the microscope at a given time. Each replicate individual represents a genetically distinct colony from the same species. Replicate specimens were of an equivalent colony size, and zooid replicates were also equivalent sizes. Larger zooid types, such as gastrozooids, were sampled as a single zooid, but smaller zooids were pooled. Pooled zooids were of a comparable maturity and sampled from the same location in a single colony. 

**Sequencing** \newline
mRNA was extracted directly from tissue using Zymo Quick RNA MicroPrep (Zymo #R1050), including a DNase step, and subsequently prepared for sequencing using the Illumina TruSeq Stranded Library Prep Kit (Illumina, #RS-122-2101). 50 base-pair single-end libraries were all sequenced on the HiSeq 2500 sequencing platform. Three sequencing runs were conducted, representing three full flow cells. To avoid potential run/lane confounding effects, where possible, libraries of multiple zooids/tissue of a single individual in a species were barcoded and pooled in a single sequencing lane, and replicate lanes of zooids/tissue from different individuals of the same species were sequenced in separate runs. Additionally, two libraries were run as technical replicates across all runs and many lanes, for a total of 20 technical replicates. Sequences, along with sampling metadata, are publicly available on NCBI under BioProject ID PRJNA540747.

**Analysis**

**_Sequencing & Differential gene expression_** \newline
Short read libraries were mapped to previously published transcriptomes (150bp paired end) [@munro2018improved] using Agalma v 2.0.0 [@Dunn:2013kw; @guang2021revising], which uses a number of existing tools for transcript quantification, including RSEM (which uses Bowtie) [@langmead2009ultrafast;@li2011rsem]. Using the `agalmar` package (https://github.com/caseywdunn/agalmar), we filtered out genes that were flagged as being rRNA, and selected only protein coding genes. We also only considered genes that were greater than 0 in at least two libraries. Differential gene expression analyses, including normalization, were conducted in R, using the `DESeq2` package [@love2014moderated]. Libraries that were found to be outliers based on mean cook's distance were removed from the DESeq object and from downstream analyses and normalization. Testing for differential expression was conducted using the `results()` function in DESeq2. Genes were considered to be significantly differentially expressed if adjusted p-values (Bonferroni correction) were less than 0.05. Differential expression analyses were only conducted on zooids/tissue with two or more replicates.

GO annotations were retrieved for each of the reference translated transcriptomes [@munro2018improved] using the PANNZER2 web server [@toronen2018pannzer2].The PANNZER2 format was modified to match the gene2GO format required for the package `topGO` [@topGO]. Gene set enrichment analyses were carried out within species using the R package `GOseq` [@young2010gene], which takes gene length into account. Over and underrepresented categories were calculated using the Wallenius approximation, and p-values were adjusted using the Benjamini and Hochberg method. Categories with an adjusted p-value below 0.05 are considered enriched. Gene set enrichment analyses were also conducted at the gene tree level, considering representative GO terms for particular gene trees. Representative GO terms were selected based on the frequency of occurrence among genes in the gene tree. As gene lengths vary among species and genes in the gene tree, the `GOseq` approach could not be used, and `topGO` was used to detect GO terms that are enriched based on Fisher's exact test. This approach assumes that each gene tree has an equal probability of having genes shared among species that are detected as differentially expressed, however results may be biased by a number of factors, including mean gene length among genes in the gene tree [@young2010gene]. Putative toxin genes were identified using blastp from the ToxProt dataset (http://www.uniprot.org/program/Toxins, last accessed 13 May 2021). See Supplementary Information for R package version numbers.

**_TPM10K_**

For all comparative expression analyses, expression values were normalized using a new method we call transcripts per million 10K (TPM10K). For gene $i$ of a given species, TPM is typically calculated as [@li2009rna]:

$$ TPM_i = \frac{10^6 \times \theta_i}{\ell_i \times \displaystyle\sum_{i=1}^{n}\frac{\theta_j}{\ell_j}} $$

Where $\theta_i$ is the number of the mapped reads to gene $i$, $\ell_i$ is the effective length of the gene, and n is the number of genes in the reference. The intent of this measure is to make libraries comparable within a single species. The sum of TPM values within a library is $10^6$, and the mean is $\frac{10^6}{n}$. One implication of this is that TPM values are not directly comparable across species, since in practice $n$ differs across species. If this were not accounted for, then it could appear, for example, that genes all have lower expression in a species with a more complete reference transcriptome and higher $n$. To account for differences in means among species, we use a new measure, TPM10K, that accounts for differences in $n$:

$$TPM10K_i= \frac{TPM_i \times n}{10^4}$$

Where the sum of TPM10K values within a library is $10^2 \times n$ and the mean is $10^2$. By multiplying by $n$ we are able to account for different sequencing depths among species, and ensure a common mean. As $n$ is large, we divide by an arbitrary number (in this case $10^4$) in order to reduce the magnitude of the expression value.

**_Strict Ortholog Filtering of Trees_**

For Strict Ortholog Filtering of Trees (SOFT) analyses, single copy orthologs were obtained using Orthofinder 2 (v2.4.0) [@emms2019orthofinder]. Linear models used in SOFT analyses were constructed using `lm()`, following methods and code developed by Breschi et al. 2016 [@breschi2016gene], https://github.com/abreschi/Rscripts/blob/master/anova.R.  All SVGs and TVGs are genes for which both species and tissue explain greater than 75% of variance. Additionally, in SVGs, the proportion of variance explained by species is two times greater than that explained by tissues; while in TVGs the proportion of variance explained by tissues is two times greater than that explained by species.

**_Phylogenetic analysis of gene expression_** 

Gene trees were generated using the transcriptomes and genomes from `r species_tree$tip.label %>% length()` species [@munro2018improved] using Agalma v 2.0.0 [@Dunn:2013kw; @guang2021revising]. Following the `treefinder` step of Agalma v 2.0.0, amino acid data were exported and supplied to Orthofinder 2 (v2.3.8) [@emms2019orthofinder] for simultaneous co-estimation of gene trees with the published maximum likelihood species tree [@munro2018improved]. Within Orthofinder 2, the selected multiple sequence alignment method was MAFFT [@katoh2013mafft] and maximum likelihood tree inference method was IQ-tree with the LG+F+R4 substitution model [@nguyen2015iq]. 

Phylogenetic analyses were conducted in R using `geiger`, `ape`, `phytools`, `Rphylopars`, and `hutan` [@ape2004; @doi:10.1093/sysbio/syv055; @goolsby2017rphylopars; @harmon2007geiger; @MEE3:MEE3169]. Phylogenetic trees were visualized in R using `ggtree` and `treeio` [@yu2017ggtree]. 

Gene trees were filtered to exclude trees with a length threshold >2 and that had more than 0.25 branches with a default length value (that are indicative of branch length=0). Gene trees nodes were annotated as speciation events or duplication events, based on assignment by Orthofinder 2 [@emms2019orthofinder]. Speciation nodes were subsequently assigned a node ID equivalent to the species tree node, using species tip names from the gene tree to determine the most common recent ancestor in the species tree, using the `phytools` package. Due to the use of the species-overlap method by Orthofinder 2, some clades of single copy genes were assigned  as speciation events, although the topology is inconsistent with the species tree. To avoid time calibration issues, due to descendant nodes being assigned the same species node ID, descendant speciation nodes with the same species node ID were marked as null, and gene trees with nodes greater than 0.3 null nodes per internal node were excluded -- indicating widespread topological differences with the species tree. Additionally, only trees with one or more speciation events were retained, as speciation events are used for time calibrations. The gene trees were then time calibrated to the species tree using `chronos()` in the `ape` package, so that the branch lengths were scaled to the same equivalent length across all gene trees [@ape2004]. Some gene trees could not be calibrated against the node constraints from the species tree and were discarded, additionally gene trees with roots that exceeded a maximum root depth of 5 were excluded. Tips without expression values were then pruned out of the tree. Gene trees with fewer than three expression values at the tips were discarded, retaining only trees with three or more values.  Pruned and unpruned calibrated gene tree files are available as supplementary data.

We then took the mean TPM10K value for each gene across replicates of the same zooid/tissue within a species and applied a log transformation. Using gene trees with expression values for each gene within a species at the tips, maximum likelihood ancestral trait values were generated at the nodes using the anc.recon() function in `Rphylopars` assuming a Brownian Motion model of evolution (Fig. \@ref(fig:methods-comp), step 2 & 3) [@goolsby2017rphylopars]. As not all zooids are present in all of the species, the trees were pruned down to the subset of tips with expression values for ancestral trait reconstructions. Node values were then added back to the unpruned tree with all of the reconstructed expression values. Change in expression was measured across a branch by taking the difference between a parent node and a child node, and then this difference is scaled by branch length (Fig. \@ref(fig:methods-comp), step 4).

To examine if expression values in our dataset evolve under Brownian Motion (BM), we simulated data set of random expression values using `fastBM()` from the `phytools` package with empirically derived mean and standard-deviation values for each gene tree. Under this model, expression variance accumulates among linages on the gene tree as a function of time, and is used as a model of evolution under drift, as well as some forms of natural selection [@felsenstein1973maximum;@revell2008testing;@revell2008phylogenetic]. 

## Acknowledgements {-}

This work was supported by the National Science Foundation (DEB-1256695 and the Waterman Award). CM was also supported in part by a RI-EPSCoR Fellowship, NSF EPS-1004057. Analyses were conducted with computational resources and services at the Center for Computation and Visualization at Brown University, supported in part by the NSF EPSCoR EPS-1004057 and the State of Rhode Island. Analyses were also conducted at the Yale Center for Research Computing (YCRC). We would like to thank Steve Haddock, Lynne Christianson, and the whole crew of the Western Flyer for their help and expertise. Additional thanks are due to Samuel Church for his assistance during a research cruise, and for productive conversations about comparative gene expression methods. 

\clearpage
## Supplementary data {-}

\setcounter{figure}{0}
\setcounter{table}{0}
\makeatletter
\renewcommand{\thefigure}{S\@arabic\c@figure}
\renewcommand{\thetable}{S\@arabic\c@table}
\makeatother

(ref:zooids-sampled) Phylogeny of the focal species sampled in this study, with details of the traits sampled for each of the species. Phylogeny modified from Munro et al. 2018, @munro2018improved. Black indicates that multiple replicates have been sampled, grey indicates that no or only one replicate has been sampled, and white indicates that this zooid/tissue is not present in this species. The category "unique zooid" indicates that a zooid type that is unique to this species was sampled.

(ref:zooids-sampledshort) Phylogeny of the focal species sampled in this study, with details of the traits sampled for each of the species.

```{r zooids-sampled, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.7\\textheight', out.width='0.7\\textheight', fig.cap= '(ref:zooids-sampled)', fig.scap='(ref:zooids-sampledshort)', fig.align = 'center'}

knitr::include_graphics("figures/DGE_figure.pdf",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

```

```{r PCA-new, echo=FALSE, message=F, warning=FALSE, verbose= FALSE}


diphyes_data <- DESeq2::plotPCA( dge$`K00162-189-HJTYGBBXX-7-NCAGTG`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_diph <- round( 100 * attr(diphyes_data, "percentVar") )

diphyes_pca <- ggplot(diphyes_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"= 15,"Gastrozooid mature"= 0, "Bract mature"= 7, "Gonophore"=3)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_diph[1],"% variance")) +  ylab(paste0("PC2:",percentVar_diph[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

physalia_data<-DESeq2::plotPCA(dge$`HWI-ST625-73-C0JUVACXX-7-TTAGGC`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_phy <- round( 100 * attr(physalia_data, "percentVar") )

physalia_pca <- ggplot(physalia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8,"Tentacular palpon mature"=10,"Tentacular palpon developing"=16,"Gas gland"=9, "Tentacle developing"=23,"Tentacle mature"=5)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_phy[1],"% variance")) +  ylab(paste0("PC2:",percentVar_phy[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

apolemia_data<-DESeq2::plotPCA(dge$`HWI-ST625-159-C4MVCACXX-5-CCGTCC`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_apol <- round( 100 * attr(apolemia_data, "percentVar") )

apolemia_pca <- ggplot(apolemia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_apol[1],"% variance")) +  ylab(paste0("PC2:",percentVar_apol[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

nanomia_data<-DESeq2::plotPCA(dge$`HWI-ST625-51-C02UNACXX-7-NANOMIA`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_nano <- round( 100 * attr(nanomia_data, "percentVar") )

nanomia_pca <- ggplot(nanomia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19,"Nectophore developing "=18, "Palpons mature"=2,"Palpons developing"=17,"Gonodendron male"=13,"Gonodendron female"=10)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_nano[1],"% variance")) +  ylab(paste0("PC2:",percentVar_nano[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

frillagalma_data<-DESeq2::plotPCA(dge$`HWI-ST625-51-C02UNACXX-6-FRILLAGALMA`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_frill <- round( 100 * attr(frillagalma_data, "percentVar") )

frillagalma_pca <- ggplot(frillagalma_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19,"Palpons mature"=2,"Gonodendron male"=13,"Gonodendron female"=10,"Bract developing"= 7)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_frill[1],"% variance")) +  ylab(paste0("PC2:",percentVar_frill[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

bargmannia_data<-DESeq2::plotPCA(dge$`HWI-ST625-51-C02UNACXX-8-BARGMANNIA`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_barg <- round( 100 * attr(bargmannia_data, "percentVar") )

bargmannia_pca <- ggplot(bargmannia_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid white mature"=0,"Gastrozooid yellow mature"=14,"Gastrozooid yellow developing"=12,"Pneumatophore"=8, "Nectophore developing"=19,"Gonodendron male"=13,"Bract developing"= 7)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_barg[1],"% variance")) +  ylab(paste0("PC2:",percentVar_barg[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))

agalma_data<-DESeq2::plotPCA(dge$`HWI-ST625-73-C0JUVACXX-7-AGALMA2`@rld, intgroup=c("treatment", "individual"), returnData=TRUE )
percentVar_agalma <- round( 100 * attr(agalma_data, "percentVar") )

agalma_pca <- ggplot(agalma_data, aes( PC1, PC2, color=individual, shape=treatment ) ) + scale_shape_manual(values=c("Gastrozooid developing"=15,"Gastrozooid mature"=0,"Pneumatophore"=8, "Nectophore developing"=19,"Palpons mature"=2,"Palpons B mature"=6,"Gonodendron male"=13,"Gonodendron female"=10,"Bract developing"= 7)) + geom_point(size=3) + xlab(paste0("PC1: ",percentVar_agalma[1],"% variance")) +  ylab(paste0("PC2:",percentVar_agalma[2],"% variance")) + guides(shape  = guide_legend(order = 1),color = guide_legend(order = 2))


```
(ref:technical-replicate) PCA of regularized log transformed expression counts of technical replicates of two different zooids in *Frillagalma vityazi* from different runs and lanes. Color indicates the run number, shape indicates the zooid, and size factor indicates number of genes sequenced.

(ref:technical-replicateshort)  PCA of regularized log transformed expression counts of technical replicates of two different zooids from different runs and lanes.

```{r technical-replicate, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.cap= '(ref:technical-replicate)', fig.scap='(ref:technical-replicateshort)', fig.align='center'}

knitr::include_graphics("Figures/technical-replicate.pdf",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

```

(ref:pca-diphyes) PCA of regularized log transformed expression counts of zooids/tissues in *Diphyes dispar*. Color indicates the replicate number, shape indicates the zooid.

(ref:pca-diphyesshort) PCA of regularized log transformed expression counts of zooids/tissues in *Diphyes dispar*.

```{r pca-diphyes, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-diphyes)', fig.scap='(ref:pca-diphyesshort)',fig.align='center'}

diphyes_pca

```

(ref:pca-apolemia) PCA of regularized log transformed expression counts of zooids/tissues in *Apolemia lanosa*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-apolemiashort) PCA of regularized log transformed expression counts of treatments in *Apolemia lanosa*.

```{r pca-apolemia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-apolemia)', fig.scap='(ref:pca-apolemiashort)',fig.align='center'}

apolemia_pca

```

(ref:pca-agalma) PCA of regularized log transformed expression counts of zooids/tissues in *Agalma elegans*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-agalmashort) PCA of regularized log transformed expression counts of treatments in *Agalma elegans*.

```{r pca-agalma, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-agalma)', fig.scap='(ref:pca-agalmashort)',fig.align='center'}

agalma_pca

```

(ref:pca-bargmannia) PCA of regularized log transformed expression counts of zooids/tissues in *Bargmannia elongata*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-bargmanniashort) PCA of regularized log transformed expression counts of treatments in *Bargmannia elongata*.

```{r pca-bargmannia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-bargmannia)', fig.scap='(ref:pca-bargmanniashort)',fig.align='center'}

bargmannia_pca

```

(ref:pca-Frillagalma) PCA of regularized log transformed expression counts of zooids/tissues in *Frillagalma vityazi*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-Frillagalmashort) PCA of regularized log transformed expression counts of treatments in *Frillagalma vityazi*.

```{r pca-Frillagalma, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-Frillagalma)', fig.scap='(ref:pca-Frillagalmashort)',fig.align='center'}

frillagalma_pca

```

(ref:pca-nanomia) PCA of regularized log transformed expression counts of zooids/tissues in *Nanomia bijuga*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-nanomiashort) PCA of regularized log transformed expression counts of treatments in *Nanomia bijuga*.

```{r pca-nanomia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-nanomia)', fig.scap='(ref:pca-nanomiashort)',fig.align='center'}

nanomia_pca

```

(ref:pca-physalia) PCA of regularized log transformed expression counts of zooids/tissues in *Physalia physalis*. Color indicates the replicate number, shape indicates the zooid. 

(ref:pca-physaliashort) PCA of regularized log transformed expression counts of treatments in *Physalia physalis*.

```{r pca-physalia, results='asis', warning=FALSE, echo=FALSE, comment=FALSE, message=FALSE, fig.width=7, fig.height=4, fig.cap= '(ref:pca-physalia)', fig.scap='(ref:pca-physaliashort)',fig.align='center'}

physalia_pca

```

(ref:unique-zooids) Number of genes identified as uniquely differentially expressed in zooids within different siphonophore species.  Unique in this case means that the genes are significantly upregulated in these zooids and are not found to be upregulated in any other zooid within the same species. Note that larger numbers of genes are identified in species where fewer zooids were sampled, this is likely due to sampling differences, as opposed to biological differences among species.

```{r unique-zooids, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.cap= '(ref:unique-zooids)', fig.align = 'center'}

unique_zooids %>% dplyr::group_by(zooid,species) %>% dplyr::summarize(number=n()) %>% ggplot()+ geom_point(aes(x=zooid, y=species, size=number),colour="#00BFC4", pch=19) + scale_size_area( max_size = 10 ) + theme(axis.text.x=element_text(angle = 30, hjust = 1))+labs(x="Zooids",y="Species")

```

(ref:transcription-factors) Putative transcription factors found to be significantly upregulated in zooids and the pneumatophore across four different siphonophore species. Transcription factor identity based on blast hit, GO DNA-binding transcription factor activity followed by manual curation. For *Bargmannia elongata* values for mature and developing gastrozooids and values for 'white' gastrozooids. 

```{r transcription-factors, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.height=12, fig.width=12, fig.cap= '(ref:transcription-factors)', fig.align = 'center'}

x %>% ggplot() + geom_point(aes(zooid, short_blast_hit,col=species),position=position_dodge2(width = 0.6)) +labs(y="Transcription factor", x="Zooid")+ theme(axis.text.x=element_text(angle = 30, hjust = 1))

```

(ref:novel-zooids) Unique siphonophore zooids that were sampled for differential gene expression. A. *Bargmannia elongata* with developing "yellow" gastrozooid surrounded by developing "white" gastrozooids. B.  *Bargmannia elongata* stem with mature "white" and "yellow" gastrozooids. C. Zooids in *Physalia physalis*, including multiple developing stages of gastrozooids, tentacular palpon and tentacle; the most mature form of either zooid is not shown. D. Stem of *Agalma elegans*, with gastric palpons, B-palpon and gastrozooid shown.

(ref:novel-zooidsshort) Unique siphonophore zooids that were sampled for differential gene expression.

```{r novel-zooids, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, out.height='0.5\\textheight', out.width='0.5\\textheight', fig.cap= '(ref:novel-zooids)', fig.scap='(ref:novel-zooidsshort)', fig.align = 'center'}

knitr::include_graphics("Figures/Zooid_type.jpg",auto_pdf = getOption("knitr.graphics.auto_pdf", FALSE))

```


(ref:changes-branches) Mean changes of TPM10K along branches across all gene trees that correspond to branches in the species trees, error bar is one standard deviation. Top panel: species phylogeny with branch IDs given as letters.Lower panel: Distribution of changes along a species-equivalent branch in a gene tree, showing mean change in the empirical dataset and in the BM simulation. Treatment type is coded in colour, Gasdev = Developing Gastrozooid, Gasmat= Mature gastrozooid, Necdev= Developing nectophore, Palmat = Mature palpon, Pne= pneumatophore.

```{r changes-branches, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=7.5, fig.height=7, fig.cap='(ref:changes-branches)', fig.align='center'}

grid.arrange(tree_reference,branch_change2,sim_branch_change2,leg2, bottom="Branch",widths = c(2,2,0.5),layout_matrix = layout)

```

(ref:changes-branches2) Mean changes of expression ratios along branches across all gene trees that correspond to branches in the species tree, error bar is one standard deviation. Top panel: species phylogeny with branch IDs given as letters. Lower panel: Distribution of changes along a branch in a gene tree, showing mean change in the empirical dataset and in the BM simulation.  Treatment type is coded in colour, GasdevGasmat = developing gastrozooid / mature gastrozooid, NecdevGasmat = developing nectophore / mature gastrozooid, PalmatGasmat = mature palpon / mature gastrozooid, PneGasmat = Pneumatophore / mature gastrozooid.

```{r changes-branches2, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=7.5, fig.height=7, fig.cap='(ref:changes-branches2)', fig.align='center'}

grid.arrange(tree_reference,branch_change,sim_branch_change,leg, bottom="Branch",widths = c(2,2,0.5),layout_matrix = layout)

```


```{r variance-structure, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=6.5, fig.height=6, fig.pos='h', out.extra = '', fig.cap="Variance of TPM10K change across a branch plotted against the age of the parent node. Branch ID is coded in colour, values are separated by treatment. Gasmat = Mature Gastrozooid, Gasdev = Developing Gastrozooid, Necdev= Developing nectophore, Palmat = Mature palpon , Pne= Pneumatophore", fig.scap="Variance of change across a branch plotted against the age of the parent node.", fig.align='center'}

grid.arrange(var_change_obs_noratio,legend2,var_change_sim_noratio, bottom="Node age of parent", left="Variance of change",widths = c(2,0.25),layout_matrix = layout3)

```

```{r variance-structure2, results='asis', echo=FALSE, message=FALSE, warning=FALSE, verbose= FALSE, fig.width=6.5, fig.height=6, fig.pos='h', out.extra = '', fig.cap="Variance of expression ratio change across a branch plotted against the age of the parent node. Branch ID is coded in colour, values are separated by treatment. GasdevGasmat = Developing Gastrozooid vs Mature Gastrozooid, NecdevGasmat= Developing nectophore vs Mature Gastrozooid, PalmatGasmat = Mature palpon vs Mature Gastrozooid, PneGasmat= Pneumatophore vs Mature Gastrozooid", fig.scap="Variance of change across a branch plotted against the age of the parent node.", fig.align='center'}

grid.arrange(var_change_obs,legend,var_change_sim, bottom="Node age of parent", left="Variance of change",widths = c(2,0.25),layout_matrix = layout2)

```


(ref:wntplot) Top panel: Maximum likelihood gene tree of the Wnt family in Cnidaria, Wnt clades that include siphonophore sequences are highlighted. Bottom panel: Pruned gene tree of wnt genes in siphonophore species with expression values (TPM10K) at the tips and nodes. Developing nectophore/mature gastrozooid expression in green (above node), developing gastrozooid/mature gastrozooid expression in orange (below node). Changes across the branch are shown for developing gastrozooid/mature gastrozooid (orange). Red circles are duplication nodes and blue circles are speciation nodes.

```{r wntplot, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.height=15, fig.width=15, fig.cap='(ref:wntplot)', fig.align = 'center'}

#pull gene tree digest

wnt_digest<-edges_tidy[grep("WNT2B_CHICK Protein Wnt-2b",edges_tidy$blast_hit),] %>% .$gene_tree %>% unique()

nhx_wider<-gene_trees_ace_calibrated[gene_trees_ace_digests==wnt_digest][[1]]
p<-nhx_wider %>% ggtree()+geom_nodepoint(aes( color=Ev )) +xlim(c(0,1.5))
p<-p + geom_cladelabel(node=133, label="Wnt2", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=166, label="Wnt4", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=171, label="Wnt6", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=191, label="Wnt1", align=T, angle=270, hjust='center', offset=0.05, barsize=1)
p<-p + geom_cladelabel(node=204, label="Wnt3", align=T, angle=270, hjust='center', offset=0.05, barsize=1)

nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==wnt_digest][[1]]
tips1<-pull(nhx@data,"Necdev")
tips2<-pull(nhx@data,"Gasmat")
tips11<-(tips1+1)/(tips2+1)

tips3<-pull(nhx@data,"Gasdev")
tips4<-pull(nhx@data,"Gasmat")
tips22<-(tips3+1)/(tips4+1)
p1<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names),pch=1, hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,2))
p1<-p1+geom_text(aes(label=round(tips11,2)),vjust=-1,hjust=1, size=3, col="#A7C136")
p1<-p1+geom_text(aes(label=round(tips22,2)),vjust=1.5,hjust=1, size=3, col="#F4A519")
p1<-p1 + geom_cladelabel(node=23, label="Wnt2", align=T, angle=270, hjust='center', offset=0.5, barsize=1)
p1<-p1 + geom_cladelabel(node=31, label="Wnt6", align=T, angle=270, hjust='center', offset=0.5, barsize=1)
p1<-p1 + geom_cladelabel(node=33, label="Wnt1", align=T, angle=270, hjust='center', offset=0.5, barsize=1)
p1<-p1 + geom_cladelabel(node=34, label="Wnt3", align=T, angle=270, hjust='center', offset=0.5, barsize=1) + theme(legend.position = "none")

edge <- change("Gasdev","Gasmat",nhx)
p3<- p1 %<+% edge + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3, col="#F4A519")  
grid.arrange(p,p3)

```

(ref:wnt4plot) Top panel: Maximum likelihood gene tree of the Wnt4 in Cnidaria. Bottom panel: Pruned gene tree of wnt4 genes in siphonophore species with expression values (TPM10K) at the tips and nodes. Developing nectophore/mature gastrozooid expression in green (above node), developing gastrozooid/mature gastrozooid expression in orange (below node). Changes across the branch are shown for developing gastrozooid/mature gastrozooid (orange). Red circles are duplication nodes and blue circles are speciation nodes.

```{r wnt4plot, echo=FALSE, message=F, warning=FALSE, verbose= FALSE, fig.height=15, fig.width=15, fig.cap='(ref:wnt4plot)', fig.align = 'center'}

#pull gene tree digest

wnt_digest<-edges_tidy[grep("Protein Wnt-4",edges_tidy$blast_hit),] %>% .$gene_tree %>% unique()

nhx_wider<-gene_trees_ace_calibrated[gene_trees_ace_digests==wnt_digest][[1]]
p<-nhx_wider %>% ggtree()+geom_nodepoint(aes( color=Ev )) +xlim(c(0,2)) +geom_tiplab()


nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==wnt_digest][[1]]

nhx<-gene_trees_ace_calibrated_pruned[gene_trees_ace_digests==wnt_digest][[1]]
tips1<-pull(nhx@data,"Necdev")
tips2<-pull(nhx@data,"Gasmat")
tips11<-(tips1+1)/(tips2+1)

tips3<-pull(nhx@data,"Gasdev")
tips4<-pull(nhx@data,"Gasmat")
tips22<-(tips3+1)/(tips4+1)
p1<-nhx %>% ggtree()+ geom_tiplab(aes( label=phy_node_names),pch=1, hjust=-0.1)+geom_nodepoint(aes( color=Ev )) +xlim(c(0,1))
p1<-p1+geom_text(aes(label=round(tips11,2)),vjust=-1,hjust=1, size=3, col="#A7C136")
p1<-p1+geom_text(aes(label=round(tips22,2)),vjust=1.5,hjust=1, size=3, col="#F4A519")


edge <- change("Gasdev","Gasmat",nhx)

p3<- p1 %<+% edge + geom_label(aes(x=branch, label=edge_num),label.size = 0.01,size=3, col="#F4A519")  
grid.arrange(p,p3)

```


\clearpage
### Software versions {-}

This manuscript was computed on `r format( Sys.time(), "%a %b %d %X %Y" )` with the following R package versions.

```{r session_summary, echo=FALSE, include=TRUE, comment=NA}
	sessionInfo()
```

\clearpage
## References {-}
